<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function allows to add miRNA and gene differential expression results
to a MirnaExperiment object. Instead of running
performMirnaDE() and performGeneDE() functions, this one allows to use
differential expression analyses carried out in other ways. Note that it is
possible to manually add differential expression results just for miRNAs or
just for genes. This is particularly useful in order to use the pipeline
implemented in MIRit for proteomic data and for expression data deriving
from different technologies."><title>Manually add differential expression results to a MirnaExperiment object — addDifferentialExpression • MIRit</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Manually add differential expression results to a MirnaExperiment object — addDifferentialExpression"><meta property="og:description" content="This function allows to add miRNA and gene differential expression results
to a MirnaExperiment object. Instead of running
performMirnaDE() and performGeneDE() functions, this one allows to use
differential expression analyses carried out in other ways. Note that it is
possible to manually add differential expression results just for miRNAs or
just for genes. This is particularly useful in order to use the pipeline
implemented in MIRit for proteomic data and for expression data deriving
from different technologies."><meta property="og:image" content="/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MIRit</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/MIRit.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jacopo-ronchi/MIRit/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Manually add differential expression results to a MirnaExperiment object</h1>
      <small class="dont-index">Source: <a href="https://github.com/jacopo-ronchi/MIRit/blob/HEAD/R/differential-expression.R" class="external-link"><code>R/differential-expression.R</code></a></small>
      <div class="d-none name"><code>addDifferentialExpression.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function allows to add miRNA and gene differential expression results
to a <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object. Instead of running
<code><a href="deAnalysis.html">performMirnaDE()</a></code> and <code><a href="deAnalysis.html">performGeneDE()</a></code> functions, this one allows to use
differential expression analyses carried out in other ways. Note that it is
possible to manually add differential expression results just for miRNAs or
just for genes. This is particularly useful in order to use the pipeline
implemented in MIRit for proteomic data and for expression data deriving
from different technologies.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">addDifferentialExpression</span><span class="op">(</span></span>
<span>  <span class="va">mirnaObj</span>,</span>
<span>  mirnaDE <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  geneDE <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  mirna.logFC <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  mirna.pCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  mirna.pAdjustment <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  gene.logFC <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  gene.pCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  gene.pAdjustment <span class="op">=</span> <span class="st">"fdr"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>mirnaObj</dt>
<dd><p>A <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object
containing miRNA and gene data</p></dd>


<dt>mirnaDE</dt>
<dd><p>A <code>data.frame</code> containing the output of miRNA differential
expression analysis. Check the <em>details</em> section to see the required format.
Default is NULL not to add miRNA differential expression results</p></dd>


<dt>geneDE</dt>
<dd><p>A <code>data.frame</code> containing the output of gene differential
expression analysis. Check the <em>details</em> section to see the required format.
Default is NULL not to add gene differential expression results</p></dd>


<dt>mirna.logFC</dt>
<dd><p>The minimum log2 fold change required to consider a miRNA
as differentially expressed. Default is 1, to retain only two-fold
differences</p></dd>


<dt>mirna.pCutoff</dt>
<dd><p>The adjusted p-value cutoff to use for miRNA statistical
significance. The default value is <code>0.05</code></p></dd>


<dt>mirna.pAdjustment</dt>
<dd><p>The p-value correction method for miRNA multiple
testing. It must be one of: <code>fdr</code> (default), <code>BH</code>, <code>none</code>, <code>holm</code>,
<code>hochberg</code>, <code>hommel</code>, <code>bonferroni</code>, <code>BY</code></p></dd>


<dt>gene.logFC</dt>
<dd><p>The minimum log2 fold change required to consider a gene as
differentially expressed. Default is 1, to retain only two-fold differences</p></dd>


<dt>gene.pCutoff</dt>
<dd><p>The adjusted p-value cutoff to use for gene statistical
significance. The default value is <code>0.05</code></p></dd>


<dt>gene.pAdjustment</dt>
<dd><p>The p-value correction method for gene multiple
testing. It must be one of: <code>fdr</code> (default), <code>BH</code>, <code>none</code>, <code>holm</code>,
<code>hochberg</code>, <code>hommel</code>, <code>bonferroni</code>, <code>BY</code></p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object containing differential
expression results. To access these results, the user may run the
<code><a href="deAccessors.html">mirnaDE()</a></code> and <code><a href="deAccessors.html">geneDE()</a></code> functions for miRNAs and genes, respectively.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The following paragraphs briefly explain the formats needed for mirnaDE,
geneDE, and differential expression parameters.</p><div class="section">
<h3 id="mirnade-and-genede">mirnaDE and geneDE<a class="anchor" aria-label="anchor" href="#mirnade-and-genede"></a></h3>


<p><code>mirnaDE</code> and <code>geneDE</code> are two objects of class <code>data.frame</code> containing
the results of miRNA and gene differential expression analysis respectively.
These tables should contain the differential expression results for all
miRNAs/genes analyzed, not just for statistically significant species.</p>
<p>Note that you can individually add differential expression results for
miRNAs and genes. For instance, it is possible to manually add gene
differential expression through this function, while performing miRNA
differential expression through the <code><a href="deAnalysis.html">performMirnaDE()</a></code> function, and vice
versa. In order to only add miRNA or gene differential expression results,
you must leave <code>mirnaDE</code> or <code>geneDE</code> slots to NULL.</p>
<p>All <code>data.frame</code> objects can be used, as long as they have:</p><ul><li><p>One column containing miRNA/gene names (according to miRBase/hgnc
nomenclature). Accepted column names are: <code>ID</code>, <code>Symbol</code>, <code>Gene_Symbol</code>,
<code>Mirna</code>, <code>mir</code>, <code>Gene</code>, <code>gene.symbol</code>, <code>Gene.symbol</code>;</p></li>
<li><p>One column with log2 fold changes. Accepted column names are: <code>logFC</code>,
<code>log2FoldChange</code>, <code>FC</code>, <code>lFC</code>;</p></li>
<li><p>One column with average expression. Accepted column names are: <code>AveExpr</code>,
<code>baseMean</code>, <code>logCPM</code>;</p></li>
<li><p>One column with the p-values resulting from the differential expression
analysis. Accepted column names are: <code>P.Value</code>, <code>pvalue</code>, <code>PValue</code>,
<code>Pvalue</code>;</p></li>
<li><p>One column containing p-values adjusted for multiple testing. Accepted
column names are: <code>adj.P.Val</code>, <code>padj</code>, <code>FDR</code>, <code>fdr</code>, <code>adj</code>, <code>adj.p</code>, <code>adjp</code>.</p></li>
</ul></div>

<div class="section">
<h3 id="differential-expression-cutoffs">Differential expression cutoffs<a class="anchor" aria-label="anchor" href="#differential-expression-cutoffs"></a></h3>


<p><code>mirna.logFC</code>, <code>mirna.pCutoff</code>, <code>mirna.pAdjustment</code>, and <code>gene.logFC</code>,
<code>gene.pCutoff</code>, <code>gene.pAdjustment</code> represent the parameters used to define
the significance of differential expression results. These are needed in
order to inform MIRit about the features that are considered as
differentially expressed.</p>
</div>

    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Jacopo Ronchi, <a href="mailto:jacopo.ronchi@unimib.it">jacopo.ronchi@unimib.it</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># load example data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">geneCounts</span>, package <span class="op">=</span> <span class="st">"MIRit"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">mirnaCounts</span>, package <span class="op">=</span> <span class="st">"MIRit"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># create samples metadata</span></span></span>
<span class="r-in"><span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="st">"primary"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">geneCounts</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    <span class="st">"mirnaCol"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">mirnaCounts</span><span class="op">)</span>, <span class="st">"geneCol"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">geneCounts</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    <span class="st">"disease"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"PTC"</span>, <span class="fl">8</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"NTH"</span>, <span class="fl">8</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    <span class="st">"patient"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Sample_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">8</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># create a 'MirnaExperiment' object</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="MirnaExperiment.html">MirnaExperiment</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    mirnaExpr <span class="op">=</span> <span class="va">mirnaCounts</span>, geneExpr <span class="op">=</span> <span class="va">geneCounts</span>,</span></span>
<span class="r-in"><span>    samplesMetadata <span class="op">=</span> <span class="va">meta</span>, pairedSamples <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># perform miRNA DE with DESeq2 separately</span></span></span>
<span class="r-in"><span><span class="va">dds_m</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSetFromMatrix</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    countData <span class="op">=</span> <span class="va">mirnaCounts</span>, colData <span class="op">=</span> <span class="va">meta</span>, design <span class="op">=</span> <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">disease</span> <span class="op">+</span> <span class="va">patient</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>some variables in design formula are characters, converting to factors</span>
<span class="r-in"><span><span class="va">dds_m</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds_m</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating size factors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating dispersions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> gene-wise dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> mean-dispersion relationship</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> final dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fitting model and testing</span>
<span class="r-in"><span><span class="va">de_m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds_m</span>,</span></span>
<span class="r-in"><span>    contrast <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"disease"</span>, <span class="st">"PTC"</span>, <span class="st">"NTH"</span><span class="op">)</span>, pAdjustMethod <span class="op">=</span> <span class="st">"fdr"</span></span></span>
<span class="r-in"><span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># perform gene DE with DESeq2 separately</span></span></span>
<span class="r-in"><span><span class="va">dds_g</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSetFromMatrix</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    countData <span class="op">=</span> <span class="va">geneCounts</span>, colData <span class="op">=</span> <span class="va">meta</span>, design <span class="op">=</span> <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">disease</span> <span class="op">+</span> <span class="va">patient</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>some variables in design formula are characters, converting to factors</span>
<span class="r-in"><span><span class="va">dds_g</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds_g</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating size factors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating dispersions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> gene-wise dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> mean-dispersion relationship</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> final dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fitting model and testing</span>
<span class="r-in"><span><span class="va">de_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds_g</span>,</span></span>
<span class="r-in"><span>    contrast <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"disease"</span>, <span class="st">"PTC"</span>, <span class="st">"NTH"</span><span class="op">)</span>, pAdjustMethod <span class="op">=</span> <span class="st">"fdr"</span></span></span>
<span class="r-in"><span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># prepare DE tables</span></span></span>
<span class="r-in"><span><span class="va">de_m</span><span class="op">$</span><span class="va">ID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">de_m</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">de_m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="va">de_m</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">de_g</span><span class="op">$</span><span class="va">ID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">de_g</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">de_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="va">de_g</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># add DE results to MirnaExperiment object</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu">addDifferentialExpression</span><span class="op">(</span><span class="va">obj</span>, <span class="va">de_m</span>, <span class="va">de_g</span>,</span></span>
<span class="r-in"><span>    mirna.logFC <span class="op">=</span> <span class="fl">1</span>, mirna.pCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span></span>
<span class="r-in"><span>    gene.logFC <span class="op">=</span> <span class="fl">1</span>, gene.pCutoff <span class="op">=</span> <span class="fl">0.05</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Jacopo Ronchi.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

