<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function allows to remove unwanted batch effects from miRNA and gene
expression matrices. In particular, this function fits a linear model to
miRNA/gene expression levels, and then removes the variability caused by
batch effects. Furthermore, a weighted surrogate variable analysis (WSVA)
can also be included to remove the effects due to surrogate variables.
If batch effects are present, it is crucial to remove them with this
function before moving to correlation analysis."><title>Correct for batch effects in miRNA and gene expression measurements — batchCorrection • MIRit</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Correct for batch effects in miRNA and gene expression measurements — batchCorrection"><meta property="og:description" content="This function allows to remove unwanted batch effects from miRNA and gene
expression matrices. In particular, this function fits a linear model to
miRNA/gene expression levels, and then removes the variability caused by
batch effects. Furthermore, a weighted surrogate variable analysis (WSVA)
can also be included to remove the effects due to surrogate variables.
If batch effects are present, it is crucial to remove them with this
function before moving to correlation analysis."><meta property="og:image" content="/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MIRit</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/MIRit.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jacopo-ronchi/MIRit/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Correct for batch effects in miRNA and gene expression measurements</h1>
      <small class="dont-index">Source: <a href="https://github.com/jacopo-ronchi/MIRit/blob/HEAD/R/batch-correction.R" class="external-link"><code>R/batch-correction.R</code></a></small>
      <div class="d-none name"><code>batchCorrection.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function allows to remove unwanted batch effects from miRNA and gene
expression matrices. In particular, this function fits a linear model to
miRNA/gene expression levels, and then removes the variability caused by
batch effects. Furthermore, a weighted surrogate variable analysis (WSVA)
can also be included to remove the effects due to surrogate variables.
If batch effects are present, it is crucial to remove them with this
function before moving to correlation analysis.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">batchCorrection</span><span class="op">(</span></span>
<span>  <span class="va">mirnaObj</span>,</span>
<span>  <span class="va">assay</span>,</span>
<span>  batch <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  batch2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  covariates <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  includeWsva <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  n.sv <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  weight.by.sd <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>mirnaObj</dt>
<dd><p>A <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object
containing miRNA and gene data</p></dd>


<dt>assay</dt>
<dd><p>The expression matrix to correct. It must be one of <code>genes</code>
and <code>microRNA</code></p></dd>


<dt>batch</dt>
<dd><p>It must be the name of a variable present in the <code>colData</code> of
a <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object (eg. "disease"), or,
alternatively, it must be a <code>character</code>/<code>factor</code> object that defines batch
memberships. See the <strong>details</strong> section for additional information</p></dd>


<dt>batch2</dt>
<dd><p>It must be the name of a variable present in the <code>colData</code> of
a <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object (eg. "disease"), or,
alternatively, it must be a <code>character</code>/<code>factor</code> object that defines
another series of batches that have additive effects to those specified
in <code>batch</code>. See the <strong>details</strong> section for additional information</p></dd>


<dt>covariates</dt>
<dd><p>Additional numeric covariates that we want to correct for.
It must be a <code>character</code> vector containing the names of numeric variables
present in the <code>colData</code> of a <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code>
object (eg. <code>c("age", "RIN", "quantity")</code>), or, alternatively, it must be a
simple <code>matrix</code> object. See the <strong>details</strong> section for additional
information</p></dd>


<dt>includeWsva</dt>
<dd><p>Logical, whether to correct for surrogate variables or
not. Default is FALSE</p></dd>


<dt>n.sv</dt>
<dd><p>The number of surrogate variables to estimate</p></dd>


<dt>weight.by.sd</dt>
<dd><p>Logical, whether to specifically tune the surrogate
variables to the more variable genes or not. Default is TRUE</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object containing batch
effect-corrected expression matrices.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Batch effects consist in unwanted sources of technical variation that
confound expression variability and limit downstream analyses. Since the
reliability of biological conclusions of integrative miRNA-mRNA analyses
depends on the association between miRNA and gene expression levels, it is
pivotal to ensure that expression measurements are not affected by technical
variations. In this regard, if batch effects are noticed in the data, the
user should run this function before using the <code><a href="mirnaIntegration.html">mirnaIntegration()</a></code>
function to perform a correlation analysis.</p>
<p>Usually, given a <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object, the user
should specify:</p><ul><li><p>the <code>assay</code> from which we want to remove batch effects (one between
<code>genes</code> and <code>microRNA</code>);</p></li>
<li><p>the <code>batch</code> variable, which is a variable that defines the different
batches;</p></li>
<li><p>the <code>batch2</code> variable, which can be included to correct for a second
series of batches that have additive effects to those specified in <code>batch</code>;</p></li>
<li><p>the <code>covariates</code> variables, which allows correction for one or more
continuous numeric effects.</p></li>
</ul><p>In particular, <code>batch</code> and <code>batch2</code> could be provided as the names of
covariates included in the <code>colData</code> of a
<code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object. Alternatively, they can
be <code>character</code>/<code>factor</code> objects that declare batch memberships.
Similarly, <code>covariates</code> can be supplied as a vector containing the names
of numeric variables listed in the <code>colData</code> of
<code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> objects, or they can be provided
as a simple <code>matrix</code>.</p>
<p>Additionally, the influence of unknown sources of technical variation can
be removed by including surrogate variables estimated through WSVA. To do
so, we can set <code>includeWsva</code> to TRUE, and then we can specify the number of
surrogate variables to use through the <code>n.sv</code> parameter. Further, the
surrogate variables can be tuned to the more variable genes by setting
<code>weight.by.sd</code> to TRUE.</p>
<p>Please note that we only recommend to remove batch effects directly from
expression measurements prior to correlation analysis. This function
can't be used to remove batch effects before differential expression
analysis, because for that purpose, it is better to include batch variables
in the linear model. In this way, we do not underestimate the residual
degrees of freedom, so that the calculated standard errors, t-statistics
and p-values are not overoptimistic.</p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>To estimate surrogate variables and to remove batch effects from expression
data, MIRit uses the <code><a href="https://rdrr.io/pkg/limma/man/wsva.html" class="external-link">limma::wsva()</a></code> and <code><a href="https://rdrr.io/pkg/limma/man/removeBatchEffect.html" class="external-link">limma::removeBatchEffect()</a></code>
functions, respectively.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W, Smyth GK (2015). “limma
powers differential expression analyses for RNA-sequencing and microarray
studies.” Nucleic Acids Research, 43(7), e47. <a href="doi:10.1093/nar/gkv007">doi:10.1093/nar/gkv007</a>.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Jacopo Ronchi, <a href="mailto:jacopo.ronchi@unimib.it">jacopo.ronchi@unimib.it</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># load example MirnaExperiment object</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="loadExamples.html">loadExamples</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># correct batch effects due to the patient from miRNA expression matrix</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu">batchCorrection</span><span class="op">(</span><span class="va">obj</span>, <span class="st">"microRNA"</span>, batch <span class="op">=</span> <span class="st">"patient"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Jacopo Ronchi.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>MIRit is provided without <strong>any warranty</strong>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

