<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Correct for batch effects in miRNA and gene expression measurements — batchCorrection • MIRit</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Correct for batch effects in miRNA and gene expression measurements — batchCorrection"><meta property="og:description" content="This function allows to remove unwanted batch effects from miRNA and gene
expression matrices. In particular, this function fits a linear model to
miRNA/gene expression levels, and then removes the variability caused by
batch effects. Furthermore, a weighted surrogate variable analysis (WSVA)
can also be included to remove the effects due to surrogate variables.
If batch effects are present, it is crucial to remove them with this
function before moving to correlation analysis."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MIRit</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MIRit.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/jacopo-ronchi/MIRit/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Correct for batch effects in miRNA and gene expression measurements</h1>
    <small class="dont-index">Source: <a href="https://github.com/jacopo-ronchi/MIRit/blob/HEAD/R/batch-correction.R" class="external-link"><code>R/batch-correction.R</code></a></small>
    <div class="hidden name"><code>batchCorrection.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function allows to remove unwanted batch effects from miRNA and gene
expression matrices. In particular, this function fits a linear model to
miRNA/gene expression levels, and then removes the variability caused by
batch effects. Furthermore, a weighted surrogate variable analysis (WSVA)
can also be included to remove the effects due to surrogate variables.
If batch effects are present, it is crucial to remove them with this
function before moving to correlation analysis.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">batchCorrection</span><span class="op">(</span></span>
<span>  <span class="va">mirnaObj</span>,</span>
<span>  <span class="va">assay</span>,</span>
<span>  batch <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  batch2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  covariates <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  includeWsva <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  n.sv <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  weight.by.sd <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>mirnaObj</dt>
<dd><p>A <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object
containing miRNA and gene data</p></dd>


<dt>assay</dt>
<dd><p>The expression matrix to correct. It must be one of <code>genes</code>
and <code>microRNA</code></p></dd>


<dt>batch</dt>
<dd><p>It must be the name of a variable present in the <code>colData</code> of
a <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object (eg. "disease"), or,
alternatively, it must be a <code>character</code>/<code>factor</code> object that defines batch
memberships. See the <strong>details</strong> section for additional information</p></dd>


<dt>batch2</dt>
<dd><p>It must be the name of a variable present in the <code>colData</code> of
a <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object (eg. "disease"), or,
alternatively, it must be a <code>character</code>/<code>factor</code> object that defines
another series of batches that have additive effects to those specified
in <code>batch</code>. See the <strong>details</strong> section for additional information</p></dd>


<dt>covariates</dt>
<dd><p>Additional numeric covariates that we want to correct for.
It must be a <code>character</code> vector containing the names of numeric variables
present in the <code>colData</code> of a <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code>
object (eg. <code>c("age", "RIN", "quantity")</code>), or, alternatively, it must be a
simple <code>matrix</code> object. See the <strong>details</strong> section for additional
information</p></dd>


<dt>includeWsva</dt>
<dd><p>Logical, whether to correct for surrogate variables or
not. Default is FALSE</p></dd>


<dt>n.sv</dt>
<dd><p>The number of surrogate variables to estimate</p></dd>


<dt>weight.by.sd</dt>
<dd><p>Logical, whether to specifically tune the surrogate
variables to the more variable genes or not. Default is TRUE</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object containing batch
effect-corrected expression matrices.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Batch effects consist in unwanted sources of technical variation that
confound expression variability and limit downstream analyses. Since the
reliability of biological conclusions of integrative miRNA-mRNA analyses
depends on the association between miRNA and gene expression levels, it is
pivotal to ensure that expression measurements are not affected by technical
variations. In this regard, if batch effects are noticed in the data, the
user should run this function before using the <code><a href="mirnaIntegration.html">mirnaIntegration()</a></code>
function to perform a correlation analysis.</p>
<p>Usually, given a <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object, the user
should specify:</p><ul><li><p>the <code>assay</code> from which we want to remove batch effects (one between
<code>genes</code> and <code>microRNA</code>);</p></li>
<li><p>the <code>batch</code> variable, which is a variable that defines the different
batches;</p></li>
<li><p>the <code>batch2</code> variable, which can be included to correct for a second
series of batches that have additive effects to those specified in <code>batch</code>;</p></li>
<li><p>the <code>covariates</code> variables, which allows correction for one or more
continuous numeric effects.</p></li>
</ul><p>In particular, <code>batch</code> and <code>batch2</code> could be provided as the names of
covariates included in the <code>colData</code> of a
<code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object. Alternatively, they can
be <code>character</code>/<code>factor</code> objects that declare batch memberships.
Similarly, <code>covariates</code> can be supplied as a vector containing the names
of numeric variables listed in the <code>colData</code> of
<code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> objects, or they can be provided
as a simple <code>matrix</code>.</p>
<p>Additionally, the influence of unknown sources of technical variation can
be removed by including surrogate variables estimated through WSVA. To do
so, we can set <code>includeWsva</code> to TRUE, and then we can specify the number of
surrogate variables to use through the <code>n.sv</code> parameter. Further, the
surrogate variables can be tuned to the more variable genes by setting
<code>weight.by.sd</code> to TRUE.</p>
<p>Please note that we only recommend to remove batch effects directly from
expression measurements prior to correlation analysis. This function
can't be used to remove batch effects before differential expression
analysis, because for that purpose, it is better to include batch variables
in the linear model. In this way, we do not underestimate the residual
degrees of freedom, so that the calculated standard errors, t-statistics
and p-values are not overoptimistic.</p>
    </div>
    <div id="note">
    <h2>Note</h2>
    <p>To estimate surrogate variables and to remove batch effects from expression
data, MIRit uses the <code><a href="https://rdrr.io/pkg/limma/man/wsva.html" class="external-link">limma::wsva()</a></code> and <code><a href="https://rdrr.io/pkg/limma/man/removeBatchEffect.html" class="external-link">limma::removeBatchEffect()</a></code>
functions, respectively.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W, Smyth GK (2015). “limma
powers differential expression analyses for RNA-sequencing and microarray
studies.” Nucleic Acids Research, 43(7), e47. <a href="doi:10.1093/nar/gkv007">doi:10.1093/nar/gkv007</a>.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Jacopo Ronchi, <a href="mailto:jacopo.ronchi@unimib.it">jacopo.ronchi@unimib.it</a></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># load example MirnaExperiment object</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="loadExamples.html">loadExamples</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># correct batch effects due to the patient from miRNA expression matrix</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu">batchCorrection</span><span class="op">(</span><span class="va">obj</span>, <span class="st">"microRNA"</span>, batch <span class="op">=</span> <span class="st">"patient"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jacopo Ronchi.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

