<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Perform an enrichment analysis of integrated microRNA targets — enrichTargets • MIRit</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Perform an enrichment analysis of integrated microRNA targets — enrichTargets"><meta property="og:description" content="This function allows to perform an over-representation analysis (ORA) of
integrated miRNA targets in order to explore the biological effects of
targets that are statistically associated/correlated with DE-miRNAs. The
enrichment analysis can be performed using different databases, namely
Gene Ontology (GO), Kyoto Encyclopedia of Genes and Genomes (KEGG), MsigDB,
WikiPathways, Reactome, Enrichr, Disease Ontology (DO), Network of Cancer
Genes (NCG), DisGeNET, and COVID19."><meta property="og:image" content="/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MIRit</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MIRit.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/jacopo-ronchi/MIRit/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Perform an enrichment analysis of integrated microRNA targets</h1>
    <small class="dont-index">Source: <a href="https://github.com/jacopo-ronchi/MIRit/blob/HEAD/R/enrichment.R" class="external-link"><code>R/enrichment.R</code></a></small>
    <div class="hidden name"><code>enrichTargets.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function allows to perform an over-representation analysis (ORA) of
integrated miRNA targets in order to explore the biological effects of
targets that are statistically associated/correlated with DE-miRNAs. The
enrichment analysis can be performed using different databases, namely
Gene Ontology (GO), Kyoto Encyclopedia of Genes and Genomes (KEGG), MsigDB,
WikiPathways, Reactome, Enrichr, Disease Ontology (DO), Network of Cancer
Genes (NCG), DisGeNET, and COVID19.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">enrichTargets</span><span class="op">(</span></span>
<span>  <span class="va">mirnaObj</span>,</span>
<span>  database <span class="op">=</span> <span class="st">"GO"</span>,</span>
<span>  category <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  organism <span class="op">=</span> <span class="st">"Homo sapiens"</span>,</span>
<span>  pCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  pAdjustment <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  minSize <span class="op">=</span> <span class="fl">10L</span>,</span>
<span>  maxSize <span class="op">=</span> <span class="fl">500L</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>mirnaObj</dt>
<dd><p>A <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object
containing miRNA and gene data</p></dd>


<dt>database</dt>
<dd><p>The name of the database used for the enrichment analysis.
It must be one of: <code>GO</code>, <code>KEGG</code>, <code>MsigDB</code>, <code>WikiPathways</code>, <code>Reactome</code>,
<code>Enrichr</code>, <code>DO</code>, <code>NCG</code>, <code>DisGeNET</code>, <code>COVID19</code>. Default is <code>GO</code></p></dd>


<dt>category</dt>
<dd><p>The desired subcategory of gene sets present in <code>database</code>.
Please, see the <em>details</em> section to check the available categories for
each database. Default is NULL to use default categories</p></dd>


<dt>organism</dt>
<dd><p>The name of the organism under consideration. The different
databases have different supported organisms. To see the list of supported
organisms for a given database, use the <code><a href="supportedOrganisms.html">supportedOrganisms()</a></code> function.
Default is <code>Homo sapiens</code></p></dd>


<dt>pCutoff</dt>
<dd><p>The adjusted p-value cutoff to use for statistical
significance. The default value is <code>0.05</code></p></dd>


<dt>pAdjustment</dt>
<dd><p>The p-value correction method for multiple testing. It
must be one of: <code>fdr</code> (default), <code>BH</code>, <code>none</code>, <code>holm</code>, <code>hochberg</code>, <code>hommel</code>,
<code>bonferroni</code>, <code>BY</code></p></dd>


<dt>minSize</dt>
<dd><p>The minimum size for a gene set. All gene sets containing
less than this number of genes will not be considered. Default is 10</p></dd>


<dt>maxSize</dt>
<dd><p>The maximum size for a gene set. All gene sets containing
more than this number of genes will not be considered. Default is 500</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>This function produces a <code>list</code> object with two elements, namely
'upregulated' and 'downregulated', each containing a
<code><a href="FunctionalEnrichment-class.html">FunctionalEnrichment</a></code></p>


<p>object storing enrichment results of upregulated and downregulated target
genes, respectively.</p>


<p>To access results of <code><a href="FunctionalEnrichment-class.html">FunctionalEnrichment</a></code></p>


<p>objects, the user can use the <code><a href="enrichmentResults.html">enrichmentResults()</a></code> function. Additionally,
MIRit provides several functions to graphically represent enrichment
analyses, including <code><a href="enrichmentBarplot.html">enrichmentBarplot()</a></code>, and <code><a href="enrichmentDotplot.html">enrichmentDotplot()</a></code>.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>For each database, different organisms are supported. To check the supported
organisms for a given database, MIRit provides the <code><a href="supportedOrganisms.html">supportedOrganisms()</a></code>
function.</p>
<p>Moreover, since different database support multiple subcategories, the
<code>category</code> parameter can be set to specify the desired resource. For
specific information regarding the available categories for the different
databases, check the <em>details</em> section of the <code><a href="enrichGenes.html">enrichGenes()</a></code> documentation.</p>
    </div>
    <div id="note">
    <h2>Note</h2>
    <p>To download gene sets from the above mentioned databases, MIRit uses the
<code>geneset</code> R package. Moreover, to perform ORA, MIRit implements the
<code>fgsea</code> package in Bioconductor.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Liu, Y., Li, G. Empowering biologists to decode omics data: the Genekitr R
package and web server. BMC Bioinformatics 24, 214 (2023).
<a href="https://doi.org/10.1186/s12859-023-05342-9" class="external-link">https://doi.org/10.1186/s12859-023-05342-9</a>.</p>
<p>Korotkevich G, Sukhov V, Sergushichev A (2019). “Fast gene set enrichment
analysis.” bioRxiv. doi:10.1101/060012,
<a href="http://biorxiv.org/content/early/2016/06/20/060012" class="external-link">http://biorxiv.org/content/early/2016/06/20/060012</a>.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Jacopo Ronchi, <a href="mailto:jacopo.ronchi@unimib.it">jacopo.ronchi@unimib.it</a></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># load example MirnaExperiment object</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="loadExamples.html">loadExamples</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># perform enrichment analysis of integrated targets with DO</span></span></span>
<span class="r-in"><span><span class="va">targets_enrichment</span> <span class="op">&lt;-</span> <span class="fu">enrichTargets</span><span class="op">(</span><span class="va">obj</span>, database <span class="op">=</span> <span class="st">"DO"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing the appropriate gene set...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Some ID occurs one-to-many match, like "26476, 127068, 101060321"...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 99.06% genes are mapped to symbol</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing the enrichment of upregulated genes...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing the enrichment of downregulated genes...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The enrichment of genes reported 113 significantly enriched terms for downregulated genes and 0 for upregulated genes.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># extract enrichment results of downregulated targets</span></span></span>
<span class="r-in"><span><span class="va">enr_down</span> <span class="op">&lt;-</span> <span class="va">targets_enrichment</span><span class="op">[[</span><span class="st">"downregulated"</span><span class="op">]</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># extract enrichment results as a data.frame</span></span></span>
<span class="r-in"><span><span class="va">enr_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="enrichmentResults.html">enrichmentResults</a></span><span class="op">(</span><span class="va">enr_down</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># create a dotplot of enriched terms</span></span></span>
<span class="r-in"><span><span class="fu"><a href="enrichmentDotplot.html">enrichmentDotplot</a></span><span class="op">(</span><span class="va">enr_down</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="enrichTargets-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jacopo Ronchi.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

