<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Find disease-associated SNPs occurring at DE-miRNA loci — findMirnaSNPs • MIRit</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Find disease-associated SNPs occurring at DE-miRNA loci — findMirnaSNPs"><meta property="og:description" content="This function allows to identify disease-associated genomic variants
affecting differentially expressed miRNA genes or their host genes.
To do so, this function uses gwasrapidd to retrieve SNPs-disease
associations, and then retains only SNPs that affect DE-miRNA genes or their
relative host genes (for intronic miRNAs)."><meta property="og:image" content="/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MIRit</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MIRit.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/jacopo-ronchi/MIRit/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Find disease-associated SNPs occurring at DE-miRNA loci</h1>
    <small class="dont-index">Source: <a href="https://github.com/jacopo-ronchi/MIRit/blob/HEAD/R/association.R" class="external-link"><code>R/association.R</code></a></small>
    <div class="hidden name"><code>findMirnaSNPs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function allows to identify disease-associated genomic variants
affecting differentially expressed miRNA genes or their host genes.
To do so, this function uses <code>gwasrapidd</code> to retrieve SNPs-disease
associations, and then retains only SNPs that affect DE-miRNA genes or their
relative host genes (for intronic miRNAs).</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">findMirnaSNPs</span><span class="op">(</span><span class="va">mirnaObj</span>, <span class="va">diseaseEFO</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>mirnaObj</dt>
<dd><p>A <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object
containing miRNA and gene data</p></dd>


<dt>diseaseEFO</dt>
<dd><p>The EFO identifier of a disease of interest. This can be
identified with the <code><a href="searchDisease.html">searchDisease()</a></code> function</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A <code>data.frame</code> containing details about disease-SNPs and the associated
differentially expressed miRNAs:</p><ul><li><p><code>variant</code> contains SNP identifiers;</p></li>
<li><p><code>gene</code> defines the gene affected by a disease-SNP (it may be the miRNA
gene itself or the host gene of an intronic miRNA);</p></li>
<li><p><code>miRNA.gene</code> specifies the DE-miRNA gene present;</p></li>
<li><p><code>miRNA.precursor</code> specifies the name of the miRNA precursor affected
by disease-SNPs;</p></li>
<li><p><code>chr</code> indicates the chromosome of SNPs;</p></li>
<li><p><code>position</code> shows the SNP position;</p></li>
<li><p><code>allele</code> displays possible alleles for this SNPs;</p></li>
<li><p><code>distance</code> specifies the distance between SNPs and miRNAs;</p></li>
<li><p><code>is_upstream</code> indicates whether SNP is upstream of miRNA gene;</p></li>
<li><p><code>is_downstream</code> indicates whether SNP is downstream of miRNA gene;</p></li>
<li><p><code>mirnaStrand</code> shows the strand;</p></li>
<li><p><code>mirnaStartPosition</code> displays the start position of DE-miRNA gene;</p></li>
<li><p><code>mirnaEndPosition</code> displays the end position of DE-miRNA gene.</p></li>
</ul></div>
    <div id="details">
    <h2>Details</h2>
    <p>SNPs occurring within miRNAs may have important effects on the biological
function of these transcripts. Indeed, a SNP present within a miRNA gene
might alter its expression or the spectrum of miRNA targets.</p>
<p>To retrieve disease-SNPs, this function uses <code>gwasrapidd</code> package, which
directly queries the NHGRI-EBI Catalog of published genome-wide association
studies. After running this function, the user can use the <code><a href="mirVariantPlot.html">mirVariantPlot()</a></code>
function to produce a trackplot for visualizing the genomic location of
SNPs within miRNA genes.</p>
    </div>
    <div id="note">
    <h2>Note</h2>
    <p>To retrieve disease-associated SNPs, this function makes use of the
<code>gwasrapidd</code> package.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Ramiro Magno, Ana-Teresa Maia, gwasrapidd: an R package to query, download
and wrangle GWAS catalog data, Bioinformatics, Volume 36, Issue 2, January
2020, Pages 649–650, <a href="https://doi.org/10.1093/bioinformatics/btz605" class="external-link">https://doi.org/10.1093/bioinformatics/btz605</a></p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Jacopo Ronchi, <a href="mailto:jacopo.ronchi@unimib.it">jacopo.ronchi@unimib.it</a></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># load example MirnaExperiment object</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="loadExamples.html">loadExamples</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="co"># search disease</span></span></span>
<span class="r-in"><span><span class="fu"><a href="searchDisease.html">searchDisease</a></span><span class="op">(</span><span class="st">"Alzheimer disease"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for cached EFO traits...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Downloading EFO traits, this may take some minutes...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Searching for disease: Alzheimer disease</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Alzheimer's disease biomarker measurement" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2] "Alzheimer's disease neuropathologic change"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3] "Alzheimer disease"                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [4] "late-onset Alzheimers disease"             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [5] "family history of Alzheimer’s disease"     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [6] "age of onset of Alzheimer disease"         </span>
<span class="r-in"><span><span class="va">disId</span> <span class="op">&lt;-</span> <span class="st">"Alzheimer disease"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># retrieve associated SNPs</span></span></span>
<span class="r-in"><span><span class="va">association</span> <span class="op">&lt;-</span> <span class="fu">findMirnaSNPs</span><span class="op">(</span><span class="va">obj</span>, <span class="va">disId</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Querying GWAS Catalog, this may take some time...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finding genomic information of differentially expressed miRNAs...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> No disease-related SNPs are present within DE-miRNA loci.</span>
<span class="r-in"><span><span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jacopo Ronchi.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

