<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This is the constructor function that allows to easily create objects of
class MirnaExperiment. This function requires as
inputs miRNA and gene expression matrices, as well as sample metadata."><title>The constructor function for MirnaExperiment — MirnaExperiment • MIRit</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="The constructor function for MirnaExperiment — MirnaExperiment"><meta property="og:description" content="This is the constructor function that allows to easily create objects of
class MirnaExperiment. This function requires as
inputs miRNA and gene expression matrices, as well as sample metadata."><meta property="og:image" content="/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MIRit</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/MIRit.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jacopo-ronchi/MIRit/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>The constructor function for <a href="MirnaExperiment-class.html">MirnaExperiment</a></h1>
      <small class="dont-index">Source: <a href="https://github.com/jacopo-ronchi/MIRit/blob/HEAD/R/AllClasses.R" class="external-link"><code>R/AllClasses.R</code></a></small>
      <div class="d-none name"><code>MirnaExperiment.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This is the constructor function that allows to easily create objects of
class <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code>. This function requires as
inputs miRNA and gene expression matrices, as well as sample metadata.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">MirnaExperiment</span><span class="op">(</span><span class="va">mirnaExpr</span>, <span class="va">geneExpr</span>, <span class="va">samplesMetadata</span>, pairedSamples <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>mirnaExpr</dt>
<dd><p>A <code>matrix</code> object containing microRNA expression levels.
Other objects coercible to <code>matrix</code> are also accepted (e.g. <code>data.frame</code>).
This object must be structured as specified in the <em>details</em> section</p></dd>


<dt>geneExpr</dt>
<dd><p>A <code>matrix</code> object containing gene expression levels.
Other objects coercible to <code>matrix</code> are also accepted (e.g. <code>data.frame</code>).
This object must be structured as specified in the <em>details</em> section</p></dd>


<dt>samplesMetadata</dt>
<dd><p>A <code>data.frame</code> object containing information about
samples used for microRNA and gene expression profiling. For further
information see the <em>details</em> section</p></dd>


<dt>pairedSamples</dt>
<dd><p>Logical, whether miRNA and gene expression levels
derive from the same subjects or not. Check the <em>details</em> section for
additional instructions. Default is <code>TRUE</code></p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A valid <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object containing
information about miRNA and gene expression.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function requires data to be prepared as described below.</p><div class="section">
<h3 id="mirnaexpr-and-geneexpr">mirnaExpr and geneExpr<a class="anchor" aria-label="anchor" href="#mirnaexpr-and-geneexpr"></a></h3>


<p><code>mirnaExpr</code> and <code>geneExpr</code> must be <code>matrix</code> objects (or objects coercible
to one) that contain miRNA and gene expression values, respectively.
Rows must represent the different miRNAs/genes analyzed while columns must
represent the different samples in study. For <code>mirnaExpr</code>, row names must
contain miRNA names according to miRBase nomenclature, whereas for
<code>geneExpr</code>, row names must contain gene symbols according to hgnc
nomenclature. The values contained in these objects can derive from both
microarray and RNA-Seq experiments.</p>
<p>For NGS experiments, <code>mirnaExpr</code> and <code>geneExpr</code> should just be un-normalized
count matrices. Instead, for microarray experiments, data should be
normalized and log2 transformed, for example with the RMA algorithm.</p>
</div>

<div class="section">
<h3 id="samplesmetadata">samplesMetadata<a class="anchor" aria-label="anchor" href="#samplesmetadata"></a></h3>


<p><code>samplesMetadata</code> must be a <code>data.frame</code> object containing information about
samples used for miRNA profiling and for gene expression analysis.
Specifically, this data.frame must contain:</p><ul><li><p>A column named <code>primary</code>, specifying an identifier for each sample;</p></li>
<li><p>A column named <code>mirnaCol</code>, containing the column names used for each
sample in the <code>mirnaExpr</code> object;</p></li>
<li><p>A column named <code>geneCol</code>, containing the column names used for each
sample in the <code>geneExpr</code> object;</p></li>
<li><p>Other eventual columns that define specific sample metadata, such as
disease condition, age, sex and so on...</p></li>
</ul><p>For unpaired samples, NAs can be used for missing entries in
<code>mirnaCol</code>/<code>geneCol</code>.</p>
</div>

<div class="section">
<h3 id="pairedsamples">pairedSamples<a class="anchor" aria-label="anchor" href="#pairedsamples"></a></h3>


<p>MicroRNA and gene expression measurements may derive from the same subjects
(i.e. samples used to generate both miRNA and gene expression data) or from
different individuals (i.e. miRNA expression assayed on a group of samples
and gene expression retrieved from a different group of samples).
<code>pairedSamples</code> is a <code>logical</code> parameter that defines the relationship
between miRNA and gene expression measurements. It must be <code>TRUE</code> if data
derive from the same individuals, while it must be <code>FALSE</code> when data derive
from different subjects.</p>
</div>

    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Jacopo Ronchi, <a href="mailto:jacopo.ronchi@unimib.it">jacopo.ronchi@unimib.it</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># load example data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">geneCounts</span>, package <span class="op">=</span> <span class="st">"MIRit"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">mirnaCounts</span>, package <span class="op">=</span> <span class="st">"MIRit"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># create samples metadata</span></span></span>
<span class="r-in"><span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="st">"primary"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">geneCounts</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    <span class="st">"mirnaCol"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">mirnaCounts</span><span class="op">)</span>, <span class="st">"geneCol"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">geneCounts</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    <span class="st">"disease"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"PTC"</span>, <span class="fl">8</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"NTH"</span>, <span class="fl">8</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    <span class="st">"patient"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Sample_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">8</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># create a 'MirnaExperiment' object</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu">MirnaExperiment</span><span class="op">(</span></span></span>
<span class="r-in"><span>    mirnaExpr <span class="op">=</span> <span class="va">mirnaCounts</span>, geneExpr <span class="op">=</span> <span class="va">geneCounts</span>,</span></span>
<span class="r-in"><span>    samplesMetadata <span class="op">=</span> <span class="va">meta</span>, pairedSamples <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Jacopo Ronchi.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>MIRit is provided without <strong>any warranty</strong>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

