<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function allows to identify microRNAs that are significantly
associated/correlated with their targets. The principle is that, since the
biological role of miRNAs is mainly to negatively regulate gene expression
post-transcriptionally, the expression of a microRNA should be negatively
correlated with the expression of its targets. To test this assumption for
matched-sample data, this function performs a correlation analysis. On the
other hand, for unpaired data, it offers different one-sided association
tests to estimate if targets of down-regulated miRNAs are enriched in
up-regulated genes and vice versa. Additionally, for unpaired data, miRNA
effects on target gene expression can also be quantified through a fast
approximation to rotation gene-set testing ('fry' method). For correlation
analyses, the default behavior is to use Spearman's correlation analysis,
whereas for association tests the default option makes use of a one-sided
Boschloo's exact test. See the details section for further information."><title>Integrate microRNA and gene expression — mirnaIntegration • MIRit</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Integrate microRNA and gene expression — mirnaIntegration"><meta property="og:description" content="This function allows to identify microRNAs that are significantly
associated/correlated with their targets. The principle is that, since the
biological role of miRNAs is mainly to negatively regulate gene expression
post-transcriptionally, the expression of a microRNA should be negatively
correlated with the expression of its targets. To test this assumption for
matched-sample data, this function performs a correlation analysis. On the
other hand, for unpaired data, it offers different one-sided association
tests to estimate if targets of down-regulated miRNAs are enriched in
up-regulated genes and vice versa. Additionally, for unpaired data, miRNA
effects on target gene expression can also be quantified through a fast
approximation to rotation gene-set testing ('fry' method). For correlation
analyses, the default behavior is to use Spearman's correlation analysis,
whereas for association tests the default option makes use of a one-sided
Boschloo's exact test. See the details section for further information."><meta property="og:image" content="/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MIRit</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/MIRit.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jacopo-ronchi/MIRit/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Integrate microRNA and gene expression</h1>
      <small class="dont-index">Source: <a href="https://github.com/jacopo-ronchi/MIRit/blob/HEAD/R/integration.R" class="external-link"><code>R/integration.R</code></a></small>
      <div class="d-none name"><code>mirnaIntegration.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function allows to identify microRNAs that are significantly
associated/correlated with their targets. The principle is that, since the
biological role of miRNAs is mainly to negatively regulate gene expression
post-transcriptionally, the expression of a microRNA should be negatively
correlated with the expression of its targets. To test this assumption for
matched-sample data, this function performs a correlation analysis. On the
other hand, for unpaired data, it offers different one-sided association
tests to estimate if targets of down-regulated miRNAs are enriched in
up-regulated genes and vice versa. Additionally, for unpaired data, miRNA
effects on target gene expression can also be quantified through a fast
approximation to rotation gene-set testing ('fry' method). For correlation
analyses, the default behavior is to use Spearman's correlation analysis,
whereas for association tests the default option makes use of a one-sided
Boschloo's exact test. See the <em>details</em> section for further information.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">mirnaIntegration</span><span class="op">(</span></span>
<span>  <span class="va">mirnaObj</span>,</span>
<span>  test <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  pCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  pAdjustment <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  corMethod <span class="op">=</span> <span class="st">"spearman"</span>,</span>
<span>  corCutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  associationMethod <span class="op">=</span> <span class="st">"boschloo"</span>,</span>
<span>  nuisanceParam <span class="op">=</span> <span class="fl">100</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>mirnaObj</dt>
<dd><p>A <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object
containing miRNA and gene data</p></dd>


<dt>test</dt>
<dd><p>The statistical test to evaluate the association between miRNAs
and genes. It must be one of <code>auto</code> (default), to automatically determine
the appropriate statistical test; <code>correlation</code>, to perform a correlation
analysis; <code>association</code>, to perform a one-sided association test; <code>fry</code> to
perform the integrative analysis through rotation gene-set testing</p></dd>


<dt>pCutoff</dt>
<dd><p>The adjusted p-value cutoff to use for statistical
significance. The default value is <code>0.05</code></p></dd>


<dt>pAdjustment</dt>
<dd><p>The p-value correction method for multiple testing. It
must be one of: <code>fdr</code> (default), <code>BH</code>, <code>none</code>, <code>holm</code>, <code>hochberg</code>, <code>hommel</code>,
<code>bonferroni</code>, <code>BY</code></p></dd>


<dt>corMethod</dt>
<dd><p>The correlation method to be used for correlation analysis.
It must be one of: <code>spearman</code> (default), <code>pearson</code>, <code>kendall</code>. See the
<em>details</em> section for further information</p></dd>


<dt>corCutoff</dt>
<dd><p>The minimum (negative) value of correlation coefficient to
consider meaningful a miRNA-target relationship. Default is <code>0.5</code></p></dd>


<dt>associationMethod</dt>
<dd><p>The statistical test used for evaluating the
association between miRNAs and their targets for unpaired data. It must be
one of <code>boschloo</code> (default), to perform a one-sided Boschloo's exact test;
<code>fisher-midp</code>, to compute a one-sided Fisher's exact test with Lancaster's
mid-p correction; <code>fisher</code>, to perform a one-sided Fisher's exact test</p></dd>


<dt>nuisanceParam</dt>
<dd><p>The number of nuisance parameter values considered for
p-value calculation in <code>boschloo</code> method. The higher this value, the better
the p-value estimation accuracy. Default is 100</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object containing integration
results. To access these results, the user can make use of the
<code><a href="integration.html">integration()</a></code> function. For additional details on how to
interpret the results of miRNA-gene integrative analysis, please see
<code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code>.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>As already pointed out, if miRNA and gene expression data derive from the
same samples, a correlation analysis is used. For evaluating these
relationships, the default method used is Spearman's correlation
coefficient, as:</p><ul><li><p>it does not need normally distributed data;</p></li>
<li><p>it does not assume linearity;</p></li>
<li><p>it is much more resistant to outliers.</p></li>
</ul><p>However, the user can also decide to use other correlation methods,
such as Pearson's and Kendall's correlation. Nevertheless, for NGS data
it may happen that a certain number of ties is present in the expression
values. This can be handled by <code>spearman</code> method as it computes a
tie-corrected version of Spearman's coefficients. However, another
correlation method that is suitable to perform rank correlation on tied data
is the Kendall's tau-b method, usable with <code>kendall</code>.</p>
<p>Regarding correlation direction, since miRNAs mainly act as negative
regulators, only negatively correlated miRNA-target pairs are evaluated, and
statistical significance is calculated through a one-tailed t-test.</p>
<p>Please notice that if strong batch effects are noticed in expression data,
it is recommended to remove them through the <code><a href="batchCorrection.html">batchCorrection()</a></code> function
implemented in MIRit.</p>
<p>Moreover, if gene expression data and miRNA expression data derive from
different samples (unpaired data), a correlation analysis can't be
performed. However, one-sided association tests can be applied in these
cases to evaluate if targets of down-regulated miRNAs are statistically
enriched in up-regulated genes, and, conversely, if targets of up-regulated
miRNAs are statistically enriched in down-regulated genes. In this case,
Fisher's exact test can be used to assess the statistical significance of
this inverse association. Moreover, Lancaster's mid-p adjustment can be
applied since it has been shown that it increases statistical power
while retaining Type I error rates. However, Fisher's exact test is a
conditional test that requires the sum of both rows and columns of a
contingency table to be fixed. Notably, this is not true for genomic data
because it is likely that different datasets may lead to a different number
of DEGs. Therefore, the default behavior in MIRit is to use a variant of
Barnard's exact test, named Boschloo's exact test, that is suitable when
group sizes of contingency tables are variable. Moreover, it is possible to
demonstrate that Boschloo's test is uniformly more powerful compared to
Fisher's exact test.</p>
<p>Finally, for unpaired data, the effect of DE-miRNAs on the expression of
target genes can be estimated through rotation gene-set tests. In particular,
a fast approximation to rotation gene-set testing called <code>fry</code>, implemented
in the <code>limma</code> package, can be used to statistically quantify the influence
of miRNAs on the expression changes of their target genes.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W, Smyth GK (2015). “limma
powers differential expression analyses for RNA-sequencing and microarray
studies.” Nucleic Acids Research, 43(7), e47. <a href="doi:10.1093/nar/gkv007">doi:10.1093/nar/gkv007</a>.</p>
<p>Di Wu and others, ROAST: rotation gene set tests for complex microarray
experiments, Bioinformatics, Volume 26, Issue 17, September 2010,
Pages 2176–2182, <a href="https://doi.org/10.1093/bioinformatics/btq401" class="external-link">https://doi.org/10.1093/bioinformatics/btq401</a>.</p>
<p>Routledge, R. D. (1994). Practicing Safe Statistics with the Mid-p. The
Canadian Journal of Statistics / La Revue Canadienne de Statistique, 22(1),
103–110, <a href="https://doi.org/10.2307/3315826" class="external-link">https://doi.org/10.2307/3315826</a>.</p>
<p>Boschloo R.D. (1970). "Raised Conditional Level of Significance for the
2x2-table when Testing the Equality of Two Probabilities".
Statistica Neerlandica. 24: 1–35.
<a href="doi:10.1111/j.1467-9574.1970.tb00104.x">doi:10.1111/j.1467-9574.1970.tb00104.x</a>.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Jacopo Ronchi, <a href="mailto:jacopo.ronchi@unimib.it">jacopo.ronchi@unimib.it</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># load example MirnaExperiment object</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="loadExamples.html">loadExamples</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># perform integration analysis with default settings</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu">mirnaIntegration</span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Since data derive from paired samples, a correlation test will be used.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing Spearman's correlation analysis...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> A statistically significant correlation between 215 miRNA-target pairs was found!</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># perform Kendall's correlation analysis with tau &gt; 0.8 and p &lt; 0.05</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu">mirnaIntegration</span><span class="op">(</span><span class="va">obj</span>,</span></span>
<span class="r-in"><span>    test <span class="op">=</span> <span class="st">"correlation"</span>,</span></span>
<span class="r-in"><span>    corMethod <span class="op">=</span> <span class="st">"kendall"</span>, corCutoff <span class="op">=</span> <span class="fl">0.8</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> As specified by the user, a correlation will be used.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing Kendall's correlation analysis...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> A statistically significant correlation between 1 miRNA-target pairs was found!</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>NULL</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>MIRit is provided without <strong>any warranty</strong>. NULL</p>
</div>

    </footer></div>

  

  

  </body></html>

