<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Create a trackplot to show the association between miRNAs and disease-SNPs — mirVariantPlot • MIRit</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Create a trackplot to show the association between miRNAs and disease-SNPs — mirVariantPlot"><meta property="og:description" content="This function plots a trackplot that shows the genomic position of
disease-associated SNPs that affect miRNA genes. This is useful to visualize
the genomic position and context of disease-associated variants that may
affect miRNA expression."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MIRit</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MIRit.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/jacopo-ronchi/MIRit/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Create a trackplot to show the association between miRNAs and disease-SNPs</h1>
    <small class="dont-index">Source: <a href="https://github.com/jacopo-ronchi/MIRit/blob/HEAD/R/visualization.R" class="external-link"><code>R/visualization.R</code></a></small>
    <div class="hidden name"><code>mirVariantPlot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function plots a trackplot that shows the genomic position of
disease-associated SNPs that affect miRNA genes. This is useful to visualize
the genomic position and context of disease-associated variants that may
affect miRNA expression.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">mirVariantPlot</span><span class="op">(</span></span>
<span>  <span class="va">variantId</span>,</span>
<span>  <span class="va">snpAssociation</span>,</span>
<span>  showContext <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  showSequence <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  snpFill <span class="op">=</span> <span class="st">"lightblue"</span>,</span>
<span>  mirFill <span class="op">=</span> <span class="st">"orange"</span>,</span>
<span>  from <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  to <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  title <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>variantId</dt>
<dd><p>A valid name of a SNP variant! (e.g. <code>"rs394581"</code>)</p></dd>


<dt>snpAssociation</dt>
<dd><p>A <code>data.frame</code> object containing the results of
<code><a href="findMirnaSNPs.html">findMirnaSNPs()</a></code> function</p></dd>


<dt>showContext</dt>
<dd><p>Logical, if <code>TRUE</code> a complete genomic context with genes
present in the region will be shown. Default is <code>FALSE</code> to just display the
variant and the miRNA gene</p></dd>


<dt>showSequence</dt>
<dd><p>Logical, whether to display a color-coded sequence at
the bottom of the trackplot. Default is <code>TRUE</code>. This parameter will be set
to <code>FALSE</code> if <code>showContext</code> is <code>TRUE</code></p></dd>


<dt>snpFill</dt>
<dd><p>It must be an R color name that specifies the fill color of
the SNP locus. Default is <code>lightblue</code>. Available color formats include
color names, such as 'blue' and 'red', and hexadecimal colors specified
as #RRGGBB</p></dd>


<dt>mirFill</dt>
<dd><p>It must be an R color name that specifies the fill color of
the miRNA locus. Default is <code>orange</code>. Available color formats include
color names, such as 'blue' and 'red', and hexadecimal colors specified
as #RRGGBB</p></dd>


<dt>from</dt>
<dd><p>The start position of the plotted genomic range. Default is
NULL to automatically determine an appropriate position</p></dd>


<dt>to</dt>
<dd><p>The end position of the plotted genomic range. Default is
NULL to automatically determine an appropriate position</p></dd>


<dt>title</dt>
<dd><p>The title of the plot. Default is <code>NULL</code> not to include a plot
title</p></dd>


<dt>...</dt>
<dd><p>Other parameters that can be passed to <code><a href="https://rdrr.io/pkg/Gviz/man/plotTracks.html" class="external-link">Gviz::plotTracks()</a></code>
function</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A trackplot with information about chromosome, SNP and miRNA gene location.</p>
    </div>
    <div id="note">
    <h2>Note</h2>
    <p>This function retrieves genomic coordinates from the output of
<code><a href="findMirnaSNPs.html">findMirnaSNPs()</a></code> function and then uses <code>Gviz</code> package to build
the trackplot.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Hahne, F., Ivanek, R. (2016). Visualizing Genomic Data Using Gviz and
Bioconductor. In: Mathé, E., Davis, S. (eds) Statistical Genomics. Methods
in Molecular Biology, vol 1418. Humana Press, New York, NY.
<a href="https://doi.org/10.1007/978-1-4939-3578-9_16" class="external-link">https://doi.org/10.1007/978-1-4939-3578-9_16</a></p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Jacopo Ronchi, <a href="mailto:jacopo.ronchi@unimib.it">jacopo.ronchi@unimib.it</a></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># load example MirnaExperiment object</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="loadExamples.html">loadExamples</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="co"># retrieve associated SNPs</span></span></span>
<span class="r-in"><span><span class="va">association</span> <span class="op">&lt;-</span> <span class="fu"><a href="findMirnaSNPs.html">findMirnaSNPs</a></span><span class="op">(</span><span class="va">obj</span>, <span class="va">disId</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># visualize association as a trackplot</span></span></span>
<span class="r-in"><span><span class="fu">mirVariantPlot</span><span class="op">(</span>variantId <span class="op">=</span> <span class="va">varId</span>, snpAssociation <span class="op">=</span> <span class="va">association</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jacopo Ronchi.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

