<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Generate multidimensional scaling (MDS) plots to explore miRNA/gene
expression distances — plotDimensions • MIRit</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Generate multidimensional scaling (MDS) plots to explore miRNA/gene
expression distances — plotDimensions"><meta property="og:description" content="This function performs multidimensional scaling in order to produce a simple
scatterplot that shows miRNA/gene expression variations among samples. In
particular, starting from a MirnaExperiment
object, this functions allows to visualize both miRNA and gene expression in
the multidimensional space. Moreover, it is possible to color samples on the
basis of specific variables, and this is extremely useful to assess
miRNA/gene expression variations between distinct biological groups."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MIRit</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MIRit.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/jacopo-ronchi/MIRit/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate multidimensional scaling (MDS) plots to explore miRNA/gene
expression distances</h1>
    <small class="dont-index">Source: <a href="https://github.com/jacopo-ronchi/MIRit/blob/HEAD/R/visualization.R" class="external-link"><code>R/visualization.R</code></a></small>
    <div class="hidden name"><code>plotDimensions.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function performs multidimensional scaling in order to produce a simple
scatterplot that shows miRNA/gene expression variations among samples. In
particular, starting from a <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code>
object, this functions allows to visualize both miRNA and gene expression in
the multidimensional space. Moreover, it is possible to color samples on the
basis of specific variables, and this is extremely useful to assess
miRNA/gene expression variations between distinct biological groups.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">plotDimensions</span><span class="op">(</span></span>
<span>  <span class="va">mirnaObj</span>,</span>
<span>  <span class="va">assay</span>,</span>
<span>  condition <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  dimensions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>  labels <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  boxedLabel <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  pointSize <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  pointAlpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  colorScale <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  title <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  fontSize <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  fontFamily <span class="op">=</span> <span class="st">""</span>,</span>
<span>  legend <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>  borderWidth <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  allBorders <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  grid <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>mirnaObj</dt>
<dd><p>A <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object
containing miRNA and gene data</p></dd>


<dt>assay</dt>
<dd><p>The results to display. It must be either 'microRNA', to plot
miRNA expression, or 'genes', to produce MDS plot for genes</p></dd>


<dt>condition</dt>
<dd><p>It must be the column name of a variable specified in the
metadata (colData) of a <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object;
or, alternatively, it must be a character/factor object that specifies
group memberships (eg. c("healthy, "healthy", "disease", "disease"))</p></dd>


<dt>dimensions</dt>
<dd><p>It is a numeric vector of length 2 that indicates the two
dimensions to represent on the plot. Default is <code>c(1, 2)</code> to plot the two
dimensions that account for the highest portion of variability</p></dd>


<dt>labels</dt>
<dd><p>Logical, whether to display labels or not. Default is FALSE</p></dd>


<dt>boxedLabel</dt>
<dd><p>Logical, whether to show labels inside a rectangular shape
(default) or just as text elements</p></dd>


<dt>pointSize</dt>
<dd><p>The size of points in the MDS plot (default is 3)</p></dd>


<dt>pointAlpha</dt>
<dd><p>The transparency of points in the MDS plot (default
is 1)</p></dd>


<dt>colorScale</dt>
<dd><p>It must be a named character vector where values
correspond to R colors, while names coincide with the groups specified in
the <code>condition</code> parameter (eg. c("healthy" = "green", "disease" = "red")).
Default is NULL, in order to use the default color scale. Available color
formats include color names, such as 'blue' and 'red', and hexadecimal
colors specified as #RRGGBB</p></dd>


<dt>title</dt>
<dd><p>The title of the plot. Default is <code>NULL</code> not to include a plot
title</p></dd>


<dt>fontSize</dt>
<dd><p>The base size for text elements within the plot.
Default is 12</p></dd>


<dt>fontFamily</dt>
<dd><p>The base family for text elements within the plot</p></dd>


<dt>legend</dt>
<dd><p>The position of the legend. Allowed values are <code>top</code>,
<code>bottom</code>, <code>right</code>, <code>left</code> and <code>none</code>. The default setting is <code>top</code> to show
a legend above the plot. If <code>none</code> is specified, the legend will not be
included in the graph.</p></dd>


<dt>borderWidth</dt>
<dd><p>The width of plot borders (default is 1)</p></dd>


<dt>allBorders</dt>
<dd><p>Logical, whetether to show all panel borders, or just the
bottom and left borders. Default is TRUE</p></dd>


<dt>grid</dt>
<dd><p>Logical, whether to show grid lines or not. Default is FALSE</p></dd>


<dt>...</dt>
<dd><p>Other parameters that can be passed to <code><a href="https://rdrr.io/pkg/limma/man/plotMDS.html" class="external-link">limma::plotMDS()</a></code> function</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>An object of class <code>ggplot</code> containing the plot.</p>
    </div>
    <div id="note">
    <h2>Note</h2>
    <p>To perform multidimensional scaling, this function internally uses
<code><a href="https://rdrr.io/pkg/limma/man/plotMDS.html" class="external-link">limma::plotMDS()</a></code> function provided by <code>limma</code> package.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W, Smyth GK (2015). “limma
powers differential expression analyses for RNA-sequencing and microarray
studies.” Nucleic Acids Research, 43(7), e47. <a href="doi:10.1093/nar/gkv007">doi:10.1093/nar/gkv007</a>.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Jacopo Ronchi, <a href="mailto:jacopo.ronchi@unimib.it">jacopo.ronchi@unimib.it</a></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># load example MirnaExperiment object</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="loadExamples.html">loadExamples</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># produce MDS plot for miRNA expression with labels</span></span></span>
<span class="r-in"><span><span class="fu">plotDimensions</span><span class="op">(</span><span class="va">obj</span>, <span class="st">"microRNA"</span>, condition <span class="op">=</span> <span class="st">"disease"</span>, labels <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="plotDimensions-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># produce MDS plot for genes without condition color</span></span></span>
<span class="r-in"><span><span class="fu">plotDimensions</span><span class="op">(</span><span class="va">obj</span>, <span class="st">"genes"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="plotDimensions-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jacopo Ronchi.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

