<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function creates a scatter plot that shows the correlation between
miRNA and gene expression levels. This is useful after correlation
analysis performed through the mirnaIntegration() function, to
graphically visualize the quantitative effect of miRNA dysregulations on
target gene expression. Furthermore, this function performs linear/monotonic
regression to better represent the relationships between miRNA-target pairs."><title>Plot correlation between miRNAs and genes within biological groups — plotCorrelation • MIRit</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Plot correlation between miRNAs and genes within biological groups — plotCorrelation"><meta property="og:description" content="This function creates a scatter plot that shows the correlation between
miRNA and gene expression levels. This is useful after correlation
analysis performed through the mirnaIntegration() function, to
graphically visualize the quantitative effect of miRNA dysregulations on
target gene expression. Furthermore, this function performs linear/monotonic
regression to better represent the relationships between miRNA-target pairs."><meta property="og:image" content="/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MIRit</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/MIRit.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jacopo-ronchi/MIRit/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Plot correlation between miRNAs and genes within biological groups</h1>
      <small class="dont-index">Source: <a href="https://github.com/jacopo-ronchi/MIRit/blob/HEAD/R/visualization.R" class="external-link"><code>R/visualization.R</code></a></small>
      <div class="d-none name"><code>plotCorrelation.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function creates a scatter plot that shows the correlation between
miRNA and gene expression levels. This is useful after correlation
analysis performed through the <code><a href="mirnaIntegration.html">mirnaIntegration()</a></code> function, to
graphically visualize the quantitative effect of miRNA dysregulations on
target gene expression. Furthermore, this function performs linear/monotonic
regression to better represent the relationships between miRNA-target pairs.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">plotCorrelation</span><span class="op">(</span></span>
<span>  <span class="va">mirnaObj</span>,</span>
<span>  <span class="va">mirna</span>,</span>
<span>  <span class="va">gene</span>,</span>
<span>  condition <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  showCoeff <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  regression <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  useRanks <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  lineCol <span class="op">=</span> <span class="st">"red"</span>,</span>
<span>  lineType <span class="op">=</span> <span class="st">"dashed"</span>,</span>
<span>  lineWidth <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  pointSize <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  colorScale <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  fontSize <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  fontFamily <span class="op">=</span> <span class="st">""</span>,</span>
<span>  legend <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>  borderWidth <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  allBorders <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  grid <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>mirnaObj</dt>
<dd><p>A <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object
containing miRNA and gene data</p></dd>


<dt>mirna</dt>
<dd><p>The name of the miRNA for which we want to observe the
correlation</p></dd>


<dt>gene</dt>
<dd><p>The name of the gene for which we want to observe the
correlation</p></dd>


<dt>condition</dt>
<dd><p>It must be NULL (default) to plot expression based on the
group variable used for differential expression analysis. Alternatively, it
must be a character/factor object that specifies group memberships
(eg. c("healthy, "healthy", "disease", "disease"))</p></dd>


<dt>showCoeff</dt>
<dd><p>Logical, whether to show the correlation coeffficient or
not. Note that the "R" is used for Pearson's correlation", "rho" for
Spearman's correlation, and "tau" for Kendall's correlation. Default is TRUE</p></dd>


<dt>regression</dt>
<dd><p>Logical, whether to display a linear/monotonic regression
line that fits miRNA-gene correlation data. Default is TRUE</p></dd>


<dt>useRanks</dt>
<dd><p>Logical, whether to represent non-parametric correlation
analyses (Spearman's and Kendall's correlations) through rank-transformed
data. Note that in this case, linear regression is performed on ranked
data instead of monotonic regression. Default is FALSE</p></dd>


<dt>lineCol</dt>
<dd><p>It must be an R color name that specifies the color of
the regression line. Default is <code>red</code>. Available color formats include
color names, such as 'blue' and 'red', and hexadecimal colors specified
as #RRGGBB</p></dd>


<dt>lineType</dt>
<dd><p>It specifies the line type used for the regression line. It
must be either 'blank', 'solid', 'dashed' (default), 'dotted', 'dotdash',
'longdash' or 'twodash'</p></dd>


<dt>lineWidth</dt>
<dd><p>The width of the fitted regression line (default is 0.8)</p></dd>


<dt>pointSize</dt>
<dd><p>The size of points in the correlation plot (default is 3)</p></dd>


<dt>colorScale</dt>
<dd><p>It must be a named character vector where values
correspond to R colors, while names coincide with the groups specified in
the <code>condition</code> parameter (eg. c("healthy" = "green", "disease" = "red")).
Default is NULL, in order to use the default color scale. Available color
formats include color names, such as 'blue' and 'red', and hexadecimal
colors specified as #RRGGBB</p></dd>


<dt>fontSize</dt>
<dd><p>The base size for text elements within the plot.
Default is 12</p></dd>


<dt>fontFamily</dt>
<dd><p>The base family for text elements within the plot</p></dd>


<dt>legend</dt>
<dd><p>The position of the legend. Allowed values are <code>top</code>,
<code>bottom</code>, <code>right</code>, <code>left</code> and <code>none</code>. The default setting is <code>top</code> to show
a legend above the plot. If <code>none</code> is specified, the legend will not be
included in the graph.</p></dd>


<dt>borderWidth</dt>
<dd><p>The width of plot borders (default is 1)</p></dd>


<dt>allBorders</dt>
<dd><p>Logical, whetether to show all panel borders, or just the
bottom and left borders. Default is TRUE</p></dd>


<dt>grid</dt>
<dd><p>Logical, whether to show grid lines or not. Default is TRUE</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>An object of class <code>ggplot</code> containing the correlation scatter plot.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>When non-parametric correlation has been performed with the
<code><a href="mirnaIntegration.html">mirnaIntegration()</a></code> function, a regression line can be fitted through
monotonic regression on expression levels, or through linear regression
performed on rank-transformed data. Since, ranks do not correspond to real
expression values, the default option is to perform monotonic regression
to fit a monotonic curve. To do so, this function makes use of the <code>MonoPoly</code>
R package, which implements the algorithm proposed by Murray et al. in 2016.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>K. Murray, S. Müller &amp; B. A. Turlach (2016) Fast and flexible methods for
monotone polynomial fitting, Journal of Statistical Computation and
Simulation, 86:15, 2946-2966, DOI: <a href="10.1080/00949655.2016.1139582">10.1080/00949655.2016.1139582</a>.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Jacopo Ronchi, <a href="mailto:jacopo.ronchi@unimib.it">jacopo.ronchi@unimib.it</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># load example MirnaExperiment object</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="loadExamples.html">loadExamples</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># perform miRNA-target integration</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="mirnaIntegration.html">mirnaIntegration</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Since data derive from paired samples, a correlation test will be used.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing Spearman's correlation analysis...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> A statistically significant correlation between 215 miRNA-target pairs was found!</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># plot correlation between miR-146b and PAX8 with monotonic regression curve</span></span></span>
<span class="r-in"><span><span class="fu">plotCorrelation</span><span class="op">(</span><span class="va">obj</span>, <span class="st">"hsa-miR-146b-5p"</span>, <span class="st">"PAX8"</span>, condition <span class="op">=</span> <span class="st">"disease"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="plotCorrelation-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Jacopo Ronchi.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

