<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function takes influential miRNA-mRNA interactions, identified by the
mirnaIntegration() function, and adds them to biological pathways
retrieved from a pathway database such as KEGG, WikiPathways and
Reactome. The pathways returned from this function are needed to perform
a topologically-aware integrative pathway analysis (TAIPA) through the
topologicalAnalysis() function."><title>Prepare miRNA-augmented pathways for integrative miRNA-mRNA pathway analyses — preparePathways • MIRit</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Prepare miRNA-augmented pathways for integrative miRNA-mRNA pathway analyses — preparePathways"><meta property="og:description" content="This function takes influential miRNA-mRNA interactions, identified by the
mirnaIntegration() function, and adds them to biological pathways
retrieved from a pathway database such as KEGG, WikiPathways and
Reactome. The pathways returned from this function are needed to perform
a topologically-aware integrative pathway analysis (TAIPA) through the
topologicalAnalysis() function."><meta property="og:image" content="/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MIRit</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/MIRit.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jacopo-ronchi/MIRit/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Prepare miRNA-augmented pathways for integrative miRNA-mRNA pathway analyses</h1>
      <small class="dont-index">Source: <a href="https://github.com/jacopo-ronchi/MIRit/blob/HEAD/R/topological-integration.R" class="external-link"><code>R/topological-integration.R</code></a></small>
      <div class="d-none name"><code>preparePathways.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function takes influential miRNA-mRNA interactions, identified by the
<code><a href="mirnaIntegration.html">mirnaIntegration()</a></code> function, and adds them to biological pathways
retrieved from a pathway database such as <code>KEGG</code>, <code>WikiPathways</code> and
<code>Reactome</code>. The pathways returned from this function are needed to perform
a topologically-aware integrative pathway analysis (TAIPA) through the
<code><a href="topologicalAnalysis.html">topologicalAnalysis()</a></code> function.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">preparePathways</span><span class="op">(</span></span>
<span>  <span class="va">mirnaObj</span>,</span>
<span>  database <span class="op">=</span> <span class="st">"KEGG"</span>,</span>
<span>  organism <span class="op">=</span> <span class="st">"Homo sapiens"</span>,</span>
<span>  minPc <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  BPPARAM <span class="op">=</span> <span class="fu">bpparam</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>mirnaObj</dt>
<dd><p>A <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object
containing miRNA and gene data</p></dd>


<dt>database</dt>
<dd><p>The name of the database to use. It must be one of: <code>KEGG</code>,
<code>Reactome</code>, and <code>WikiPathways</code>. Default is <code>KEGG</code></p></dd>


<dt>organism</dt>
<dd><p>The name of the organism under consideration. The different
databases have different supported organisms. To see the list of supported
organisms for a given database, use the <code><a href="supportedOrganisms.html">supportedOrganisms()</a></code> function.
Default specie is <code>Homo sapiens</code></p></dd>


<dt>minPc</dt>
<dd><p>The minimum percentage of measured features that a pathway must
have for being considered in the analysis. Default is 10. See the <em>details</em>
section for additional information</p></dd>


<dt>BPPARAM</dt>
<dd><p>The desired parallel computing behavior. This parameter
defaults to <code><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">BiocParallel::bpparam()</a></code>, but this can be edited. See
<code><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">BiocParallel::bpparam()</a></code> for information on parallel computing in R</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A <code>list</code> object containing the miRNA-augmented pathways as <code>graphNEL</code></p>


<p>objects.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>To create augmented pathways, this function uses the <code>graphite</code> R package
to download biological networks from the above mentioned databases. Then,
each pathway is converted to a <code>graph</code> object, and significant miRNA-mRNA
interactions are added to the network. Further, edge weights are added
according to interaction type.</p>
<p>At this point, biological pathways with few nodes measured are excluded
from this analysis. This is required because, during differential expression
analysis, lowly expressed features are removed. Therefore, some pathways
might result significantly affected even if only 1% of nodes is perturbed.
The default behavior is to exclude pathways with less than 10% of
representation (<code>minPc = 10</code>).</p>
<p>Finally, this function performs a breadth-first search (BFS) algorithm to
topologically sort pathway nodes so that each individual node occurs after
all its upstream nodes. Nodes within cycles are considered leaf nodes.</p>
<p>Information about pathway coverage, i.e. the percentage of nodes with
expression measurments, edge weights, topological sorting order, and the
parameters used to create the networks are all stored in the <code>graphData</code>
slot of each <code>graphNEL</code> object.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Sales, G., Calura, E., Cavalieri, D. et al. graphite - a Bioconductor
package to convert pathway topology to gene network.
BMC Bioinformatics 13, 20 (2012),
<a href="https://doi.org/10.1186/1471-2105-13-20" class="external-link">https://doi.org/10.1186/1471-2105-13-20</a>.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Jacopo Ronchi, <a href="mailto:jacopo.ronchi@unimib.it">jacopo.ronchi@unimib.it</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># load example MirnaExperiment object</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="loadExamples.html">loadExamples</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># perform integration analysis with default settings</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="mirnaIntegration.html">mirnaIntegration</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Since data derive from paired samples, a correlation test will be used.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing Spearman's correlation analysis...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> A statistically significant correlation between 215 miRNA-target pairs was found!</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="co"># retrieve pathways from KEGG and augment them with miRNA-gene interactions</span></span></span>
<span class="r-in"><span><span class="va">paths</span> <span class="op">&lt;-</span> <span class="fu">preparePathways</span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Downloading pathways from KEGG database...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting identifiers to gene symbols...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding miRNA-gene interactions to biological pathways...</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>152 pathways have been ignored because they contain too few nodes with gene expression measurement.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing topological sorting of pathway nodes...</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># perform the integrative pathway analysis with 1000 permutations</span></span></span>
<span class="r-in"><span><span class="va">ipa</span> <span class="op">&lt;-</span> <span class="fu"><a href="topologicalAnalysis.html">topologicalAnalysis</a></span><span class="op">(</span><span class="va">obj</span>, <span class="va">paths</span>, nPerm <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating pathway scores...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Generating random permutations...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating p-values with 1000 permutations...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Correcting p-values through the max-T procedure...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The topologically-aware integrative pathway analysis reported 1 significantly altered pathways!</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># access the results of pathway analysis</span></span></span>
<span class="r-in"><span><span class="fu"><a href="integratedPathways.html">integratedPathways</a></span><span class="op">(</span><span class="va">ipa</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                             pathway  coverage    score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Thyroid hormone synthesis Thyroid hormone synthesis 0.3469388 12.12941</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           normalized.score       P.Val adj.P.Val</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Thyroid hormone synthesis         7.977012 0.000999001     0.019</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># create a dotplot of integrated pathways</span></span></span>
<span class="r-in"><span><span class="fu"><a href="integrationDotplot.html">integrationDotplot</a></span><span class="op">(</span><span class="va">ipa</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="preparePathways-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># explore a specific biological network</span></span></span>
<span class="r-in"><span><span class="fu"><a href="visualizeNetwork.html">visualizeNetwork</a></span><span class="op">(</span><span class="va">ipa</span>, <span class="st">"Thyroid hormone synthesis"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="preparePathways-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Jacopo Ronchi.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>MIRit is provided without <strong>any warranty</strong>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

