<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function allows to perform an integrative pathway analysis that aims
to identify the biological networks that are most affected by miRNomic
and transcriptomic dysregulations. This function takes miRNA-augmented
pathways, created by the preparePathways() function, and then calculates
a score that estimates the degree of impairment for each pathway. Later,
statistical significance is calculated through a permutation test. The main
advantages of this method are that it doesn't require matched samples, and
that it allows to perform an integrative miRNA-mRNA pathway analysis that
takes into account the topology of biological networks. See the details
section for additional information."><title>Perform a topologically-aware integrative pathway analysis (TAIPA) — topologicalAnalysis • MIRit</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Perform a topologically-aware integrative pathway analysis (TAIPA) — topologicalAnalysis"><meta property="og:description" content="This function allows to perform an integrative pathway analysis that aims
to identify the biological networks that are most affected by miRNomic
and transcriptomic dysregulations. This function takes miRNA-augmented
pathways, created by the preparePathways() function, and then calculates
a score that estimates the degree of impairment for each pathway. Later,
statistical significance is calculated through a permutation test. The main
advantages of this method are that it doesn't require matched samples, and
that it allows to perform an integrative miRNA-mRNA pathway analysis that
takes into account the topology of biological networks. See the details
section for additional information."><meta property="og:image" content="/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MIRit</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/MIRit.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jacopo-ronchi/MIRit/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Perform a topologically-aware integrative pathway analysis (TAIPA)</h1>
      <small class="dont-index">Source: <a href="https://github.com/jacopo-ronchi/MIRit/blob/HEAD/R/topological-integration.R" class="external-link"><code>R/topological-integration.R</code></a></small>
      <div class="d-none name"><code>topologicalAnalysis.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function allows to perform an integrative pathway analysis that aims
to identify the biological networks that are most affected by miRNomic
and transcriptomic dysregulations. This function takes miRNA-augmented
pathways, created by the <code><a href="preparePathways.html">preparePathways()</a></code> function, and then calculates
a score that estimates the degree of impairment for each pathway. Later,
statistical significance is calculated through a permutation test. The main
advantages of this method are that it doesn't require matched samples, and
that it allows to perform an integrative miRNA-mRNA pathway analysis that
takes into account the topology of biological networks. See the <em>details</em>
section for additional information.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">topologicalAnalysis</span><span class="op">(</span></span>
<span>  <span class="va">mirnaObj</span>,</span>
<span>  <span class="va">pathways</span>,</span>
<span>  pCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  pAdjustment <span class="op">=</span> <span class="st">"max-T"</span>,</span>
<span>  nPerm <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>  progress <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  BPPARAM <span class="op">=</span> <span class="fu">bpparam</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>mirnaObj</dt>
<dd><p>A <code><a href="MirnaExperiment-class.html">MirnaExperiment</a></code> object
containing miRNA and gene data</p></dd>


<dt>pathways</dt>
<dd><p>A <code>list</code> of miRNA-augmented pathways returned by the
<code><a href="preparePathways.html">preparePathways()</a></code> function</p></dd>


<dt>pCutoff</dt>
<dd><p>The adjusted p-value cutoff to use for statistical
significance. The default value is <code>0.05</code></p></dd>


<dt>pAdjustment</dt>
<dd><p>The p-value correction method for multiple testing. It
must be one of: <code>max-T</code> (default), <code>fdr</code>, <code>BH</code>, <code>none</code>, <code>holm</code>, <code>hochberg</code>,
<code>hommel</code>, <code>bonferroni</code>, <code>BY</code></p></dd>


<dt>nPerm</dt>
<dd><p>The number of permutation used for assessing the statistical
significance of each pathway. Default is 10000. See the <em>details</em> section
for additional information</p></dd>


<dt>progress</dt>
<dd><p>Logical, whether to show a progress bar during p-value
calculation or not. Default is FALSE, not to include a progress bar. Please
note that setting <code>progress = TRUE</code> with high values of <code>tasks</code> leads to
less efficient parallelization. See the <em>details</em> section for additional
information</p></dd>


<dt>tasks</dt>
<dd><p>An integer between 0 and 100 that specifies how frequently the
progress bar must be updated. Default is 0 to simply split the computation
among the workers. High values of <code>tasks</code> can lead to 15-30% slower p-value
calculation. See the <em>details</em> section for additional information</p></dd>


<dt>BPPARAM</dt>
<dd><p>The desired parallel computing behavior. This parameter
defaults to <code><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">BiocParallel::bpparam()</a></code>, but this can be edited. See
<code><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">BiocParallel::bpparam()</a></code> for information on parallel computing in R</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>An object of class
<code><a href="IntegrativePathwayAnalysis-class.html">IntegrativePathwayAnalysis</a></code> that stores
the results of the analysis. See the relative help page for further details.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    
<div class="section">
<h3 id="topologically-aware-integrative-pathway-analysis-taipa-">Topologically-Aware Integrative Pathway Analysis (TAIPA)<a class="anchor" aria-label="anchor" href="#topologically-aware-integrative-pathway-analysis-taipa-"></a></h3>


<p>This analysis aims to identify the biological pathways that result affected
by miRNA and mRNA dysregulations. In this analysis, biological pathways are
retrieved from a pathway database such as KEGG, and the interplay between
miRNAs and genes is then added to the networks. Each network is defined as
a graph \(G(V, E)\), where \(V\) represents nodes, and \(E\)
represents the relationships between nodes.</p>
<p>Then, nodes that are not significantly differentially expressed are assigned
a weight \(w_i = 1\), whereas differentially expressed nodes are assigned
a weight \(w_i = \left| \Delta E_i \right|\), where \(\Delta E_i\) is
the linear fold change of the node. Moreover, to consider the biological
interaction between two nodes, namely \(i\) and \(j\), we define an
interaction parameter \(\beta_{i \rightarrow j} = 1\) for activation
interactions and \(\beta_{i \rightarrow j} = -1\) for repression
interactions. Subsequently, the concordance coefficient
\(\gamma_{i \rightarrow j}\) is defined as:</p>
<p>$$\gamma_{i \rightarrow j} = \begin{cases} \beta_{i \rightarrow j}
&amp;\text{if } sign(\Delta E_i) = sign(\Delta E_j) \\ - \beta_{i \rightarrow j}
&amp;\text{if } sign(\Delta E_i) \not= sign(\Delta E_j) \end{cases}\,.$$</p>
<p>Later in the process, a breadth-first search (BFS) algorithm is applied to
topologically sort pathway nodes so that each individual node occurs after
all its upstream nodes. Nodes within cycles are considered leaf nodes. At
this point, a node score \(\phi\) is calculated for each pathway node
\(i\) as:</p>
<p>$$\phi_i = w_i + \sum_{j=1}^{U} \gamma_{i \rightarrow j} \cdot k_j\,.$$</p>
<p>where \(U\) represents the number of upstream nodes,
\(\gamma_{i \rightarrow j}\) denotes the concordance coefficient, and
\(k_j\) is a propagation factor defined as:</p>
<p>$$k_j = \begin{cases} w_j &amp;\text{if } \phi_j = 0 \\ \phi_j &amp;\text{if }
\phi_j \not = 0 \end{cases}\,.$$</p>
<p>Finally, the pathway score \(\Psi\) is calculated as:</p>
<p>$$\Psi = \frac{1 - M}{N} \cdot \sum_{i=1}^{N} \phi_i\,,$$</p>
<p>where \(M\) represents the proportion of miRNAs in the pathway, and
\(N\) represents the total number of nodes in the network.</p>
<p>Then, to compute the statistical significance of each pathway score, a
permutation procedure is applied. Later, both observed pathway scores and
permuted scores are standardized by subtracting the mean score of the
permuted sets \(\mu_{\Psi_P}\) and then dividing by the standard deviation
of the permuted scores \(\sigma_{\Psi_P}\).</p>
<p>Finally, the p-value is defined based on the fraction of permutations that
reported a higher normalized pathway score than the observed one.
However, to prevent p-values equal to zero, we define p-values as:</p>
<p>$$p = \frac{\sum_{n=1}^{N_p} \left[ \Psi_{P_N} \ge \Psi_N \right] + 1}
{N_p + 1}\,.$$</p>
<p>In the end, p-values are corrected for multiple testing either through the
max-T procedure (default option) which is particularly suited for
permutation tests, or through the standard multiple testing approaches.</p>
</div>

<div class="section">
<h3 id="implementation-details">Implementation details<a class="anchor" aria-label="anchor" href="#implementation-details"></a></h3>


<p>For computational efficiency, pathway score computation has been implemented
in C++ language. Moreover, to define the statistical significance of each
network, a permutation test is applied following the number of permutations
specified with <code>nPerm</code>. The default setting is to perform 10000 permutations.
The higher is the number of permutations, the more stable are the calculated
p-values, even though the time needed will increase. In this regard, since
computing pathway score for 10000 networks for each pathway is
computationally intensive, parallel computing has been employed to reduce
running time. The user can modify the parallel computing behavior by
specifying the <code>BPPARAM</code> parameter. See <code><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">BiocParallel::bpparam()</a></code> for
further details. Further, a progress bar can also be included to show the
completion percentage by setting <code>progress = TRUE</code>. Moreover, the user can
define how frequently the progress bar gets updated by tweaking the <code>tasks</code>
parameter. When using <code>progress = TRUE</code>, setting <code>tasks</code> to 100 tells the
function to update the progress bar 100 times, so that the user can see
increases of 1%. Instead, setting <code>tasks</code> to 50, means that the progress bar
gets updated every 2% of completion. However, keep in mind that <code>tasks</code>
values from 50 to 100 lead to 15-30% slower p-value calculation due to
increased data transfer to the workers. Instead, lower <code>tasks</code> values like
20 determine less frequent progress updates but are only slightly less
efficient than not including a progress bar.</p>
</div>

    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Peter H. Westfall and S. Stanley Young. Resampling-Based Multiple Testing:
Examples and Methods for p-Value Adjustment. John Wiley &amp; Sons.
ISBN 978-0-471-55761-6.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Jacopo Ronchi, <a href="mailto:jacopo.ronchi@unimib.it">jacopo.ronchi@unimib.it</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># load example MirnaExperiment object</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="loadExamples.html">loadExamples</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># perform integration analysis with default settings</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="mirnaIntegration.html">mirnaIntegration</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Since data derive from paired samples, a correlation test will be used.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing Spearman's correlation analysis...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> A statistically significant correlation between 215 miRNA-target pairs was found!</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="co"># retrieve pathways from KEGG and augment them with miRNA-gene interactions</span></span></span>
<span class="r-in"><span><span class="va">paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="preparePathways.html">preparePathways</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading KEGG pathways from cache...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding miRNA-gene interactions to biological pathways...</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>152 pathways have been ignored because they contain too few nodes with gene expression measurement.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing topological sorting of pathway nodes...</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># perform the integrative pathway analysis with 1000 permutations</span></span></span>
<span class="r-in"><span><span class="va">ipa</span> <span class="op">&lt;-</span> <span class="fu">topologicalAnalysis</span><span class="op">(</span><span class="va">obj</span>, <span class="va">paths</span>, nPerm <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating pathway scores...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Generating random permutations...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating p-values with 1000 permutations...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Correcting p-values through the max-T procedure...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The topologically-aware integrative pathway analysis reported 2 significantly altered pathways!</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># access the results of pathway analysis</span></span></span>
<span class="r-in"><span><span class="fu"><a href="integratedPathways.html">integratedPathways</a></span><span class="op">(</span><span class="va">ipa</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                             pathway  coverage    score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Thyroid hormone synthesis Thyroid hormone synthesis 0.3469388 12.12941</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Thyroid cancer                       Thyroid cancer 0.2820513 11.56291</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           normalized.score       P.Val adj.P.Val</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Thyroid hormone synthesis         8.436331 0.000999001     0.014</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Thyroid cancer                    7.099923 0.000999001     0.040</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># create a dotplot of integrated pathways</span></span></span>
<span class="r-in"><span><span class="fu"><a href="integrationDotplot.html">integrationDotplot</a></span><span class="op">(</span><span class="va">ipa</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="topologicalAnalysis-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># explore a specific biological network</span></span></span>
<span class="r-in"><span><span class="fu"><a href="visualizeNetwork.html">visualizeNetwork</a></span><span class="op">(</span><span class="va">ipa</span>, <span class="st">"Thyroid hormone synthesis"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="topologicalAnalysis-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Jacopo Ronchi.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>MIRit is provided without <strong>any warranty</strong>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

