<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="pandoc">
<meta http-equiv="X-UA-Compatible" content="IE=EDGE">
<title>Integrate miRNA and gene expression data with MIRit</title>
<script src="MIRit_files/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1">
<link href="MIRit_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet">
<script src="MIRit_files/bootstrap-3.3.5/js/bootstrap.min.js"></script><script src="MIRit_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script><script src="MIRit_files/bootstrap-3.3.5/shim/respond.min.js"></script><style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       </style>
<script src="MIRit_files/jqueryui-1.13.2/jquery-ui.min.js"></script><link href="MIRit_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet">
<script src="MIRit_files/tocify-1.9.1/jquery.tocify.js"></script><script src="MIRit_files/navigation-1.1/tabsets.js"></script><script src="MIRit_files/navigation-1.1/codefolding.js"></script><style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>
<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script><style type="text/css">
  p.abstract{
    text-align: center;
    font-weight: bold;
  }
  div.abstract{
    margin: auto;
    width: 90%;
  }
</style>
<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>
<link rel="stylesheet" href="/home/jacopo/R/x86_64-pc-linux-gnu-library/4.3/BiocStyle/resources/html/bioconductor.css" type="text/css">
<style type="text/css">
.main-container {
  max-width: 828px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
</style>
<!-- tabsets --><style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>
<!-- code folding --><style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {

}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 246px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>
<script>
function toggle_visibility(id1) {
  var e = document.getElementById(id1);
  e.style.display = ((e.style.display!="none") ? "none" : "block");
}
</script>
</head>
<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Integrate miRNA and gene expression data with MIRit</h1>
<p class="author-name">Jacopo Ronchi<span class="affil-mark">1*</span></p>
<p class="author-affiliation"><span class="affil-mark">1</span>University of Milano-Bicocca, Italy</p>
<p class="author-email"><span class="affil-mark">*</span><a href="mailto:jacopo.ronchi@unimib.it">jacopo.ronchi@unimib.it</a></p>
<h4 class="date">4 dicembre 2023</h4>
<h4 class="abstract">Abstract</h4>
<p>In this vignette, we are going to see how to use MIRit for investigating the compromised miRNA-gene regulatory networks in thyroid cancer. In particular, an RNA-Seq experiment will be used as an example to demonstrate how to perform an integrative analysis with MIRit, including differential expression analysis, functional enrichment and characterization, correlation analysis and, lastly, the construction and visualization of the impaired miRNAs regulatory axes within biological pathways.</p>
<h4 class="package">Package</h4>
<p>MIRit 0.99.0</p>

</div>


<div class="section level2" number="1">
<h2 id="introduction">
<span class="header-section-number">1</span> Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><img src="../reference/figures/logo.svg" style="float:right; padding:20px" height="192" alt=""></p>
<div class="section level3" number="1.1">
<h3 id="what-is-mirit">
<span class="header-section-number">1.1</span> What is MIRit<a class="anchor" aria-label="anchor" href="#what-is-mirit"></a>
</h3>
<p><code>MIRit</code> (miRNA integration tool) is an open-source R package that aims to facilitate the comprehension of microRNA (miRNA) biology through the integrative analysis of gene and miRNA expression data deriving from different platforms, including microarrays, RNA-Seq, miRNA-Seq, proteomics and single-cell transcriptomics. Given their regulatory importance, a complete characterization of miRNA dysregulations results crucial to explore the molecular networks that may lead to the insurgence of complex diseases. Unfortunately, there are no currently available options for thoroughly interpreting the biological consequences of miRNA dysregulations, thus limiting the ability to identify the affected pathways and reconstruct the compromised molecular networks. To tackle this limitation, we developed MIRit, an all-in-one framework that provides flexible and powerful methods for performing integrative miRNA-mRNA multi-omic analyses from start to finish. In particular, MIRit includes multiple modules that allow to perform:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Differential expression analysis</strong>, which identifies miRNAs and genes that vary across biological conditions for both RNA-Seq and microarray experiments (even though other technologies can also be used);</li>
<li>
<strong>Functional enrichment analysis</strong>, which allows to understand the consequences of gene dysregulations through different strategies, including over-representation analysis (ORA), gene-set enrichment analysis (GSEA) and Correlation Adjusted MEan RAnk gene set test (CAMERA);</li>
<li>
<strong>SNP association</strong>, that links miRNA dysregulations to disease-associated SNPs occurring at miRNA gene loci;</li>
<li>
<strong>Target identification</strong>, which retrieves predicted and validated miRNA-target interactions through innovative state-of-the-art approaches that limit false discoveries;</li>
<li>
<strong>Expression levels integration</strong>, which integrates the expression levels of miRNAs and mRNAs for both paired data, through correlation analyses, and unpaired data, through association tests and rotation gene-set tests;</li>
<li>
<strong>Topological analysis</strong>, which implements a novel approach called <em>Topologically-Aware Integrative Pathway Analysis (TAIPA)</em> for identifying the impaired molecular networks that affect biological pathways retrieved from KEGG, Reactome and WikiPathways.</li>
</ol>
</div>
<div class="section level3" number="1.2">
<h3 id="how-to-cite-mirit">
<span class="header-section-number">1.2</span> How to cite MIRit<a class="anchor" aria-label="anchor" href="#how-to-cite-mirit"></a>
</h3>
<p>If you use MIRit in published research, please cite:</p>
<blockquote>
<p>Ronchi J and Foti M. ‘MIRit: an integrative R framework for the identification of impaired miRNA-mRNA regulatory networks in complex diseases’. bioRxiv (2023). <a href="doi:10.1101/2023.11.24.568528" class="uri">doi:10.1101/2023.11.24.568528</a></p>
</blockquote>
<p>This package internally uses different R/Bioconductor packages, remember to cite the appropriate publication.</p>
</div>
<div class="section level3" number="1.3">
<h3 id="installation">
<span class="header-section-number">1.3</span> Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>Before starting, MIRit must be installed on your system. You can do this through Bioconductor.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## install MIRit from Bioconductor</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"MIRit"</span><span class="op">)</span></span></code></pre></div>
<p>If needed, you could also install the development version of MIRit directly from GitHub:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## install the development version from GitHub</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"jacopo-ronchi/MIRit"</span><span class="op">)</span></span></code></pre></div>
<p>When MIRit is installed, we can load it through:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## load MIRit</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jacopo-ronchi/MIRit" class="external-link">MIRit</a></span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2" number="2">
<h2 id="data-preparation">
<span class="header-section-number">2</span> Data preparation<a class="anchor" aria-label="anchor" href="#data-preparation"></a>
</h2>
<div class="section level3" number="2.1">
<h3 id="load-example-data">
<span class="header-section-number">2.1</span> Load example data<a class="anchor" aria-label="anchor" href="#load-example-data"></a>
</h3>
<p>To demonstrate the capabilities of MIRit we will use RNA-Seq data from <span class="citation">Riesco-Eizaguirre et al. (<a href="#ref-riesco-eizaguirre_mir-146b-3ppax8nis_2015">2015</a>)</span>. This experiment collected samples from 8 papillary thyroid carcinoma tumors and contralateral normal thyroid tissue from the same patients. These samples were profiled for gene expression through RNA-Seq, and for miRNA expression through miRNA-Seq. To provide an easy access to the user, raw count matrices have been retrieved from GEO and included in this package.</p>
<p>To load the example data, we can simply use the <code><a href="https://rdrr.io/r/utils/data.html" class="external-link">data()</a></code> function for both gene and miRNA count matrices.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## load count matrix for genes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">geneCounts</span>, package <span class="op">=</span> <span class="st">"MIRit"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## load count matrix for miRNAs</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">mirnaCounts</span>, package <span class="op">=</span> <span class="st">"MIRit"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="2.2">
<h3 id="paired-vs-unpaired-data">
<span class="header-section-number">2.2</span> Paired vs unpaired data<a class="anchor" aria-label="anchor" href="#paired-vs-unpaired-data"></a>
</h3>
<p>When using MIRit, we must specify whether miRNA and gene expression values derive from the same individuals or not. As already mentioned, <strong>paired data</strong> are those where individuals used to measure gene expression are the same subjects used to measure miRNA expression. On the other hand, <strong>unpaired data</strong> are those where gene expression and miRNA expression derive from different cohorts of donors. Importantly, MIRit considers as paired samples those data sets where paired measurements are available for at least some samples.</p>
<p>In our case, miRNA and gene expression data originate from the same subjects, and therefore we will conduct a <em>paired samples</em> analysis.</p>
</div>
<div class="section level3" number="2.3">
<h3 id="set-up-expression-matrices">
<span class="header-section-number">2.3</span> Set up expression matrices<a class="anchor" aria-label="anchor" href="#set-up-expression-matrices"></a>
</h3>
<p>As input data, MIRit requires miRNA and gene expression measurements as matrices with samples as columns, and genes/miRNAs as rows. Further, the row names of miRNA expression matrix should contain miRNA names according to miRBase nomenclature (e.g. hsa-miR-151a-5p, hsa-miR-21-5p), whereas for gene expression matrix, row names must contain gene symbols according to HGNC (e.g. TYK2, BDNF, NTRK2).</p>
<p>These matrices may handle different types of values deriving from multiple technologies, including microarrays, RNA-Seq and proteomics. The only requirement is that, for microarray studies, expression matrices must be normalized and log2 transformed. This can be achieved by applying the RMA algorithm implemented in the <em><a href="https://bioconductor.org/packages/3.17/oligo" class="external-link">oligo</a></em> <span class="citation">(<a href="#ref-carvalho_framework_2010">Carvalho and Irizarry 2010</a>)</span> package, or by applying other quantile normalization strategies. On the contrary, for RNA-Seq and miRNA-Seq experiments, the simple count matrix must be supplied.</p>
<p>Eventually, expression matrices required by MIRit should appear as those in <code>mirnaCounts</code> and <code>geneCounts</code>, which are displayed in Tables <a href="#tab:geneExpr">1</a> and <a href="#tab:mirnaExpr">2</a>.</p>
<div class="horizontal-scroll"><table class="table">
<caption>
<span id="tab:geneExpr">Table 1: </span><span class="caption-title">Gene expression matrix</span><br>An expression matrix containing normalized values, with row names according to hgnc.</caption>
<thead><tr class="header">
<th align="left"></th>
<th align="right">PTC 1</th>
<th align="right">PTC 2</th>
<th align="right">PTC 3</th>
<th align="right">PTC 4</th>
<th align="right">PTC 5</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">A1BG</td>
<td align="right">7</td>
<td align="right">6</td>
<td align="right">9</td>
<td align="right">12</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">A1BG-AS1</td>
<td align="right">20</td>
<td align="right">8</td>
<td align="right">22</td>
<td align="right">6</td>
<td align="right">16</td>
</tr>
<tr class="odd">
<td align="left">A1CF</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">A2M</td>
<td align="right">9</td>
<td align="right">11</td>
<td align="right">9</td>
<td align="right">37</td>
<td align="right">18</td>
</tr>
<tr class="odd">
<td align="left">A2M-AS1</td>
<td align="right">1486</td>
<td align="right">722</td>
<td align="right">801</td>
<td align="right">968</td>
<td align="right">1787</td>
</tr>
</tbody>
</table></div>
<div class="horizontal-scroll"><table class="table">
<caption>
<span id="tab:mirnaExpr">Table 2: </span><span class="caption-title">MiRNA expression matrix</span><br>An expression matrix containing normalized values, with row names according to miRBase.</caption>
<thead><tr class="header">
<th align="left"></th>
<th align="right">PTC 1</th>
<th align="right">PTC 2</th>
<th align="right">PTC 3</th>
<th align="right">PTC 4</th>
<th align="right">PTC 5</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">hsa-let-7a-2-3p</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">9</td>
<td align="right">1</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">hsa-let-7a-3p</td>
<td align="right">472</td>
<td align="right">82</td>
<td align="right">228</td>
<td align="right">122</td>
<td align="right">313</td>
</tr>
<tr class="odd">
<td align="left">hsa-let-7a-5p</td>
<td align="right">141101</td>
<td align="right">45543</td>
<td align="right">105598</td>
<td align="right">45503</td>
<td align="right">159598</td>
</tr>
<tr class="even">
<td align="left">hsa-let-7b-3p</td>
<td align="right">412</td>
<td align="right">81</td>
<td align="right">120</td>
<td align="right">147</td>
<td align="right">164</td>
</tr>
<tr class="odd">
<td align="left">hsa-let-7b-5p</td>
<td align="right">16337</td>
<td align="right">6586</td>
<td align="right">8121</td>
<td align="right">7993</td>
<td align="right">16516</td>
</tr>
</tbody>
</table></div>
</div>
<div class="section level3" number="2.4">
<h3 id="meta">
<span class="header-section-number">2.4</span> Define sample metadata<a class="anchor" aria-label="anchor" href="#meta"></a>
</h3>
<p>Once we have expression matrices in the proper format, we need to inform MIRit about the samples in study and the biological conditions of interest. To do so, we need to create a <code>data.frame</code> that must contain:</p>
<ul>
<li>a column named <code>primary</code>, specifying a unique identifier for each different subject;</li>
<li>a column named <code>mirnaCol</code>, matching the column name used for each sample in the miRNA expression matrix;</li>
<li>a column named <code>geneCol</code>, matching the column name used for each sample in the gene expression matrix;</li>
<li>a column that defines the biological condition of interest;</li>
<li>other optional columns that store specific sample metadata, such as age, sex and so on…</li>
</ul>
<p>Firstly, let’s take a look at the column names used for miRNA and gene expression matrices.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## print sample names in geneCounts</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">geneCounts</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "PTC 1" "PTC 2" "PTC 3" "PTC 4" "PTC 5" "PTC 6" "PTC 7" "PTC 8" "NTH 1"</span></span>
<span><span class="co">#&gt; [10] "NTH 2" "NTH 3" "NTH 4" "NTH 5" "NTH 6" "NTH 7" "NTH 8"</span></span>
<span></span>
<span><span class="co">## print sample names in mirnaCounts</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">mirnaCounts</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "PTC 1" "PTC 2" "PTC 3" "PTC 4" "PTC 5" "PTC 6" "PTC 7" "PTC 8" "NTH 1"</span></span>
<span><span class="co">#&gt; [10] "NTH 2" "NTH 3" "NTH 4" "NTH 5" "NTH 6" "NTH 7" "NTH 8"</span></span>
<span></span>
<span><span class="co">## check if samples in geneCounts are equal to those in mirnaCounts</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/identical.html" class="external-link">identical</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">geneCounts</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">mirnaCounts</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
<p>In our case, we see that both expression matrices have the same column names, and therefore <code>mirnaCol</code> and <code>geneCol</code> will contain the same values. However, note that is not always the case, especially for unpaired data, where miRNA and gene expression values derive from different subjects. In these cases, <code>mirnaCol</code> and <code>geneCol</code> must map each column of miRNA and gene expression matrices to the relative subjects indicated in the <code>primary</code> column. Notably, for unpaired data, NAs can be used for missing entries in <code>mirnaCol</code>/<code>geneCol</code>.</p>
<p>That said, we can proceed to create the <code>data.frame</code> with sample metadata as follows.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## create a data.frame containing sample metadata</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    primary <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">mirnaCounts</span><span class="op">)</span>,</span>
<span>    mirnaCol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">mirnaCounts</span><span class="op">)</span>,</span>
<span>    geneCol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">geneCounts</span><span class="op">)</span>,</span>
<span>    disease <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"PTC"</span>, <span class="fl">8</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"NTH"</span>, <span class="fl">8</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    patient <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Sample_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">8</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="2.5">
<h3 id="create-a-mirnaexperiment-object">
<span class="header-section-number">2.5</span> Create a <code>MirnaExperiment</code> object<a class="anchor" aria-label="anchor" href="#create-a-mirnaexperiment-object"></a>
</h3>
<p>At this point, after setting up expression matrices, and after defining sample metadata, we need to create an object of class <code>MirnaExperiment</code>, which is the main class used in MIRit to store all the data that are necessary for the integrative miRNA-mRNA analysis. In particular, this class extends the <code>MultiAssayExperiment</code> class from the homonym package <span class="citation">(<a href="#ref-ramos_software_2017">Ramos et al. 2017</a>)</span> to store expression levels of both miRNAs and genes, differential expression results, miRNA-target pairs and integrative miRNA-gene analysis.</p>
<p>The easiest way to create a valid <code>MirnaExperiment</code> object is to use the <code><a href="../reference/MirnaExperiment.html">MirnaExperiment()</a></code> function, which automatically handles the formatting of input data and the creation of the object.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## create the MirnaExperiment object</span></span>
<span><span class="va">experiment</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MirnaExperiment.html">MirnaExperiment</a></span><span class="op">(</span></span>
<span>    mirnaExpr <span class="op">=</span> <span class="va">mirnaCounts</span>,</span>
<span>    geneExpr <span class="op">=</span> <span class="va">geneCounts</span>,</span>
<span>    samplesMetadata <span class="op">=</span> <span class="va">meta</span>,</span>
<span>    pairedSamples <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2" number="3">
<h2 id="differential-expression-analysis">
<span class="header-section-number">3</span> Differential expression analysis<a class="anchor" aria-label="anchor" href="#differential-expression-analysis"></a>
</h2>
<p>Now that the <code>MirnaExperiment</code> object has been created, we can move to differential expression analysis, which aims to define differentially expressed features across biological conditions.</p>
<div class="section level3" number="3.1">
<h3 id="visualize-expression-variability">
<span class="header-section-number">3.1</span> Visualize expression variability<a class="anchor" aria-label="anchor" href="#visualize-expression-variability"></a>
</h3>
<p>Firstly, before doing anything else, it is useful to explore miRNA and gene expression variability through dimensionality reduction techniques. This is useful because it allows us to visualize the main drivers of expression variability. In this regard, MIRit offers the <code><a href="../reference/plotDimensions.html">plotDimensions()</a></code> function, which enables to visualize both miRNA and gene expression in the multidimensional space (MDS plots). Moreover, it is possible to color samples based on specific variables, hence allowing to explore miRNA/gene expression variation between distinct biological groups.</p>
<p>In our example, let’s examine expression variability for both miRNAs and genes, and let’s color the samples based on “disease”, a variable included in the previously defined metadata.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geneMDS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotDimensions.html">plotDimensions</a></span><span class="op">(</span><span class="va">experiment</span>,</span>
<span>    assay <span class="op">=</span> <span class="st">"genes"</span>,</span>
<span>    condition <span class="op">=</span> <span class="st">"disease"</span>,</span>
<span>    title <span class="op">=</span> <span class="st">"MDS plot for genes"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">mirnaMDS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotDimensions.html">plotDimensions</a></span><span class="op">(</span><span class="va">experiment</span>,</span>
<span>    assay <span class="op">=</span> <span class="st">"microRNA"</span>,</span>
<span>    condition <span class="op">=</span> <span class="st">"disease"</span>,</span>
<span>    title <span class="op">=</span> <span class="st">"MDS plot for miRNAs"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span><span class="va">geneMDS</span>, <span class="va">mirnaMDS</span>,</span>
<span>    nrow <span class="op">=</span> <span class="fl">1</span>, labels <span class="op">=</span> <span class="st">"AUTO"</span>,</span>
<span>    common.legend <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:mds"></span>
<img src="MIRit_files/figure-html/mds-1.png" alt="MDS plots for miRNAs and genes. Both plots show that the main source of variability is given by the disease condition." width="100%" class="widefigure"><p class="caption">
Figure 1: <span class="caption-title">MDS plots for miRNAs and genes</span><br>Both plots show that the main source of variability is given by the disease condition.
</p>
</div>
<p>As we can see from Figures <a href="#fig:mds">1</a>A and <a href="#fig:mds">1</a>B, the samples are very well separated on the basis of disease condition, thus suggesting that this is the major factor that influences expression variability. This is exactly what we want, since we aim to evaluate the differences between cancer and normal tissue. If this wasn’t the case, we should identify the confounding variables, and include them in the model used for differential expression analysis (see Section <a href="#model">3.2.2</a>).</p>
</div>
<div class="section level3" number="3.2">
<h3 id="perform-mirna-and-gene-differential-expression">
<span class="header-section-number">3.2</span> Perform miRNA and gene differential expression<a class="anchor" aria-label="anchor" href="#perform-mirna-and-gene-differential-expression"></a>
</h3>
<div class="section level4" number="3.2.1">
<h4 id="deMethods">
<span class="header-section-number">3.2.1</span> Available methods for RNA-Seq and microarrays<a class="anchor" aria-label="anchor" href="#deMethods"></a>
</h4>
<p>Now, we are ready to perform differential expression analysis. In this concern, MIRit provides different options based on the technology used for generating expression data. Indeed, when expression measurements derive from microarrays, MIRit calculates differentially expressed features through the pipeline implemented in the <em><a href="https://bioconductor.org/packages/3.17/limma" class="external-link">limma</a></em> package. On the other hand, when expression values derive from RNA-Seq experiments, MIRit allows to choose between different approaches, including:</p>
<ul>
<li>the approach defined in the <em><a href="https://bioconductor.org/packages/3.17/edgeR" class="external-link">edgeR</a></em> package;</li>
<li>the approach defined in the <em><a href="https://bioconductor.org/packages/3.17/DESeq2" class="external-link">DESeq2</a></em> package;</li>
<li>the <code>limma-voom</code> approach defined in the <em><a href="https://bioconductor.org/packages/3.17/limma" class="external-link">limma</a></em> package.</li>
</ul>
<p>Moreover, MIRit gives the possibility of fully customizing the parameters used for differential expression analysis, thus allowing a finer control that makes it easy to adopt strategies that differ from the standard pipelines proposed in these packages. For additional information, see Section <a href="#param">3.2.4</a>.</p>
</div>
<div class="section level4" number="3.2.2">
<h4 id="model">
<span class="header-section-number">3.2.2</span> Model design<a class="anchor" aria-label="anchor" href="#model"></a>
</h4>
<p>After identifying the variable of interest and the confounding factors, we must indicate the experimental model used for fitting expression values. Notably, MIRit will automatically take care of model fitting, so that we only need to indicate a formula with the appropriate variables.</p>
<p>In our case, we want to evaluate the differences between cancer and normal thyroid. Therefore, disease condition is our variable of interest. However, in this experimental design, each individual has been assayed twice, one time for cancer tissue, and one time for healthy contralateral thyroid. Thus, we also need to include the patient ID as a covariate in order to prevent the individual differences between subjects from confounding the differences due to the disease.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## design the linear model for both genes and miRNAs</span></span>
<span><span class="va">model</span> <span class="op">&lt;-</span> <span class="op">~</span> <span class="va">disease</span> <span class="op">+</span> <span class="va">patient</span></span></code></pre></div>
<p>If other variables affecting miRNA and gene expression are observed, they should be included in this formula.</p>
</div>
<div class="section level4" number="3.2.3">
<h4 id="the-performmirnade-and-performgenede-functions">
<span class="header-section-number">3.2.3</span> The <code>performMirnaDE()</code> and <code>performGeneDE()</code> functions<a class="anchor" aria-label="anchor" href="#the-performmirnade-and-performgenede-functions"></a>
</h4>
<p>Once the linear model has been defined, we can perform the differential expression analysis through the <code><a href="../reference/deAnalysis.html">performMirnaDE()</a></code> and <code><a href="../reference/deAnalysis.html">performGeneDE()</a></code> functions. Indeed, these two functions take as input a <code>MirnaExperiment</code> object, and compute differential expression for miRNAs and genes.</p>
<p>Additionally, when we run these functions, we must define different arguments, namely:</p>
<ul>
<li>
<code>group</code>, which corresponds to the name of the variable of interest as specified in Section <a href="#meta">2.4</a>. In our case, we are interested in studying the differences between cancer tissue and normal tissue, and therefore our variable of interest is “<em>disease</em>”.</li>
<li>
<code>contrast</code>, which indicates the levels of the variable of interest to be compared. In particular, this parameter takes as input a string where the levels are separated by a dash, and where the second level corresponds to the reference group. In our example, we want to compare samples from papillary thyroid cancer (PTC) against normal thyroid tissue (NTH), thus we set <code>contrast</code> to “<em>PTC-NTH</em>”.</li>
<li>
<code>design</code>, which specifies the linear model with the variable of interest and eventual covariates. To do so, we pass to this parameter the R formula that we defined in Section <a href="#model">3.2.2</a>.</li>
<li>
<code>method</code>, which tells MIRit which pipeline we want to use for computing differentially expressed features. As stated in Section <a href="#deMethods">3.2.1</a>, for microarray studies the only option available is <code>limma</code>, while for RNA-Seq experiments, the user can choose between <code>edgeR</code>, <code>DESeq2</code>, and <code>voom</code> (for limma-voom). In our case we are going to perform differential expression analysis through the pipeline implemented in the <em><a href="https://bioconductor.org/packages/3.17/edgeR" class="external-link">edgeR</a></em> package.</li>
</ul>
<p>Following our example, let’s calculate differentially expressed genes and differentially expressed miRNAs in thyroid cancer.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## perform differential expression for genes</span></span>
<span><span class="va">experiment</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deAnalysis.html">performGeneDE</a></span><span class="op">(</span><span class="va">experiment</span>, group <span class="op">=</span> <span class="st">"disease"</span>, contrast <span class="op">=</span> <span class="st">"PTC-NTH"</span>, design <span class="op">=</span> <span class="va">model</span><span class="op">)</span></span>
<span><span class="co">#&gt; Performing differential expression analysis with edgeR...</span></span>
<span><span class="co">#&gt; Differential expression analysis reported 265 significant genes with p &lt; 0.05 (correction: fdr). You can use the 'geneDE()' function to access results.</span></span>
<span></span>
<span><span class="co">## perform differential expression for miRNAs</span></span>
<span><span class="va">experiment</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deAnalysis.html">performMirnaDE</a></span><span class="op">(</span><span class="va">experiment</span>, group <span class="op">=</span> <span class="st">"disease"</span>, contrast <span class="op">=</span> <span class="st">"PTC-NTH"</span>, design <span class="op">=</span> <span class="va">model</span><span class="op">)</span></span>
<span><span class="co">#&gt; Performing differential expression analysis with edgeR...</span></span>
<span><span class="co">#&gt; Differential expression analysis reported 40 significant miRNAs with p &lt; 0.05 (correction: fdr). You can use the 'mirnaDE()' function to access results.</span></span></code></pre></div>
<p>In the end, the analysis reported 265 differentially expressed genes, and 40 differentially expressed miRNAs. If not specified, the <code><a href="../reference/deAnalysis.html">performMirnaDE()</a></code> and <code><a href="../reference/deAnalysis.html">performGeneDE()</a></code> functions will define differentially expressed genes/miRNAs as those having an adjusted p-value lower than 0.05, and an absolute log2 fold change higher than 1 (FC &gt; 2). However, this behavior can be changed by tweaking the <code>pCutoff</code> parameter, that specifies the statistical significance threshold; the <code>logFC</code> parameter, which indicates the minimum log2 fold change that features must display for being considered as differentially expressed; and the <code>pAdjustment</code> parameter, which specifies the approach used for multiple testing correction (default is <code>fdr</code> to use the Benjamini-Hochberg method).</p>
</div>
<div class="section level4" number="3.2.4">
<h4 id="param">
<span class="header-section-number">3.2.4</span> Advanced parameters<a class="anchor" aria-label="anchor" href="#param"></a>
</h4>
<p>In addition to the above mentioned settings, other parameters can be passed to the <code><a href="../reference/deAnalysis.html">performMirnaDE()</a></code> and <code><a href="../reference/deAnalysis.html">performGeneDE()</a></code> functions. Specifically, depending on the method adopted for differential expression analysis, the user can finely control the arguments passed to each function involved in the pipeline. In particular, these two functions include the following advanced parameters:</p>
<ul>
<li>
<code>filterByExpr.args</code>,</li>
<li>
<code>calcNormFactors.args</code>,</li>
<li>
<code>estimateDisp.args</code>,</li>
<li>
<code>glmQLFit.args</code>,</li>
<li>
<code>glmQLFTest.args</code>,</li>
<li>
<code>DESeq.args</code>,</li>
<li>
<code>useVoomWithQualityWeights</code>,</li>
<li>
<code>voom.args</code>,</li>
<li>
<code>lmFit.args</code>,</li>
<li>
<code>eBayes.args</code>,</li>
<li>
<code>useArrayWeights</code>,</li>
<li>
<code>useWsva</code>,</li>
<li>
<code>arrayWeights.args</code>,</li>
<li>
<code>wsva.args</code>.</li>
<li><code>useDuplicateCorrelation</code></li>
<li><code>correlationBlockVariable</code></li>
<li><code>duplicateCorrelation.args</code></li>
</ul>
<p>In this regard, when using limma-voom strategy, the <code>useVoomWithQualityWeights</code> parameter tells MIRit whether to use <code>voomWithWualityWeights()</code> instead of the standard <code>voom()</code> function. In the same way, for microarray studies, the <code>useArrayWeights</code> specifies whether to consider array quality weights during the <code>limma</code> pipeline. Similarly, <code>useWsva</code> can be set to TRUE to include a weighted surrogate variable analysis for batch effect correction. Moreover, <code>useDuplicateCorrelation</code> can be set to TRUE if you want to consider the effect of correlated samples through the <code>duplicateCorrelation()</code> function in <code>limma</code>. In this concern, the <code>correlationBlockVariable</code> specifies the blocking variable to use. All the other parameters ending with “<em>.args</em>”, accept a <code>list</code> object with additional parameters to be passed to the relative functions. In this way, the user has <strong>full control</strong> over the strategy used for differential expression analysis.</p>
<p>For a complete reference on the usage of these parameters, check the help page of these functions. Instead, for further instructions on how to use these tools, please refer to their original manuals, which represent exceptional resources for learning the basics of differential expression analysis:</p>
<ul>
<li>limma User’s Guide,</li>
<li>edgeR User’s Guide,</li>
<li>Analysing RNA-Seq data with DESeq2.</li>
</ul>
</div>
<div class="section level4" number="3.2.5">
<h4 id="add-differential-expression-results-from-other-technologies">
<span class="header-section-number">3.2.5</span> Add differential expression results from other technologies<a class="anchor" aria-label="anchor" href="#add-differential-expression-results-from-other-technologies"></a>
</h4>
<p>Even though MIRit implements all the most commonly used strategies for differential expression analyses, these methods may not be suitable for all kind of experiments. For instance, expression data deriving from technologies different from microarrays and RNA-Seq can’t be processed through <code><a href="../reference/deAnalysis.html">performGeneDE()</a></code> and <code><a href="../reference/deAnalysis.html">performMirnaDE()</a></code> functions. Therefore, MIRit grants the possibility to perform differential expression analysis with every method of choice, and then add the results to an existing <code>MirnaExperiment</code> object. This is particularly valuable for proteomic studies, where different normalization strategies are used in differential expression pipelines. In this way, MIRit fully supports the use of proteomic data for conducting miRNA integrative analyses.</p>
<p>To do so, we can make use of the <code><a href="../reference/addDifferentialExpression.html">addDifferentialExpression()</a></code> function, which allows to manually add the results of the analysis. This function takes as input a <code>MirnaExperiment</code> object, and a table containing the differential expression results for all miRNAs/genes analyzed, not just for statistically significant species. If we want to manually set differential expression results for both miRNAs and genes, two different tables must be supplied. These tables must include:</p>
<ul>
<li>One column containing miRNA/gene names (according to miRBase/HGNC nomenclature). Accepted column names are: <code>ID</code>, <code>Symbol</code>, <code>Gene_Symbol</code>, <code>Mirna</code>, <code>mir</code>, <code>Gene</code>, <code>gene.symbol</code>, <code>Gene.symbol</code>.</li>
<li>One column with log2 fold changes. Accepted column names are: <code>logFC</code>, <code>log2FoldChange</code>, <code>FC</code>, <code>lFC</code>.</li>
<li>One column with average expression. Accepted column names are: <code>AveExpr</code>, <code>baseMean</code>, <code>logCPM</code>.</li>
<li>One column with the p-values resulting from the differential expression analysis. Accepted column names are: <code>P.Value</code>, <code>pvalue</code>, <code>PValue</code>, <code>Pvalue</code>.</li>
<li>One column containing p-values adjusted for multiple testing. Accepted column names are: <code>adj.P.Val</code>, <code>padj</code>, <code>FDR</code>, <code>fdr</code>, <code>adj</code>, <code>adj.p</code>, <code>adjp</code>.</li>
</ul>
<p>Further, we must specify the cutoff levels used to consider miRNAs/genes as significantly differentially expressed. This can be done through the <code>mirna.logFC</code>, <code>mirna.pCutoff</code>, <code>mirna.pAdjustment</code>, <code>gene.logFC</code>, <code>gene.pCutoff</code>, <code>gene.pAdjustment</code> parameters.</p>
</div>
</div>
<div class="section level3" number="3.3">
<h3 id="visualize-differentially-expressed-features">
<span class="header-section-number">3.3</span> Visualize differentially expressed features<a class="anchor" aria-label="anchor" href="#visualize-differentially-expressed-features"></a>
</h3>
<div class="section level4" number="3.3.1">
<h4 id="access-differential-expression-tables">
<span class="header-section-number">3.3.1</span> Access differential expression tables<a class="anchor" aria-label="anchor" href="#access-differential-expression-tables"></a>
</h4>
<p>Once differential expression analysis has been performed, we can use the <code><a href="../reference/deAccessors.html">mirnaDE()</a></code> and <code><a href="../reference/deAccessors.html">geneDE()</a></code> functions to access a table with differentially expressed features. Therefore, let’s access the results of differential expression analysis in thyroid cancer for both miRNAs and genes.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## access DE results for genes</span></span>
<span><span class="va">deGenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deAccessors.html">geneDE</a></span><span class="op">(</span><span class="va">experiment</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## access DE results for miRNAs</span></span>
<span><span class="va">deMirnas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deAccessors.html">mirnaDE</a></span><span class="op">(</span><span class="va">experiment</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4" number="3.3.2">
<h4 id="create-a-volcano-plot-for-mirnas-and-genes">
<span class="header-section-number">3.3.2</span> Create a volcano plot for miRNAs and genes<a class="anchor" aria-label="anchor" href="#create-a-volcano-plot-for-mirnas-and-genes"></a>
</h4>
<p>In addition to differential expression tables, we can also generate a graphical overview of differential expression through volcano plots, which are extremely useful for visualizing features changing across biological conditions. To produce volcano plots, MIRit offers the <code><a href="../reference/plotVolcano.html">plotVolcano()</a></code> function.</p>
<p>In our example, let’s create volcano plots for both miRNA and gene differential expression.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## create a volcano plot for genes</span></span>
<span><span class="va">geneVolcano</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotVolcano.html">plotVolcano</a></span><span class="op">(</span><span class="va">experiment</span>,</span>
<span>    assay <span class="op">=</span> <span class="st">"genes"</span>,</span>
<span>    title <span class="op">=</span> <span class="st">"Gene differential expression"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## create a volcano plot for miRNAs</span></span>
<span><span class="va">mirnaVolcano</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotVolcano.html">plotVolcano</a></span><span class="op">(</span><span class="va">experiment</span>,</span>
<span>    assay <span class="op">=</span> <span class="st">"microRNA"</span>,</span>
<span>    title <span class="op">=</span> <span class="st">"miRNA differential expression"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## plot graphs side by side</span></span>
<span><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span><span class="va">geneVolcano</span>, <span class="va">mirnaVolcano</span>,</span>
<span>    nrow <span class="op">=</span> <span class="fl">1</span>, labels <span class="op">=</span> <span class="st">"AUTO"</span>,</span>
<span>    common.legend <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:volcano"></span>
<img src="MIRit_files/figure-html/volcano-1.png" alt="Volcano plots of gene and miRNA differential expression." width="100%" class="widefigure"><p class="caption">
Figure 2: <span class="caption-title">Volcano plots of gene and miRNA differential expression</span><br></p>
</div>
</div>
<div class="section level4" number="3.3.3">
<h4 id="produce-differential-expression-bar-plots">
<span class="header-section-number">3.3.3</span> Produce differential expression bar plots<a class="anchor" aria-label="anchor" href="#produce-differential-expression-bar-plots"></a>
</h4>
<p>Finally, if we are interested in specific genes/miRNAs, MIRit implements the <code><a href="../reference/plotDE.html">plotDE()</a></code> function that allows to represent expression changes of specific features as box plots, bar plots, or violin plots. In our example, we can use this function to visualize expression changes of different genes involved in the normal functioning of thyroid gland. Note that we use the <code>linear = FALSE</code> option to plot data in log scale (useful when multiple genes have very different expression levels).</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## create a bar plot for all thyroid features</span></span>
<span><span class="va">thyrBar</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotDE.html">plotDE</a></span><span class="op">(</span><span class="va">experiment</span>,</span>
<span>    features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TG"</span>, <span class="st">"TPO"</span>, <span class="st">"DIO2"</span>, <span class="st">"PAX8"</span><span class="op">)</span>,</span>
<span>    graph <span class="op">=</span> <span class="st">"barplot"</span>, linear <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## show the resulting plot</span></span>
<span><span class="va">thyrBar</span></span>
<span><span class="co">#&gt; Warning: Computation failed in `stat_summary()`</span></span>
<span><span class="co">#&gt; Caused by error in `get()`:</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> object 'mean_sd' of mode 'function' was not found</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:thyroidBars"></span>
<img src="MIRit_files/figure-html/thyroidBars-1.png" alt="Differential expression bar plots for different thyroid genes. Differential expression analysis demonstrated how TG, TPO, DIO2 and PAX8 result downregulated in thyroid cancer." width="100%"><p class="caption">
Figure 3: <span class="caption-title">Differential expression bar plots for different thyroid genes</span><br>Differential expression analysis demonstrated how TG, TPO, DIO2 and PAX8 result downregulated in thyroid cancer.
</p>
</div>
</div>
</div>
</div>
<div class="section level2" number="4">
<h2 id="functional-enrichment-analysis">
<span class="header-section-number">4</span> Functional enrichment analysis<a class="anchor" aria-label="anchor" href="#functional-enrichment-analysis"></a>
</h2>
<p>After finding differentially expressed genes, we usually end up having long lists of features whose expression changes between biological conditions. However, this is usually not very informative, and we seek to understand which biological functions result impaired in our experiments. In this regard, different methods exist for determining which cellular processes are dysregulated in our conditions.</p>
<div class="section level3" number="4.1">
<h3 id="available-approaches-ora-gsea-and-camera">
<span class="header-section-number">4.1</span> Available approaches: ORA, GSEA and CAMERA<a class="anchor" aria-label="anchor" href="#available-approaches-ora-gsea-and-camera"></a>
</h3>
<p>In particular, MIRit supports different strategies for functional enrichment of genes, including over-representation analysis (ORA), gene-set enrichment analysis (GSEA), and Correlation Adjusted MEan RAnk gene set test (CAMERA). In this way, the user can infer compromised biological functions according to the approach of choice.</p>
<p>Among these methods, the first one that has been developed is named <strong>over-representation analysis</strong> <span class="citation">(<a href="#ref-boyle_gotermfinderopen_2004">Boyle et al. 2004</a>)</span>, often abbreviated as <strong>ORA</strong>. This analysis aims to define whether genes annotated to specific gene sets (such as ontological terms or biological pathways) are differentially expressed more than would be expected by chance. To do this, a p-value is calculated by the hypergeometric distribution for each gene set as in Equation <a href="#eq:hyper">(1)</a>.</p>
<p><span class="math display" id="eq:hyper">\[\begin{equation}
  p = 1 - \sum_{i = 0}^{k - 1}{\frac{\binom{M}{i}\binom{N - M}
  {n - 1}}{\binom{N}{n}}}
  \tag{1}
\end{equation}\]</span></p>
<p>Here, <span class="math inline">\(N\)</span> is the total number of genes tested, <span class="math inline">\(M\)</span> is the number of genes that are annotated to a particular gene set, <span class="math inline">\(n\)</span> is the number of differentially expressed genes, and <span class="math inline">\(k\)</span> is the number of differentially expressed genes that are annotated to the gene set.</p>
<p>Additionally, another available approach is the <strong>gene set enrichment analysis</strong> <span class="citation">(<a href="#ref-subramanian_gene_2005">Subramanian et al. 2005</a>)</span>, often refereed to with the acronym <strong>GSEA</strong>, which is suitable to find categories whose genes change in a small but coordinated way. The GSEA algorithm takes as input a list of genes ranked with a particular criterion, and then walks down the list to evaluate whether members of a given gene set are normally distributed or are mainly present at the top or at the bottom of the list. To check this out, the algorithm uses a running-sum that increases when finding a gene belonging to a given category, and decreases when a gene not contained in that specific set is found. The maximum distance from zero occurred in the running score is defined as the enrichment score (ES). To estimate the statistical significance of enrichment scores, a permutation test is performed by swapping gene labels annotated to a gene set.</p>
<p>Even though GSEA is arguably the most commonly used approach for functional enrichment, <span class="citation">Wu and Smyth (<a href="#ref-wu_camera_2012">2012</a>)</span> demonstrated that inter-gene correlations might affect the reliability of functional enrichment analyses. To overcome this issue, they developed <strong>Correlation Adjusted MEan RAnk gene set test (CAMERA)</strong>, which is another competitive test used for functional enrichment analysis of genes. The main advantage of this method is that it adjusts the gene set test statistic according to inter-gene correlations.</p>
</div>
<div class="section level3" number="4.2">
<h3 id="categories">
<span class="header-section-number">4.2</span> Available databases and categories<a class="anchor" aria-label="anchor" href="#categories"></a>
</h3>
<p>As described above, functional enrichment analysis relies on gene sets, which consist in collections of genes that are annotated to specific functions or terms. Independently from the strategy used for the analysis, functional enrichment methods need access to these properly curated collections of genes. In the effort of providing access to a vast number of these resources, MIRit uses the <em><a href="https://bioconductor.org/packages/3.17/geneset" class="external-link">geneset</a></em> package to support multiple databases, including Gene Ontology (GO), Kyoto Encyclopedia of Genes and Genomes (KEGG), MsigDB, WikiPathways, Reactome, Enrichr, Disease Ontology (DO), Network of Cancer Genes (NCG), DisGeNET, and COVID-19. However, the majority of these databases contain multiple categories. To see the complete list of available gene sets for each database refer to the documentation of the <code><a href="../reference/enrichGenes.html">enrichGenes()</a></code> function.</p>
</div>
<div class="section level3" number="4.3">
<h3 id="supported-species">
<span class="header-section-number">4.3</span> Supported species<a class="anchor" aria-label="anchor" href="#supported-species"></a>
</h3>
<p>The above-mentioned collections have their own lists of supported species. To check the available species for a given database, MIRit provides a practical helper function named <code><a href="../reference/supportedOrganisms.html">supportedOrganisms()</a></code>. For example, to retrieve the species supported by Reactome database, we can simply run the following piece of code.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## list available species for Reactome database</span></span>
<span><span class="fu"><a href="../reference/supportedOrganisms.html">supportedOrganisms</a></span><span class="op">(</span><span class="st">"Reactome"</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "Bos taurus"               "Caenorhabditis elegans"  </span></span>
<span><span class="co">#&gt;  [3] "Danio rerio"              "Drosophila melanogaster" </span></span>
<span><span class="co">#&gt;  [5] "Gallus gallus"            "Homo sapiens"            </span></span>
<span><span class="co">#&gt;  [7] "Mus musculus"             "Rattus norvegicus"       </span></span>
<span><span class="co">#&gt;  [9] "Saccharomyces cerevisiae" "Sus scrofa"              </span></span>
<span><span class="co">#&gt; [11] "Xenopus tropicalis"</span></span></code></pre></div>
</div>
<div class="section level3" number="4.4">
<h3 id="enrichment">
<span class="header-section-number">4.4</span> Perform functional enrichment with the <code>enrichGenes()</code> function<a class="anchor" aria-label="anchor" href="#enrichment"></a>
</h3>
<p>The main function in MIRit for the functional enrichment analysis of genes is <code><a href="../reference/enrichGenes.html">enrichGenes()</a></code>, which requires as input:</p>
<ul>
<li>the <code>MirnaExperiment</code> object that we get after running differential expression analysis;</li>
<li>
<code>method</code>, which specifies the desired approach among <code>ORA</code>, <code>GSEA</code>, and <code>CAMERA</code>;</li>
<li>
<code>database</code> and <code>category</code>, which define the gene set that you want to use for the enrichment (see Section <a href="#categories">4.2</a> for a complete reference of available databases and categories);</li>
<li>
<code>organism</code>, which indicates the specie under investigation (defaults to “Homo sapiens”);</li>
<li>
<code>pCutoff</code> and <code>pAdjustment</code>, which specify the cutoff for statistical significance and the multiple testing correction, respectively.</li>
</ul>
<p>In our example, we are going to perform the enrichment analysis by using ORA with GO database (biological processes).</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## perform over-representation analysis with GO</span></span>
<span><span class="va">ora</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/enrichGenes.html">enrichGenes</a></span><span class="op">(</span><span class="va">experiment</span>, method <span class="op">=</span> <span class="st">"ORA"</span>, database <span class="op">=</span> <span class="st">"GO"</span>, organism <span class="op">=</span> <span class="st">"Homo sapiens"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Since not specified, 'category' for GO database is set to bp (default).</span></span>
<span><span class="co">#&gt; Preparing the appropriate gene set...</span></span>
<span><span class="co">#&gt; Performing the enrichment of upregulated genes...</span></span>
<span><span class="co">#&gt; Performing the enrichment of downregulated genes...</span></span>
<span><span class="co">#&gt; The enrichment of genes reported 11 significantly enriched terms for downregulated genes and 0 for upregulated genes.</span></span></code></pre></div>
<p>MIRit performs ORA separately for upregulated and downregulated genes, as it has been shown that this is more powerful compared to enriching all DE genes <span class="citation">(<a href="#ref-hong_separate_2014-1">Hong et al. 2014</a>)</span>. Therefore, when we use ORA, <code><a href="../reference/enrichGenes.html">enrichGenes()</a></code> returns a <code>list</code> containing two objects of class <code>FunctionalEnrichment</code>, one storing enrichment results for upregulated genes, and one for downregulated genes.</p>
<p>Before exploring the results of the analysis, we will also demonstrate the capabilities of MIRit by performing GSEA with the gene sets provided by the KEGG pathway database.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## perform gene-set enrichment analysis with KEGG</span></span>
<span><span class="va">gse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/enrichGenes.html">enrichGenes</a></span><span class="op">(</span><span class="va">experiment</span>, method <span class="op">=</span> <span class="st">"GSEA"</span>, database <span class="op">=</span> <span class="st">"KEGG"</span>, organism <span class="op">=</span> <span class="st">"Homo sapiens"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Since not specified, 'category' for KEGG database is set to pathway (default).</span></span>
<span><span class="co">#&gt; Preparing the appropriate gene set...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Registered S3 methods overwritten by 'treeio':</span></span>
<span><span class="co">#&gt;   method              from    </span></span>
<span><span class="co">#&gt;   MRCA.phylo          tidytree</span></span>
<span><span class="co">#&gt;   MRCA.treedata       tidytree</span></span>
<span><span class="co">#&gt;   Nnode.treedata      tidytree</span></span>
<span><span class="co">#&gt;   Ntip.treedata       tidytree</span></span>
<span><span class="co">#&gt;   ancestor.phylo      tidytree</span></span>
<span><span class="co">#&gt;   ancestor.treedata   tidytree</span></span>
<span><span class="co">#&gt;   child.phylo         tidytree</span></span>
<span><span class="co">#&gt;   child.treedata      tidytree</span></span>
<span><span class="co">#&gt;   full_join.phylo     tidytree</span></span>
<span><span class="co">#&gt;   full_join.treedata  tidytree</span></span>
<span><span class="co">#&gt;   groupClade.phylo    tidytree</span></span>
<span><span class="co">#&gt;   groupClade.treedata tidytree</span></span>
<span><span class="co">#&gt;   groupOTU.phylo      tidytree</span></span>
<span><span class="co">#&gt;   groupOTU.treedata   tidytree</span></span>
<span><span class="co">#&gt;   inner_join.phylo    tidytree</span></span>
<span><span class="co">#&gt;   inner_join.treedata tidytree</span></span>
<span><span class="co">#&gt;   is.rooted.treedata  tidytree</span></span>
<span><span class="co">#&gt;   nodeid.phylo        tidytree</span></span>
<span><span class="co">#&gt;   nodeid.treedata     tidytree</span></span>
<span><span class="co">#&gt;   nodelab.phylo       tidytree</span></span>
<span><span class="co">#&gt;   nodelab.treedata    tidytree</span></span>
<span><span class="co">#&gt;   offspring.phylo     tidytree</span></span>
<span><span class="co">#&gt;   offspring.treedata  tidytree</span></span>
<span><span class="co">#&gt;   parent.phylo        tidytree</span></span>
<span><span class="co">#&gt;   parent.treedata     tidytree</span></span>
<span><span class="co">#&gt;   root.treedata       tidytree</span></span>
<span><span class="co">#&gt;   rootnode.phylo      tidytree</span></span>
<span><span class="co">#&gt;   sibling.phylo       tidytree</span></span>
<span><span class="co">#&gt; Some ID occurs one-to-many match, like "79154, 7920, 79143"...</span></span>
<span><span class="co">#&gt; 99.96% genes are mapped to symbol</span></span>
<span><span class="co">#&gt; Ranking genes based on signed.pval...</span></span>
<span><span class="co">#&gt; Performing gene-set enrichment analysis (GSEA)...</span></span>
<span><span class="co">#&gt; GSEA reported 3 significantly enriched terms.</span></span></code></pre></div>
<p>In this case, the <code><a href="../reference/enrichGenes.html">enrichGenes()</a></code> function returns a single object of class <code>FunctionalEnrichment</code>, containing GSEA results.</p>
</div>
<div class="section level3" number="4.5">
<h3 id="visualize-enriched-sets">
<span class="header-section-number">4.5</span> Visualize enriched sets<a class="anchor" aria-label="anchor" href="#visualize-enriched-sets"></a>
</h3>
<div class="section level4" number="4.5.1">
<h4 id="access-results-table">
<span class="header-section-number">4.5.1</span> Access results table<a class="anchor" aria-label="anchor" href="#access-results-table"></a>
</h4>
<p>After running the <code><a href="../reference/enrichGenes.html">enrichGenes()</a></code> function, we get <code>FunctionalEnrichment</code> objects holding the results of enrichment analyses. To access the full table containing significantly affected biological functions, we can use the <code><a href="../reference/enrichmentResults.html">enrichmentResults()</a></code> function. In our case, we can check the results of our GSEA analysis to investigate the human pathways that result affected in thyroid cancer. In Table <a href="#tab:gseaTab">3</a> we can see the output of <code>enrichmentResults(gse)</code>.</p>
<div class="horizontal-scroll"><table class="table">
<caption>
<span id="tab:gseaTab">Table 3: </span><span class="caption-title">GSEA results</span><br>A table listing the affected KEGG pathways in thyroid cancer.</caption>
<colgroup>
<col width="47%">
<col width="5%">
<col width="5%">
<col width="8%">
<col width="6%">
<col width="6%">
<col width="5%">
<col width="14%">
</colgroup>
<thead><tr class="header">
<th align="left">pathway</th>
<th align="right">pval</th>
<th align="right">padj</th>
<th align="right">log2err</th>
<th align="right">ES</th>
<th align="right">NES</th>
<th align="right">size</th>
<th align="left">leadingEdge</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Thyroid hormone synthesis</td>
<td align="right">0</td>
<td align="right">0.03</td>
<td align="right">0.52</td>
<td align="right">-0.67</td>
<td align="right">-1.98</td>
<td align="right">27</td>
<td align="left">SLC26A4,….</td>
</tr>
<tr class="even">
<td align="left">Protein processing in endoplasmic reticulum</td>
<td align="right">0</td>
<td align="right">0.03</td>
<td align="right">0.50</td>
<td align="right">-0.54</td>
<td align="right">-1.86</td>
<td align="right">55</td>
<td align="left">BCL2, HY….</td>
</tr>
<tr class="odd">
<td align="left">Coronavirus disease - COVID-19</td>
<td align="right">0</td>
<td align="right">0.05</td>
<td align="right">0.48</td>
<td align="right">-0.45</td>
<td align="right">-1.67</td>
<td align="right">85</td>
<td align="left">JUN, RPL….</td>
</tr>
</tbody>
</table></div>
</div>
<div class="section level4" number="4.5.2">
<h4 id="enrichment-dot-plots-and-bar-plots">
<span class="header-section-number">4.5.2</span> Enrichment dot plots and bar plots<a class="anchor" aria-label="anchor" href="#enrichment-dot-plots-and-bar-plots"></a>
</h4>
<p>Further, in addition to exploring results table, MIRit offers several options for the visualization of enrichment analyses, including dot plots and bar plots. These plots are available for every <code>FunctionalEnrichment</code> object independently from the method used.</p>
<p>Following our example, we can visualize the results of the ORA that we performed in Section <a href="#enrichment">4.4</a> through a simple dot plot.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## create a dot plot for ORA</span></span>
<span><span class="fu"><a href="../reference/enrichmentDotplot.html">enrichmentDotplot</a></span><span class="op">(</span><span class="va">ora</span><span class="op">$</span><span class="va">downregulated</span>, title <span class="op">=</span> <span class="st">"Depleted functions"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:oraDot"></span>
<img src="MIRit_files/figure-html/oraDot-1.png" alt="ORA results for downregulated genes. The enrichment of downregulated genes through the gene sets provided by GO database." width="100%"><p class="caption">
Figure 4: <span class="caption-title">ORA results for downregulated genes</span><br>The enrichment of downregulated genes through the gene sets provided by GO database.
</p>
</div>
</div>
<div class="section level4" number="4.5.3">
<h4 id="other-plots-for-gsea">
<span class="header-section-number">4.5.3</span> Other plots for GSEA<a class="anchor" aria-label="anchor" href="#other-plots-for-gsea"></a>
</h4>
<p>Additionally, MIRit provides specific visualization methods that are exclusive for GSEA, including ridge plots and GSEA plots. For instance, after running GSEA through <code><a href="../reference/enrichGenes.html">enrichGenes()</a></code>, we can produce a GSEA-style plot through the <code><a href="../reference/gseaPlot.html">gseaPlot()</a></code> function. In our case, we are going to use this plotting method for the “Thyroid hormone synthesis” pathway.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## create a GSEA plot</span></span>
<span><span class="fu"><a href="../reference/gseaPlot.html">gseaPlot</a></span><span class="op">(</span><span class="va">gse</span>, <span class="st">"Thyroid hormone synthesis"</span>, rankingMetric <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:gsePlot"></span>
<img src="MIRit_files/figure-html/gsePlot-1.png" alt="GSEA-style plot for Thyroid hormone synthesis. This type of plot shows the running sum that GSEA uses to determinate the enrichment score for each pathway." width="100%"><p class="caption">
Figure 5: <span class="caption-title">GSEA-style plot for Thyroid hormone synthesis</span><br>This type of plot shows the running sum that GSEA uses to determinate the enrichment score for each pathway.
</p>
</div>
</div>
</div>
</div>
<div class="section level2" number="5">
<h2 id="associate-mirnas-with-disease-snps">
<span class="header-section-number">5</span> Associate miRNAs with disease-SNPs<a class="anchor" aria-label="anchor" href="#associate-mirnas-with-disease-snps"></a>
</h2>
<p>Interestingly, MIRit enables to explore the presence of disease-associated SNPs occurring at loci of differentially expressed miRNAs. In this concern, SNPs occurring within miRNAs may have important effects on the biological function of these transcripts, as they might alter its expression or the spectrum of miRNA targets. To verify the presence of disease-SNPs within miRNA loci, MIRit directly queries the NHGRI-EBI Catalog of published genome-wide association studies through the <em><a href="https://CRAN.R-project.org/package=gwasrapidd" class="external-link">gwasrapidd</a></em> package, and then retains only SNPs that affect DE-miRNA genes or their relative host genes (for intragenic miRNAs).</p>
<p>In our case, there are no SNPs associated with thyroid cancer that occur witin DE-miRNA loci. Therefore, we will explain how this function works with reference to the analysis on Alzheimer’s disease reported in the MIRit paper.</p>
<div class="section level3" number="5.1">
<h3 id="search-disease-related-snps">
<span class="header-section-number">5.1</span> Search disease-related SNPs<a class="anchor" aria-label="anchor" href="#search-disease-related-snps"></a>
</h3>
<p>First, we need to identify the Experimental Factor Ontology (EFO) identifier of a given disease of interest. To do so, MIRit provides the <code><a href="../reference/searchDisease.html">searchDisease()</a></code> function. For example, to identify the relevant EFO ID for Alzheimer disease we can use the following code chunk.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## identify the EFO ID corresponding to Alzheimer's disease</span></span>
<span><span class="fu"><a href="../reference/searchDisease.html">searchDisease</a></span><span class="op">(</span><span class="st">"alzheimer"</span><span class="op">)</span></span></code></pre></div>
<p>The relevant EFO trait is “Alzheimer disease”.</p>
</div>
<div class="section level3" number="5.2">
<h3 id="identify-mirna-genes-overlapping-with-disease-snps">
<span class="header-section-number">5.2</span> Identify miRNA genes overlapping with disease-SNPs<a class="anchor" aria-label="anchor" href="#identify-mirna-genes-overlapping-with-disease-snps"></a>
</h3>
<p>Now, we can use the <code><a href="../reference/findMirnaSNPs.html">findMirnaSNPs()</a></code> function to identify the disease-related SNPs that affect miRNA loci.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## detect disease-SNPs occuring at DE-miRNAs loci</span></span>
<span><span class="va">association</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findMirnaSNPs.html">findMirnaSNPs</a></span><span class="op">(</span><span class="va">experiment</span>, <span class="st">"Alzheimer disease"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="5.3">
<h3 id="build-a-track-plot-to-display-mirna-snp-associations">
<span class="header-section-number">5.3</span> Build a track plot to display miRNA-SNP associations<a class="anchor" aria-label="anchor" href="#build-a-track-plot-to-display-mirna-snp-associations"></a>
</h3>
<p>Finally, if any disease-related SNPs is present within DE-miRNA loci, we can use the <code><a href="../reference/mirVariantPlot.html">mirVariantPlot()</a></code> function to graphically build a track plot displaying the polymorphism along with the relevant genomic context, including the corresponding miRNA locus.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## create a track plot to represent disease-SNPs at DE-miRNA loci</span></span>
<span><span class="fu"><a href="../reference/mirVariantPlot.html">mirVariantPlot</a></span><span class="op">(</span><span class="st">"rs2632516"</span>, <span class="va">association</span>, showContext <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2" number="6">
<h2 id="retrieve-mirna-targets">
<span class="header-section-number">6</span> Retrieve miRNA targets<a class="anchor" aria-label="anchor" href="#retrieve-mirna-targets"></a>
</h2>
<p>Before performing integrative miRNA-mRNA analyses, we need to identify the targets of differentially expressed miRNAs, so that we can test whether they really affect the levels of their targets or not.</p>
<div class="section level3" number="6.1">
<h3 id="databases-with-mirna-mrna-interactions">
<span class="header-section-number">6.1</span> Databases with miRNA-mRNA interactions<a class="anchor" aria-label="anchor" href="#databases-with-mirna-mrna-interactions"></a>
</h3>
<p>Different resources have been developed over the years to predict and collect miRNA-target interactions, and we can categorize them in two main types:</p>
<ul>
<li>
<strong>Prediction databases</strong>, that contain information about computationally determined miRNA-target interactions;</li>
<li>
<strong>Validated databases</strong>, which only contain interactions that have been proven through biomolecular experiments.</li>
</ul>
<p>The choice of which type of resources to use for identifying miRNA targets drastically influences the outcome of the analysis. In this regard, some researchers tend to give the priority to validated interactions, even though they are usually fewer than predicted ones. On the other hand, predicted pairs are much more numerous, but they exhibit a high number of false positive hits.</p>
</div>
<div class="section level3" number="6.2">
<h3 id="the-mirdip-approach">
<span class="header-section-number">6.2</span> The mirDIP approach<a class="anchor" aria-label="anchor" href="#the-mirdip-approach"></a>
</h3>
<p>The downside of miRNA target prediction algorithms is also the scarce extend of overlap existing between different tools. To address this issue, several ensemble methods have been developed, trying to aggregate the predictions obtained by different algorithms. Initially, several researchers determined as significant miRNA-target pairs those predicted by more than one tool (intersection method). However, this method is not able to capture an important number of meaningful interactions. Alternatively, other strategies used to merge predictions from several algorithms (union method). Despite identifying more true relationships, the union method leads to a higher proportion of false discoveries. Therefore, other ensemble methods started using other statistics to rank miRNA-target predictions obtained by multiple algorithms. Among these newly developed ensemble methods, one of the best performing one is the <strong>microRNA Data Integration Portal (mirDIP)</strong> database, which aggregates miRNA target predictions from 24 different resources by using an integrated score inferred from different prediction metrics. In this way, mirDIP reports more accurate predictions compared to those of individual tools. For additional information on mirDIP database and its ranking metric refer to <span class="citation">Tokar et al. (<a href="#ref-tokar_mirdip_2018">2018</a>)</span> and <span class="citation">Hauschild et al. (<a href="#ref-hauschild_mirdip_2023">2023</a>)</span>.</p>
</div>
<div class="section level3" number="6.3">
<h3 id="download-predicted-and-validated-interactions-with-gettargets">
<span class="header-section-number">6.3</span> Download predicted and validated interactions with <code>getTargets()</code><a class="anchor" aria-label="anchor" href="#download-predicted-and-validated-interactions-with-gettargets"></a>
</h3>
<p>Given the above, MIRit allows the prediction of miRNA-target interactions via the <strong>mirDIP</strong> database (version 5.2). In addition, in order to raise the number of true interactions, MIRit combines the miRNA-target pairs returned by mirDIP with the experimentally validated interactions contained in the <strong>miRTarBase</strong> database (version 9) <span class="citation">(<a href="#ref-huang_mirtarbase_2022">Huang et al. 2022</a>)</span>. In practice, to identify miRNA targets MIRit implements the <code><a href="../reference/getTargets.html">getTargets()</a></code> function, which allows to download both type of interactions. Specifically, this function also includes a parameter called <code>score</code> that determines the degree of confidence required for the targets predicted by mirDIP. The value of this parameter must be one of <code>Very High</code>, <code>High</code> (default), <code>Medium</code>, and <code>Low</code>, which correspond to ranks among top 1%, top 5% (excluding top 1%), top 1/3 (excluding top 5%) and remaining predictions, respectively. Moreover, the <code>includeValidated</code> parameter tells MIRit whether to include experimentally validate interactions deriving from miRTarBase (default is TRUE). Please note that mirDIP database is only available for human miRNAs; thus, for species other than Homo sapiens, only validated interactions contained in miRTarBase are used.</p>
<p>In our example, we are going to retrieve both predicted and validated interactions by using default settings.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## retrieve miRNA target genes</span></span>
<span><span class="va">experiment</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getTargets.html">getTargets</a></span><span class="op">(</span><span class="va">experiment</span><span class="op">)</span></span></code></pre></div>
<p>After running this function, we obtain a <code>MirnaExperiment</code> object containing miRNA-target interactions in its <code>targets</code> slot. The user can access a <code>data.frame</code> detailing these interactions through the <code><a href="../reference/mirnaTargets.html">mirnaTargets()</a></code> function.</p>
</div>
</div>
<div class="section level2" number="7">
<h2 id="investigate-the-effects-of-mirna-expression-changes-on-target-genes">
<span class="header-section-number">7</span> Investigate the effects of miRNA expression changes on target genes<a class="anchor" aria-label="anchor" href="#investigate-the-effects-of-mirna-expression-changes-on-target-genes"></a>
</h2>
<p>Now that we have defined the targets of differentially expressed miRNAs, we can continue with the integrative analysis of miRNA and gene expression levels. This analysis is useful as it allows to only consider miNA-target pairs where an inverse relationship is observed. As already mentioned, MIRit can work with both paired and unpaired data by using different statistical approaches, including:</p>
<ul>
<li>
<strong>Correlation analysis</strong>, which is the recommended method when samples are paired;</li>
<li>
<strong>Association tests</strong>, like Fisher’s exact test and Boschloo’s exact test;</li>
<li>
<strong>Rotation gene-set tests</strong>, as implemented in the <code>fry</code> function from <em><a href="https://bioconductor.org/packages/3.17/limma" class="external-link">limma</a></em> package.</li>
</ul>
<p>For unpaired data, only association tests and rotation gene-set tests are available, whereas correlation analysis is the best performing strategy for paired data. The integrative analysis, either performed through correlation, association tests, or rotation gene-set tests, is implemented in the <code><a href="../reference/mirnaIntegration.html">mirnaIntegration()</a></code> function. When using the default option <code>test = "auto"</code>, MIRit automatically performs the appropriate test for paired and unpaired samples. If only some samples of the data set have paired measurements, a correlation analysis will be carried out only for those subjects.</p>
<div class="section level3" number="7.1">
<h3 id="correlation-analysis-for-paired-data">
<span class="header-section-number">7.1</span> Correlation analysis for paired data<a class="anchor" aria-label="anchor" href="#correlation-analysis-for-paired-data"></a>
</h3>
<p>When both miRNA and gene expression measurements are available for the same samples, a correlation analysis is the recommended procedure. In statistics, correlation is a measure that expresses the extent to which two random variables are dependent. In our case, we want to assess whether a statistical relationship is present between the expression of a miRNA and the expression of its targets.</p>
<div class="section level4" number="7.1.1">
<h4 id="statistical-correlation-coefficients">
<span class="header-section-number">7.1.1</span> Statistical correlation coefficients<a class="anchor" aria-label="anchor" href="#statistical-correlation-coefficients"></a>
</h4>
<p>Several statistical coefficients can be used to weigh the degree of a correlation. Among them, the most commonly used are <em>Pearson’s correlation coefficient</em> <span class="math inline">\(r\)</span>, <em>Spearman’s correlation coefficient</em> <span class="math inline">\(\rho\)</span>, and <em>Kendall’s Tau-b correlation coefficient</em> <span class="math inline">\(\tau_b\)</span>. Pearson’s <span class="math inline">\(r\)</span> is probably the most diffused for determining the association between miRNA and gene expression. However, it assumes that the relationships between miRNA and gene expression values is linear. This is typically not true for miRNAs, whose interactions with their targets are characterized by imperfect complementarity. Additionally, miRNAs can target multiple genes with different binding sites, and this implies that a simple linear relationship may not be sufficient to properly describe the complexity of these interactions. In contrast, Spearman’s and Kendall’s Tau-b correlation coefficients result more suitable for representing the interplay between miRNAs and target genes, because they are robust to non-linear relationships and outliers. However, Kendall’s correlation just relies on the number of concordant and discordant pairs, and is less sensitive then Spearman’s correlation; so, when many ties are present or when the sample size is small, it may have a lower detection power. This is the reason why <strong>Spearman’s correlation coefficient</strong> is the default coefficient used in the <code><a href="../reference/mirnaIntegration.html">mirnaIntegration()</a></code> function to measure the correlation between miRNA and gene expression. Moreover, since miRNAs mainly act as negative regulators, only negatively correlated miRNA-target pairs are considered, and statistical significance is estimated through a one-tailed t-test.</p>
</div>
<div class="section level4" number="7.1.2">
<h4 id="correlation">
<span class="header-section-number">7.1.2</span> Perform a correlation analysis in MIRit<a class="anchor" aria-label="anchor" href="#correlation"></a>
</h4>
<p>To sum up the steps that MIRit follows when evaluating the correlation between miRNAs and genes, what the <code><a href="../reference/mirnaIntegration.html">mirnaIntegration()</a></code> function does during a correlation analysis is to:</p>
<ol style="list-style-type: decimal">
<li>consider the miRNA-target interactions retrieved with the <code><a href="../reference/getTargets.html">getTargets()</a></code> function;</li>
<li>calculate the correlation coefficient for each miRNA-target pair based on their expression values;</li>
<li>compute the statistical significance of all miRNA-target pairs;</li>
<li>adjust p-values for multiple testing before reporting significant results.</li>
</ol>
<p>In our thyroid cancer example, we want to find the miRNA-target pairs that exhibit a negative correlation with a Spearman’s coefficient lower than -0.5 and with an adjusted p-value lower than 0.05.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## perform a correlation analysis</span></span>
<span><span class="va">experiment</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mirnaIntegration.html">mirnaIntegration</a></span><span class="op">(</span><span class="va">experiment</span>, test <span class="op">=</span> <span class="st">"correlation"</span><span class="op">)</span></span>
<span><span class="co">#&gt; As specified by the user, a correlation will be used.</span></span>
<span><span class="co">#&gt; Performing Spearman's correlation analysis...</span></span>
<span><span class="co">#&gt; A statistically significant correlation between 215 miRNA-target pairs was found!</span></span></code></pre></div>
<p>Please note that all the parameters used for the correlation analysis are customizable. For instance, the user can change the significance threshold and the multiple testing correction method by setting the <code>pCutoff</code> and <code>pAdjustment</code> parameters, respectively. Further, it is also possible to change the correlation coefficient used, by editing the <code>corMethod</code> option, and the minimum required value of the correlation coefficient, by changing the <code>corCutoff</code> setting.</p>
</div>
<div class="section level4" number="7.1.3">
<h4 id="account-for-batch-effects-prior-to-correlation-analysis">
<span class="header-section-number">7.1.3</span> Account for batch effects prior to correlation analysis<a class="anchor" aria-label="anchor" href="#account-for-batch-effects-prior-to-correlation-analysis"></a>
</h4>
<p>Sometimes, when exploring expression variability through MDS plots, as we do with the <code><a href="../reference/plotDimensions.html">plotDimensions()</a></code> function, we notice the presence of batch effects that prevent a clear separation of our biological groups. Indeed, batch effects consist in unwanted sources of technical variation that confound expression variability and limit downstream analyses. Since the reliability of biological conclusions of integrative miRNA-mRNA analyses depends on the correlation between miRNA and gene expression levels, it is pivotal to ensure that expression measurements are not affected by technical variation. In this regard, if batch effects are noticed in the data, MIRit provides the <code><a href="../reference/batchCorrection.html">batchCorrection()</a></code> function, which removes batch effects from expression data before moving to correlation analysis. Please note that this procedure can only be used prior to correlation analysis, because for differential expression analysis it is more appropriate to include batch variables in the linear model, as specified in Section <a href="#model">3.2.2</a>. For additional information, please refer to the manual of the <code><a href="../reference/batchCorrection.html">batchCorrection()</a></code> function.</p>
</div>
<div class="section level4" number="7.1.4">
<h4 id="explore-the-succesfully-integrated-mirna-target-pairs">
<span class="header-section-number">7.1.4</span> Explore the succesfully integrated miRNA-target pairs<a class="anchor" aria-label="anchor" href="#explore-the-succesfully-integrated-mirna-target-pairs"></a>
</h4>
<p>Before moving on to the identification of the altered miRNAs regulatory networks, we can explore correlated miRNA-target pairs thanks to the <code><a href="../reference/integration.html">integration()</a></code> function, which returns a <code>data.frame</code> object with comprehensive details about the computed correlations.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## extract correlation results</span></span>
<span><span class="va">integrationResults</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/integration.html">integration</a></span><span class="op">(</span><span class="va">experiment</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## take a look at correlation results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">integrationResults</span><span class="op">)</span></span>
<span><span class="co">#&gt;                     microRNA Target microRNA.Direction Corr.Coefficient</span></span>
<span><span class="co">#&gt; hsa.miR.1179.7  hsa-miR-1179  ITGA6      downregulated       -0.8617647</span></span>
<span><span class="co">#&gt; hsa.miR.1179.10 hsa-miR-1179  RBMS2      downregulated       -0.7764706</span></span>
<span><span class="co">#&gt; hsa.miR.1179.12 hsa-miR-1179 SPIRE1      downregulated       -0.7117647</span></span>
<span><span class="co">#&gt; hsa.miR.1179.14 hsa-miR-1179   TAF5      downregulated       -0.5970588</span></span>
<span><span class="co">#&gt; hsa.miR.1275.1  hsa-miR-1275  CCND2      downregulated       -0.7794118</span></span>
<span><span class="co">#&gt; hsa.miR.1275.2  hsa-miR-1275   CD44      downregulated       -0.8382353</span></span>
<span><span class="co">#&gt;                 Corr.P.Value Corr.Adjusted.P.Val</span></span>
<span><span class="co">#&gt; hsa.miR.1179.7  8.902592e-06        0.0006579303</span></span>
<span><span class="co">#&gt; hsa.miR.1179.10 2.021754e-04        0.0034840545</span></span>
<span><span class="co">#&gt; hsa.miR.1179.12 9.920216e-04        0.0072485216</span></span>
<span><span class="co">#&gt; hsa.miR.1179.14 7.304866e-03        0.0266876638</span></span>
<span><span class="co">#&gt; hsa.miR.1275.1  1.858401e-04        0.0034840545</span></span>
<span><span class="co">#&gt; hsa.miR.1275.2  2.505171e-05        0.0013423542</span></span></code></pre></div>
</div>
<div class="section level4" number="7.1.5">
<h4 id="visualize-the-correlation-between-mirnas-and-genes">
<span class="header-section-number">7.1.5</span> Visualize the correlation between miRNAs and genes<a class="anchor" aria-label="anchor" href="#visualize-the-correlation-between-mirnas-and-genes"></a>
</h4>
<p>Additionally, for correlation analyses, MIRit allows to graphically represent inverse correlations through a scatter plot. Indeed, we can make use of the <code><a href="../reference/plotCorrelation.html">plotCorrelation()</a></code> function to display the correlation between specific miRNA-target pairs. For example, the correlation analysis performed in Section <a href="#correlation">7.1.2</a> revealed how miR-146b-5p, the most upregulated miRNA, is inversely correlated with the expression of DIO2, which is crucial for thyroid hormone functioning. Furthermore, it has also emerged that miR-146b-3p results negatively correlated with PAX8, which directly induces TG transcription.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## plot the correlation between miR-146b-5p and DIO2</span></span>
<span><span class="va">cor1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotCorrelation.html">plotCorrelation</a></span><span class="op">(</span><span class="va">experiment</span>,</span>
<span>    mirna <span class="op">=</span> <span class="st">"hsa-miR-146b-5p"</span>,</span>
<span>    gene <span class="op">=</span> <span class="st">"DIO2"</span>, condition <span class="op">=</span> <span class="st">"disease"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## plot the correlation between miR-146b-3p and PAX8</span></span>
<span><span class="va">cor2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotCorrelation.html">plotCorrelation</a></span><span class="op">(</span><span class="va">experiment</span>,</span>
<span>    mirna <span class="op">=</span> <span class="st">"hsa-miR-146b-3p"</span>,</span>
<span>    gene <span class="op">=</span> <span class="st">"PAX8"</span>, condition <span class="op">=</span> <span class="st">"disease"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## plot graphs side by side</span></span>
<span><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span><span class="va">cor1</span>, <span class="va">cor2</span>,</span>
<span>    nrow <span class="op">=</span> <span class="fl">1</span>, labels <span class="op">=</span> <span class="st">"AUTO"</span>,</span>
<span>    common.legend <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:corPlot"></span>
<img src="MIRit_files/figure-html/corPlot-1.png" alt="Correlation between miRNAs and key thyroid genes. These plots suggest that the upregulation of miR-146b-5p and miR-146b-3p may be responsible for the downregulation of DIO2 and PAX8, respectively." width="100%" class="widefigure"><p class="caption">
Figure 6: <span class="caption-title">Correlation between miRNAs and key thyroid genes</span><br>These plots suggest that the upregulation of miR-146b-5p and miR-146b-3p may be responsible for the downregulation of DIO2 and PAX8, respectively.
</p>
</div>
</div>
</div>
<div class="section level3" number="7.2">
<h3 id="association-tests-for-unpaired-data">
<span class="header-section-number">7.2</span> Association tests for unpaired data<a class="anchor" aria-label="anchor" href="#association-tests-for-unpaired-data"></a>
</h3>
<p>For unpaired data, we cannot directly quantify the influence of miRNA expression on the levels of their targets, because we do not have any sample correspondence between miRNA and gene measurements. However, <strong>one-sided association tests</strong> can be applied in these cases to evaluate if targets of downregulated miRNAs are statistically enriched in upregulated genes, and, conversely, if targets of upregulated miRNAs are statistically enriched in downregulated genes. In this regard, to estimate the effects of differentially expressed miRNAs on their target genes, MIRit can use two different one-sided association tests, namely:</p>
<ul>
<li>
<strong>Fisher’s exact test</strong>,</li>
<li>
<strong>Boschloo’s exact test</strong> (default).</li>
</ul>
<p>Both these tests consist in a statistical procedure that estimates the association between two dichotomous categorical variables. In our case, for each miRNA, we want to evaluate whether the proportion of targets within the differentially expressed genes significantly differs from the proportion of targets in non-differentially expressed genes. To do this, a 2x2 contingency table is built as shown in Table <a href="#tab:contingency">4</a>.</p>
<div class="horizontal-scroll"><table class="table">
<caption>
<span id="tab:contingency">Table 4: </span><span class="caption-title">The 2x2 contingency table that MIRit uses for one-sided association tests</span><br>This is the table that the <code><a href="../reference/mirnaIntegration.html">mirnaIntegration()</a></code> function creates to determine if differentially expressed genes are enriched in miRNA targets.</caption>
<colgroup>
<col width="25%">
<col width="24%">
<col width="24%">
<col width="24%">
</colgroup>
<thead><tr class="header">
<th align="right"></th>
<th align="center">Target genes</th>
<th align="center">Non target genes</th>
<th align="center">Row total</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right"><strong>Differentially expressed</strong></td>
<td align="center"><span class="math inline">\(a\)</span></td>
<td align="center"><span class="math inline">\(b\)</span></td>
<td align="center"><span class="math inline">\(a + b\)</span></td>
</tr>
<tr class="even">
<td align="right"><strong>Non differentially expressed</strong></td>
<td align="center"><span class="math inline">\(c\)</span></td>
<td align="center"><span class="math inline">\(d\)</span></td>
<td align="center"><span class="math inline">\(c + d\)</span></td>
</tr>
<tr class="odd">
<td align="right"><strong>Column total</strong></td>
<td align="center"><span class="math inline">\(a + c\)</span></td>
<td align="center"><span class="math inline">\(b + d\)</span></td>
<td align="center"><span class="math inline">\(a + b + c + d = n\)</span></td>
</tr>
</tbody>
</table></div>
<div class="section level4" number="7.2.1">
<h4 id="fishers-exact-test">
<span class="header-section-number">7.2.1</span> Fisher’s exact test<a class="anchor" aria-label="anchor" href="#fishers-exact-test"></a>
</h4>
<p>After that the contingency table has been defined, Fisher’s exact test p-value can be calculated through Equation <a href="#eq:fisher">(2)</a>.</p>
<p><span class="math display" id="eq:fisher">\[\begin{equation}
  p = \frac{(a + b)!\ (c + d)!\ (a + c)!\ (b + d)!}{a!\ b!\ c!\ d!\ n!}
  \tag{2}
\end{equation}\]</span></p>
<p>Additionally, it is also possible to compute Fisher’s p-values with <strong>Lancaster’s mid-p adjustment</strong>, since it has been proven that it increases statistical power while retaining Type I error rates.</p>
</div>
<div class="section level4" number="7.2.2">
<h4 id="boschloos-exact-test">
<span class="header-section-number">7.2.2</span> Boschloo’s exact test<a class="anchor" aria-label="anchor" href="#boschloos-exact-test"></a>
</h4>
<p>In contrast to Fisher’s exact test, a more appropriate method for the integrative analysis between miRNAs and genes is Boschloo’s exact test. Indeed, the major drawback of the Fisher’s exact test is that it consists in a conditional test that requires the sum of both rows and columns of a contingency table to be fixed. Notably, this is not true for genomic data because it is likely that different data sets may lead to a different number of DEGs. Therefore, the <strong>default</strong> behavior in MIRit is to use a variant of Barnard’s exact test, named <strong>Boschloo’s exact test</strong>, that is suitable when group sizes of contingency tables are variable. Moreover, it is possible to demonstrate that Boschloo’s test is uniformly more powerful compared to Fisher’s exact test. However, keep in mind that Boschloo’s test is much more computational intensive compared to Fisher’s exact test, and it may require some time.</p>
</div>
<div class="section level4" number="7.2.3">
<h4 id="perform-one-sideded-association-tests-in-mirit">
<span class="header-section-number">7.2.3</span> Perform one-sideded association tests in MIRit<a class="anchor" aria-label="anchor" href="#perform-one-sideded-association-tests-in-mirit"></a>
</h4>
<p>In MIRit, the <code><a href="../reference/mirnaIntegration.html">mirnaIntegration()</a></code> function automatically performs association tests for unpaired data when <code>test = "auto"</code>. Moreover, the type of association test to use can be specified through the <code>associationMethod</code> parameter, which can be set to:</p>
<ul>
<li>
<code>fisher</code>, to perform a simple one-sided Fisher’s exact test;</li>
<li>
<code>fisher-midp</code>, to perform a one-sided Fisher’s exact test with Lancaster’s mid-p correction;</li>
<li>
<code>boschloo</code>, to perform a one-sided Boschloo’s exact test (<em>default option</em>).</li>
</ul>
<p>For example, we could use the Boschloo’s exact to evaluate the inverse association between miRNA and gene expression values through a simple call to <code><a href="../reference/mirnaIntegration.html">mirnaIntegration()</a></code> function.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## perform a one-sided inverse association</span></span>
<span><span class="va">exp.association</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mirnaIntegration.html">mirnaIntegration</a></span><span class="op">(</span><span class="va">experiment</span>,</span>
<span>    test <span class="op">=</span> <span class="st">"association"</span>,</span>
<span>    associationMethod <span class="op">=</span> <span class="st">"boschloo"</span>,</span>
<span>    pAdjustment <span class="op">=</span> <span class="st">"none"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Finally, after performing the association analysis, results can be accessed through the <code><a href="../reference/integration.html">integration()</a></code> function in the same way as we can do for correlation analyses.</p>
</div>
</div>
<div class="section level3" number="7.3">
<h3 id="rotation-gene-set-tests-for-unpaired-data">
<span class="header-section-number">7.3</span> Rotation gene-set tests for unpaired data<a class="anchor" aria-label="anchor" href="#rotation-gene-set-tests-for-unpaired-data"></a>
</h3>
<p>Lastly, for unpaired data, the effect of DE-miRNAs on the expression of target genes can be estimated through rotation gene-set tests. In this approach, we want to evaluate for each miRNA whether its target genes tend to be differentially expressed in the opposite direction. In particular, a fast approximation to rotation gene-set testing called <code>fry</code>, implemented in the <em><a href="https://bioconductor.org/packages/3.17/limma" class="external-link">limma</a></em> package, can be used to statistically quantify the influence of miRNAs on expression changes of their target genes.</p>
<p>To perform the integrative analysis through rotation gene-set tests, we must simply set <code>test = "fry"</code> when calling <code><a href="../reference/mirnaIntegration.html">mirnaIntegration()</a></code> function.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## perform the integrative analysis through 'fry' method</span></span>
<span><span class="va">exp.fry</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mirnaIntegration.html">mirnaIntegration</a></span><span class="op">(</span><span class="va">experiment</span>,</span>
<span>    test <span class="op">=</span> <span class="st">"fry"</span>,</span>
<span>    pCutoff <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="7.4">
<h3 id="functional-enrichment-of-integrated-target-genes">
<span class="header-section-number">7.4</span> Functional enrichment of integrated target genes<a class="anchor" aria-label="anchor" href="#functional-enrichment-of-integrated-target-genes"></a>
</h3>
<p>Additionally, after finding miRNA-target pairs that appear to have an inverse association, we can try to identify the impaired biological functions as a result of miRNA dysregulations through ORA. To do this, MIRit provides the <code><a href="../reference/enrichTargets.html">enrichTargets()</a></code> function, which automatically performs ORA for the functional enrichment of target genes that result associated with differentially expressed miRNAs.</p>
<p>In our example, we are going to enrich the significantly anti-correlated targets that we have found in Section <a href="#correlation">7.1.2</a> through the Disease Ontology database.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## enrichment of integrated targets</span></span>
<span><span class="va">oraTarg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/enrichTargets.html">enrichTargets</a></span><span class="op">(</span><span class="va">experiment</span>, database <span class="op">=</span> <span class="st">"DO"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Preparing the appropriate gene set...</span></span>
<span><span class="co">#&gt; Some ID occurs one-to-many match, like "26476, 127068, 101060321"...</span></span>
<span><span class="co">#&gt; 99.06% genes are mapped to symbol</span></span>
<span><span class="co">#&gt; Performing the enrichment of upregulated genes...</span></span>
<span><span class="co">#&gt; Performing the enrichment of downregulated genes...</span></span>
<span><span class="co">#&gt; The enrichment of genes reported 113 significantly enriched terms for downregulated genes and 0 for upregulated genes.</span></span>
<span></span>
<span><span class="co">## show a dot plot for the enrichment of downregulated genes</span></span>
<span><span class="fu"><a href="../reference/enrichmentDotplot.html">enrichmentDotplot</a></span><span class="op">(</span><span class="va">oraTarg</span><span class="op">$</span><span class="va">downregulated</span>,</span>
<span>    showTerms <span class="op">=</span> <span class="fl">7</span>,</span>
<span>    showTermsParam <span class="op">=</span> <span class="st">"padj"</span>, title <span class="op">=</span> <span class="st">"Depleted diseases"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:intTargEnr"></span>
<img src="MIRit_files/figure-html/intTargEnr-1.png" alt="Functional enrichment of integrated targets. This dot plot shows the enriched diseases for downregulated genes." width="100%"><p class="caption">
Figure 7: <span class="caption-title">Functional enrichment of integrated targets</span><br>This dot plot shows the enriched diseases for downregulated genes.
</p>
</div>
<p>In Figure <a href="#fig:intTargEnr">7</a>, we appreciate the depletion of diseases where thyroid gland is overly active, such as goiter and hyperthyroidism, therefore suggesting the involvement of miRNAs in thyroid malfunctioning.</p>
</div>
</div>
<div class="section level2" number="8">
<h2 id="identify-the-impaired-mirna-mrna-regulatory-networks">
<span class="header-section-number">8</span> Identify the impaired miRNA-mRNA regulatory networks<a class="anchor" aria-label="anchor" href="#identify-the-impaired-mirna-mrna-regulatory-networks"></a>
</h2>
<p>Once the dysregulated miRNA-mRNA regulatory networks have been identified, the typical goal is to infer altered cellular processes and functions. To do so, MIRit introduces a novel approach named <strong>Topology-Aware Integrative Pathway Analysis (TAIPA)</strong>, which specifically focuses on detecting altered molecular networks in miRNA-mRNA multi-omic analyses by considering the topology of biological pathways and miRNA-mRNA interactions.</p>
<div class="section level3" number="8.1">
<h3 id="topologically-aware-integrative-pathway-analysis-taipa">
<span class="header-section-number">8.1</span> Topologically-Aware Integrative Pathway Analysis (TAIPA)<a class="anchor" aria-label="anchor" href="#topologically-aware-integrative-pathway-analysis-taipa"></a>
</h3>
<p>This analysis aims to identify the biological pathways that result affected by miRNA and mRNA dysregulations. In this analysis, biological pathways are retrieved from a pathway database such as KEGG, and the interplay between miRNAs and genes is then added to the networks. Each network is defined as a graph <span class="math inline">\(G(V, E)\)</span>, where <span class="math inline">\(V\)</span> represents nodes, and <span class="math inline">\(E\)</span> represents the relationships between nodes. Then, nodes that are not significantly differentially expressed are assigned a weight <span class="math inline">\(w_i = 1\)</span>, whereas differentially expressed nodes are assigned a weight <span class="math inline">\(w_i = \left| \Delta E_i \right|\)</span>, where <span class="math inline">\(\Delta E_i\)</span> is the linear fold change of the node. Moreover, to consider the biological interaction between two nodes, namely <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>, we define an interaction parameter <span class="math inline">\(\beta_{i \rightarrow j} = 1\)</span> for activation interactions and <span class="math inline">\(\beta_{i \rightarrow j} = -1\)</span> for repression interactions. Subsequently, the concordance coefficient <span class="math inline">\(\gamma_{i \rightarrow j}\)</span> is defined as in Equation <a href="#eq:gamma">(3)</a>:</p>
<p><span class="math display" id="eq:gamma">\[\begin{equation}
  \gamma_{i \rightarrow j} = \begin{cases} \beta_{i \rightarrow j}
&amp;\text{if } sign(\Delta E_i) = sign(\Delta E_j) \\ - \beta_{i \rightarrow j}
&amp;\text{if } sign(\Delta E_i) \not= sign(\Delta E_j) \end{cases}\,.
  \tag{3}
\end{equation}\]</span></p>
<p>Later in the process, a breadth-first search (BFS) algorithm is applied to topologically sort pathway nodes so that each individual node occurs after all its upstream nodes. Nodes within cycles are considered leaf nodes. At this point, a node score <span class="math inline">\(\phi\)</span> is calculated for each pathway node <span class="math inline">\(i\)</span> as in Equation <a href="#eq:phi">(4)</a>:</p>
<p><span class="math display" id="eq:phi">\[\begin{equation}
  \phi_i = w_i + \sum_{j=1}^{U} \gamma_{i \rightarrow j} \cdot k_j\,,
  \tag{4}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(U\)</span> represents the number of upstream nodes, <span class="math inline">\(\gamma_{i \rightarrow j}\)</span> denotes the concordance coefficient, and <span class="math inline">\(k_j\)</span> is a propagation factor defined as in Equation <a href="#eq:k">(5)</a>:</p>
<p><span class="math display" id="eq:k">\[\begin{equation}
  k_j = \begin{cases} w_j &amp;\text{if } \phi_j = 0 \\ \phi_j &amp;\text{if }
\phi_j \not = 0 \end{cases}\,.
  \tag{5}
\end{equation}\]</span></p>
<p>Finally, the pathway score <span class="math inline">\(\Psi\)</span> is calculated as in Equation <a href="#eq:Psi">(6)</a>:</p>
<p><span class="math display" id="eq:Psi">\[\begin{equation}
  \Psi = \frac{1 - M}{N} \cdot \sum_{i=1}^{N} \phi_i\,,
  \tag{6}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(M\)</span> represents the proportion of miRNAs in the pathway, and <span class="math inline">\(N\)</span> represents the total number of nodes in the network. Then, to compute the statistical significance of each pathway score, a permutation procedure is applied. Later, both observed pathway scores and permuted scores are standardized by subtracting the mean score of the permuted sets <span class="math inline">\(\mu_{\Psi_P}\)</span> and then dividing by the standard deviation of the permuted scores <span class="math inline">\(\sigma_{\Psi_P}\)</span>.</p>
<p>Finally, the p-value is defined based on the fraction of permutations that reported a higher normalized pathway score than the observed one. However, to prevent p-values equal to zero, we define p-values as in Equation <a href="#eq:pval">(7)</a>:</p>
<p><span class="math display" id="eq:pval">\[\begin{equation}
  p = \frac{\sum_{n=1}^{N_p} \left[ \Psi_{P_N} \ge \Psi_N \right] + 1}
{N_p + 1}\,.
  \tag{7}
\end{equation}\]</span></p>
<p>In the end, either p-values are corrected for multiple testing through the max-T procedure (default option) which is particularly suited for permutation tests, or through the standard multiple testing approaches.</p>
</div>
<div class="section level3" number="8.2">
<h3 id="perform-taipa-in-mirit">
<span class="header-section-number">8.2</span> Perform TAIPA in MIRit<a class="anchor" aria-label="anchor" href="#perform-taipa-in-mirit"></a>
</h3>
<p>Before performing TAIPA, we need to create miRNA-augmented networks. To do so, MIRit implements the <code><a href="../reference/preparePathways.html">preparePathways()</a></code> function, which automatically uses the <em><a href="https://bioconductor.org/packages/3.17/graphite" class="external-link">graphite</a></em> R package to download biological networks from multiple pathway databases, namely <code>KEGG</code>, <code>WikiPathways</code> and <code>Reactome</code>. Then, each pathway is converted to a <code>graph</code> object and significant miRNA-mRNA pairs are added to the network. Further, edge weights are included according to interaction type. After running this function, we obtain a <code>list</code> containing all the miRNA-augmented nwtworks as <code>graph</code> objects.</p>
<p>In our example, we want to use the significant miRNA-target pairs that we identified in Section <a href="#correlation">7.1.2</a> to augment biological pathways retrieved from the KEGG database.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## create miRNA-augmented networks using KEGG pathways</span></span>
<span><span class="va">networks</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preparePathways.html">preparePathways</a></span><span class="op">(</span><span class="va">experiment</span>,</span>
<span>    database <span class="op">=</span> <span class="st">"KEGG"</span>,</span>
<span>    organism <span class="op">=</span> <span class="st">"Homo sapiens"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>After running this function, pathways with less than 10% of nodes with expression measurements are removed. This option can be changed by specifying the <code>minPc</code> parameter.</p>
<p>Now, we are ready to perform TAIPA through the <code><a href="../reference/topologicalAnalysis.html">topologicalAnalysis()</a></code> function, which is used to calculate pathway scores for all the augmented networks and to evaluate their statistical significance through permutation tests.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## set seed for reproducible results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## perform TAIPA</span></span>
<span><span class="va">taipa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/topologicalAnalysis.html">topologicalAnalysis</a></span><span class="op">(</span><span class="va">experiment</span>,</span>
<span>    pathways <span class="op">=</span> <span class="va">networks</span>,</span>
<span>    nPerm <span class="op">=</span> <span class="fl">10000</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>As a result of the analysis, an object of class <code>IntegrativePathwayAnalysis</code> storing the results of TAIPA is returned. Notably, the user can change the behavior of the <code><a href="../reference/topologicalAnalysis.html">topologicalAnalysis()</a></code> in several ways. For example, the <code>pCutoff</code> and <code>pAdjustment</code> parameters can be used to change the significance threshold and the multiple testing correction method, respectively. Moreover, the <code>nPerm</code> parameter can be tweaked to change the number of permutations to use for evaluating statistical significance. In this regard, we recommend using at least 10000 permutations, with no less than 1000.</p>
</div>
<div class="section level3" number="8.3">
<h3 id="c-code-and-parallel-computing-with-biocparallel">
<span class="header-section-number">8.3</span> C++ code and parallel computing with <code>BiocParallel</code><a class="anchor" aria-label="anchor" href="#c-code-and-parallel-computing-with-biocparallel"></a>
</h3>
<p>For computational efficiency, pathway score computation has been implemented in C++ language. Furthermore, since computing pathway score for 10000 networks for each pathway is computationally intensive, parallel computing has been employed to reduce running time. The user can modify the parallel computing behavior by specifying the <code>BPPARAM</code> parameter. See the help page of the <em><a href="https://bioconductor.org/packages/3.17/BiocParallel" class="external-link">BiocParallel</a></em> package for further details. Both the <code><a href="../reference/preparePathways.html">preparePathways()</a></code> and the <code><a href="../reference/topologicalAnalysis.html">topologicalAnalysis()</a></code> functions accept the <code>BPPARAM</code> option.</p>
</div>
<div class="section level3" number="8.4">
<h3 id="visualize-the-significantly-affected-pathways">
<span class="header-section-number">8.4</span> Visualize the significantly affected pathways<a class="anchor" aria-label="anchor" href="#visualize-the-significantly-affected-pathways"></a>
</h3>
<p>After running the <code><a href="../reference/topologicalAnalysis.html">topologicalAnalysis()</a></code> function, we can inspect the significantly perturbed pathways contained in the <code>IntegrativePathwayAnalysis</code> object by using the <code><a href="../reference/integratedPathways.html">integratedPathways()</a></code> function, which returns a <code>data.frame</code> reporting the results of TAIPA.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## extract the results of TAIPA</span></span>
<span><span class="va">perturbedNetworks</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/integratedPathways.html">integratedPathways</a></span><span class="op">(</span><span class="va">taipa</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="8.5">
<h3 id="visualize-the-impaired-networks-within-biological-pathways">
<span class="header-section-number">8.5</span> Visualize the impaired networks within biological pathways<a class="anchor" aria-label="anchor" href="#visualize-the-impaired-networks-within-biological-pathways"></a>
</h3>
<p>As with functional enrichment analyses, we can plot perturbed miRNA-mRNA networks as dot plots. To do so, the <code><a href="../reference/integrationDotplot.html">integrationDotplot()</a></code> function can be used. In particular, we will graphically represent the most perturbed pathways in thyroid cancer.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## produce a dotplot that shows the most affected networks</span></span>
<span><span class="va">intDot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/integrationDotplot.html">integrationDotplot</a></span><span class="op">(</span><span class="va">taipa</span><span class="op">)</span></span>
<span><span class="va">intDot</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:integrationDot"></span>
<img src="MIRit_files/figure-html/integrationDot-1.png" alt="The perturbation of miRNA-mRNA networks in thyroid cancer. This dot plot display the impairment of the biological processes involved in the production of thyroid hormone, further highlighting the disruption of this mechanism in this disease." width="100%"><p class="caption">
Figure 8: <span class="caption-title">The perturbation of miRNA-mRNA networks in thyroid cancer</span><br>This dot plot display the impairment of the biological processes involved in the production of thyroid hormone, further highlighting the disruption of this mechanism in this disease.
</p>
</div>
<p>Finally, after identifying the impaired molecular networks, MIRit provides the possibility of exploring the molecular perturbations. In this concern, the <code>visualizeNetworks()</code> function can be used to visually represent the compromised pathways along with expression changes of both miRNAs and genes, so that users can easily interpret the functional consequences of miRNA and gene dysregulations. For example, we can explore the perturbed molecular events that are responsible for diminished production of thyroid hormone in thyroid cancer.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## plot the impaired network responsible for reduced TG synthesis</span></span>
<span><span class="fu"><a href="../reference/visualizeNetwork.html">visualizeNetwork</a></span><span class="op">(</span><span class="va">taipa</span>, <span class="st">"Thyroid hormone synthesis"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:thyroidNetwork"></span>
<img src="MIRit_files/figure-html/thyroidNetwork-1.png" alt="Impaired miRNA-mRNA regulatory network prevents thyroid hormone synthesis in thyroid cancer. The network created by MIRit suggests that the upregulation of miR-146b-5p and miR-146b-3p may be responsible for diminished expression of PAX8, which in turn causes reduced transcription of thyroid hormone." width="100%" class="widefigure"><p class="caption">
Figure 9: <span class="caption-title">Impaired miRNA-mRNA regulatory network prevents thyroid hormone synthesis in thyroid cancer</span><br>The network created by MIRit suggests that the upregulation of miR-146b-5p and miR-146b-3p may be responsible for diminished expression of PAX8, which in turn causes reduced transcription of thyroid hormone.
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<pre><code><span><span class="co">#&gt; R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 20.04.6 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=it_IT.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=it_IT.UTF-8        LC_COLLATE=it_IT.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=it_IT.UTF-8    LC_MESSAGES=it_IT.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=it_IT.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=it_IT.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Rome</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] fstcore_0.9.14   MIRit_0.99.0     BiocStyle_2.28.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] later_1.3.1                 splines_4.3.2              </span></span>
<span><span class="co">#&gt;   [3] genekitr_1.2.5              bitops_1.0-7               </span></span>
<span><span class="co">#&gt;   [5] ggplotify_0.1.2             urltools_1.7.3             </span></span>
<span><span class="co">#&gt;   [7] tibble_3.2.1                triebeard_0.4.1            </span></span>
<span><span class="co">#&gt;   [9] polyclip_1.10-4             graph_1.78.0               </span></span>
<span><span class="co">#&gt;  [11] lifecycle_1.0.3             rstatix_0.7.2              </span></span>
<span><span class="co">#&gt;  [13] edgeR_3.42.4                processx_3.8.2             </span></span>
<span><span class="co">#&gt;  [15] lattice_0.22-5              MASS_7.3-60                </span></span>
<span><span class="co">#&gt;  [17] MultiAssayExperiment_1.26.0 backports_1.4.1            </span></span>
<span><span class="co">#&gt;  [19] magrittr_2.0.3              openxlsx_4.2.5.2           </span></span>
<span><span class="co">#&gt;  [21] limma_3.56.2                sass_0.4.7                 </span></span>
<span><span class="co">#&gt;  [23] rmarkdown_2.25              remotes_2.4.2.1            </span></span>
<span><span class="co">#&gt;  [25] jquerylib_0.1.4             yaml_2.3.7                 </span></span>
<span><span class="co">#&gt;  [27] httpuv_1.6.11               zip_2.3.0                  </span></span>
<span><span class="co">#&gt;  [29] sessioninfo_1.2.2           pkgbuild_1.4.2             </span></span>
<span><span class="co">#&gt;  [31] ggvenn_0.1.10               cowplot_1.1.1              </span></span>
<span><span class="co">#&gt;  [33] DBI_1.1.3                   RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt;  [35] pkgload_1.3.3               abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [37] zlibbioc_1.46.0             quadprog_1.5-8             </span></span>
<span><span class="co">#&gt;  [39] GenomicRanges_1.52.0        purrr_1.0.2                </span></span>
<span><span class="co">#&gt;  [41] ggraph_2.1.0                BiocGenerics_0.46.0        </span></span>
<span><span class="co">#&gt;  [43] RCurl_1.98-1.12             yulab.utils_0.1.0          </span></span>
<span><span class="co">#&gt;  [45] tweenr_2.0.2                GenomeInfoDbData_1.2.10    </span></span>
<span><span class="co">#&gt;  [47] IRanges_2.34.1              S4Vectors_0.38.1           </span></span>
<span><span class="co">#&gt;  [49] enrichplot_1.20.3           ggrepel_0.9.3              </span></span>
<span><span class="co">#&gt;  [51] tidytree_0.4.5              codetools_0.2-19           </span></span>
<span><span class="co">#&gt;  [53] DelayedArray_0.26.7         DOSE_3.26.1                </span></span>
<span><span class="co">#&gt;  [55] xml2_1.3.5                  ggforce_0.4.1              </span></span>
<span><span class="co">#&gt;  [57] tidyselect_1.2.0            aplot_0.2.1                </span></span>
<span><span class="co">#&gt;  [59] farver_2.1.1                viridis_0.6.4              </span></span>
<span><span class="co">#&gt;  [61] matrixStats_1.0.0           stats4_4.3.2               </span></span>
<span><span class="co">#&gt;  [63] jsonlite_1.8.7              fst_0.9.8                  </span></span>
<span><span class="co">#&gt;  [65] ellipsis_0.3.2              tidygraph_1.2.3            </span></span>
<span><span class="co">#&gt;  [67] tools_4.3.2                 progress_1.2.2             </span></span>
<span><span class="co">#&gt;  [69] treeio_1.24.3               Rcpp_1.0.11                </span></span>
<span><span class="co">#&gt;  [71] glue_1.6.2                  gridExtra_2.3              </span></span>
<span><span class="co">#&gt;  [73] BiocBaseUtils_1.2.0         xfun_0.40                  </span></span>
<span><span class="co">#&gt;  [75] usethis_2.2.2               qvalue_2.32.0              </span></span>
<span><span class="co">#&gt;  [77] MatrixGenerics_1.12.3       GenomeInfoDb_1.36.3        </span></span>
<span><span class="co">#&gt;  [79] dplyr_1.1.3                 withr_2.5.0                </span></span>
<span><span class="co">#&gt;  [81] BiocManager_1.30.22         fastmap_1.1.1              </span></span>
<span><span class="co">#&gt;  [83] fansi_1.0.4                 callr_3.7.3                </span></span>
<span><span class="co">#&gt;  [85] digest_0.6.33               mime_0.12                  </span></span>
<span><span class="co">#&gt;  [87] R6_2.5.1                    gridGraphics_0.5-1         </span></span>
<span><span class="co">#&gt;  [89] colorspace_2.1-0            GO.db_3.17.0               </span></span>
<span><span class="co">#&gt;  [91] RSQLite_2.3.1               utf8_1.2.3                 </span></span>
<span><span class="co">#&gt;  [93] tidyr_1.3.0                 generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [95] data.table_1.14.8           htmlwidgets_1.6.2          </span></span>
<span><span class="co">#&gt;  [97] prettyunits_1.1.1           graphlayouts_1.0.1         </span></span>
<span><span class="co">#&gt;  [99] httr_1.4.7                  S4Arrays_1.2.0             </span></span>
<span><span class="co">#&gt; [101] scatterpie_0.2.1            pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt; [103] gtable_0.3.4                blob_1.2.4                 </span></span>
<span><span class="co">#&gt; [105] XVector_0.40.0              clusterProfiler_4.8.3      </span></span>
<span><span class="co">#&gt; [107] shadowtext_0.1.2            htmltools_0.5.7            </span></span>
<span><span class="co">#&gt; [109] carData_3.0-5               geneset_0.2.7              </span></span>
<span><span class="co">#&gt; [111] profvis_0.3.8               bookdown_0.35              </span></span>
<span><span class="co">#&gt; [113] fgsea_1.26.0                scales_1.2.1               </span></span>
<span><span class="co">#&gt; [115] Biobase_2.60.0              png_0.1-8                  </span></span>
<span><span class="co">#&gt; [117] ggfun_0.1.3                 knitr_1.45                 </span></span>
<span><span class="co">#&gt; [119] rstudioapi_0.15.0           reshape2_1.4.4             </span></span>
<span><span class="co">#&gt; [121] nlme_3.1-163                cachem_1.0.8               </span></span>
<span><span class="co">#&gt; [123] stringr_1.5.0               miniUI_0.1.1.1             </span></span>
<span><span class="co">#&gt; [125] parallel_4.3.2              HDO.db_0.99.1              </span></span>
<span><span class="co">#&gt; [127] AnnotationDbi_1.62.2        pillar_1.9.0               </span></span>
<span><span class="co">#&gt; [129] grid_4.3.2                  vctrs_0.6.3                </span></span>
<span><span class="co">#&gt; [131] urlchecker_1.0.1            promises_1.2.1             </span></span>
<span><span class="co">#&gt; [133] ggpubr_0.6.0                car_3.1-2                  </span></span>
<span><span class="co">#&gt; [135] xtable_1.8-4                Rgraphviz_2.44.0           </span></span>
<span><span class="co">#&gt; [137] evaluate_0.21               cli_3.6.1                  </span></span>
<span><span class="co">#&gt; [139] locfit_1.5-9.8              compiler_4.3.2             </span></span>
<span><span class="co">#&gt; [141] rlang_1.1.1                 crayon_1.5.2               </span></span>
<span><span class="co">#&gt; [143] ggsignif_0.6.4              labeling_0.4.3             </span></span>
<span><span class="co">#&gt; [145] ps_1.7.5                    plyr_1.8.8                 </span></span>
<span><span class="co">#&gt; [147] fs_1.6.3                    stringi_1.7.12             </span></span>
<span><span class="co">#&gt; [149] viridisLite_0.4.2           BiocParallel_1.34.2        </span></span>
<span><span class="co">#&gt; [151] munsell_0.5.0               Biostrings_2.68.1          </span></span>
<span><span class="co">#&gt; [153] lazyeval_0.2.2              devtools_2.4.5             </span></span>
<span><span class="co">#&gt; [155] GOSemSim_2.26.1             Matrix_1.6-3               </span></span>
<span><span class="co">#&gt; [157] europepmc_0.4.3             hms_1.1.3                  </span></span>
<span><span class="co">#&gt; [159] patchwork_1.1.3             bit64_4.0.5                </span></span>
<span><span class="co">#&gt; [161] ggplot2_3.4.3               shiny_1.7.5                </span></span>
<span><span class="co">#&gt; [163] KEGGREST_1.40.0             statmod_1.5.0              </span></span>
<span><span class="co">#&gt; [165] SummarizedExperiment_1.30.2 highr_0.10                 </span></span>
<span><span class="co">#&gt; [167] igraph_1.5.1                broom_1.0.5                </span></span>
<span><span class="co">#&gt; [169] memoise_2.0.1               bslib_0.6.1                </span></span>
<span><span class="co">#&gt; [171] ggtree_3.8.2                fastmatch_1.1-4            </span></span>
<span><span class="co">#&gt; [173] bit_4.0.5                   MonoPoly_0.3-10            </span></span>
<span><span class="co">#&gt; [175] downloader_0.4              ape_5.7-1                  </span></span>
<span><span class="co">#&gt; [177] gson_0.1.0</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-boyle_gotermfinderopen_2004" class="csl-entry">
Boyle, Elizabeth I., Shuai Weng, Jeremy Gollub, Heng Jin, David Botstein, J. Michael Cherry, and Gavin Sherlock. 2004. <span>“<span>GO</span>::<span>TermFinder</span>—Open Source Software for Accessing Gene Ontology Information and Finding Significantly Enriched Gene Ontology Terms Associated with a List of Genes.”</span> <em>Bioinformatics</em> 20 (18): 3710–15. <a href="https://doi.org/10.1093/bioinformatics/bth456" class="external-link">https://doi.org/10.1093/bioinformatics/bth456</a>.
</div>
<div id="ref-carvalho_framework_2010" class="csl-entry">
Carvalho, Benilton S., and Rafael A. Irizarry. 2010. <span>“A Framework for Oligonucleotide Microarray Preprocessing.”</span> <em>Bioinformatics</em> 26 (19): 2363–67. <a href="https://doi.org/10.1093/bioinformatics/btq431" class="external-link">https://doi.org/10.1093/bioinformatics/btq431</a>.
</div>
<div id="ref-hauschild_mirdip_2023" class="csl-entry">
Hauschild, Anne-Christin, Chiara Pastrello, Gitta Kirana Anindya Ekaputeri, Dylan Bethune-Waddell, Mark Abovsky, Zuhaib Ahmed, Max Kotlyar, Richard Lu, and Igor Jurisica. 2023. <span>“<span>MirDIP</span> 5.2: Tissue Context Annotation and Novel <span class="nocase">microRNA</span> Curation.”</span> <em>Nucleic Acids Research</em> 51 (January): D217–25. <a href="https://doi.org/10.1093/nar/gkac1070" class="external-link">https://doi.org/10.1093/nar/gkac1070</a>.
</div>
<div id="ref-hong_separate_2014-1" class="csl-entry">
Hong, Guini, Wenjing Zhang, Hongdong Li, Xiaopei Shen, and Zheng Guo. 2014. <span>“Separate Enrichment Analysis of Pathways for up- and Downregulated Genes.”</span> <em>Journal of The Royal Society Interface</em> 11 (92): 20130950. <a href="https://doi.org/10.1098/rsif.2013.0950" class="external-link">https://doi.org/10.1098/rsif.2013.0950</a>.
</div>
<div id="ref-huang_mirtarbase_2022" class="csl-entry">
Huang, Hsi-Yuan, Yang-Chi-Dung Lin, Shidong Cui, Yixian Huang, Yun Tang, Jiatong Xu, Jiayang Bao, et al. 2022. <span>“<span class="nocase">miRTarBase</span> Update 2022: An Informative Resource for Experimentally Validated <span class="nocase">miRNA</span>–Target Interactions.”</span> <em>Nucleic Acids Research</em> 50 (January): D222–30. <a href="https://doi.org/10.1093/nar/gkab1079" class="external-link">https://doi.org/10.1093/nar/gkab1079</a>.
</div>
<div id="ref-ramos_software_2017" class="csl-entry">
Ramos, Marcel, Lucas Schiffer, Angela Re, Rimsha Azhar, Azfar Basunia, Carmen Rodriguez, Tiffany Chan, et al. 2017. <span>“Software for the Integration of Multiomics Experiments in Bioconductor.”</span> <em>Cancer Research</em> 77 (21): e39–42. <a href="https://doi.org/10.1158/0008-5472.CAN-17-0344" class="external-link">https://doi.org/10.1158/0008-5472.CAN-17-0344</a>.
</div>
<div id="ref-riesco-eizaguirre_mir-146b-3ppax8nis_2015" class="csl-entry">
Riesco-Eizaguirre, Garcilaso, León Wert-Lamas, Javier Perales-Patón, Ana Sastre-Perona, Lara P. Fernández, and Pilar Santisteban. 2015. <span>“The <span class="nocase">miR</span>-146b-3p/<span>PAX</span>8/<span>NIS</span> Regulatory Circuit Modulates the Differentiation Phenotype and Function of Thyroid Cells During Carcinogenesis.”</span> <em>Cancer Research</em> 75 (19): 4119–30. <a href="https://doi.org/10.1158/0008-5472.CAN-14-3547" class="external-link">https://doi.org/10.1158/0008-5472.CAN-14-3547</a>.
</div>
<div id="ref-subramanian_gene_2005" class="csl-entry">
Subramanian, Aravind, Pablo Tamayo, Vamsi K. Mootha, Sayan Mukherjee, Benjamin L. Ebert, Michael A. Gillette, Amanda Paulovich, et al. 2005. <span>“Gene Set Enrichment Analysis: A Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles.”</span> <em>Proceedings of the National Academy of Sciences</em> 102 (43): 15545–50. <a href="https://doi.org/10.1073/pnas.0506580102" class="external-link">https://doi.org/10.1073/pnas.0506580102</a>.
</div>
<div id="ref-tokar_mirdip_2018" class="csl-entry">
Tokar, Tomas, Chiara Pastrello, Andrea E M Rossos, Mark Abovsky, Anne-Christin Hauschild, Mike Tsay, Richard Lu, and Igor Jurisica. 2018. <span>“<span class="nocase">mirDIP</span> 4.1—Integrative Database of Human <span class="nocase">microRNA</span> Target Predictions.”</span> <em>Nucleic Acids Research</em> 46 (January): D360–70. <a href="https://doi.org/10.1093/nar/gkx1144" class="external-link">https://doi.org/10.1093/nar/gkx1144</a>.
</div>
<div id="ref-wu_camera_2012" class="csl-entry">
Wu, Di, and Gordon K. Smyth. 2012. <span>“Camera: A Competitive Gene Set Test Accounting for Inter-Gene Correlation.”</span> <em>Nucleic Acids Research</em> 40 (17): e133. <a href="https://doi.org/10.1093/nar/gks461" class="external-link">https://doi.org/10.1093/nar/gks461</a>.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script><!-- tabsets --><script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script><!-- code folding --><script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script><script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script><script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": {
      styles: {
        ".MathJax_Display": {
           "text-align": "center",
           padding: "0px 150px 0px 65px",
           margin: "0px 0px 0.5em"
        },
        "@media screen and (max-width: 991px)": {
            ".MathJax_Display": {
               "text-align": "center",
               padding: "0 0 0 0"
            }
         }
      }
    }
  });
</script><script type="text/javascript">
$(document).ready(function ()  {
  
  // Map "enter" keypress to the same action as a cursor click
  function navigateLink(e) {
    if (e.key === "Enter") {
      $(this).trigger("click");
    }
  }

  var toc_items = document.querySelectorAll(".tocify-item");
  for (var i = 0; i < toc_items.length; i++) {
    // The link role tells screen readers this is for navigation
    toc_items.item(i).setAttribute("role", "link");
    // tabindex = 0 allows selection via keyboard tab presses
    toc_items.item(i).setAttribute("tabindex", "0");
    // Listen for "Enter" keypress when item is selected
    toc_items.item(i).addEventListener("keydown", navigateLink);
  }
});
</script>
</body>
</html>
